<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"md.zxzyl.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>


  <meta property="og:type" content="website">
<meta property="og:title" content="Zhongxu&#39;s blog">
<meta property="og:url" content="https://md.zxzyl.com/page/6/index.html">
<meta property="og:site_name" content="Zhongxu&#39;s blog">
<meta property="og:locale" content="en_US">
<meta property="article:author" content="Jason">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://md.zxzyl.com/page/6/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : true,
    isPost : false,
    lang   : 'en'
  };
</script>

  <title>Zhongxu's blog</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a3b5d3f466edb3e3f15e8238d8f04be8";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Zhongxu's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content index posts-expand">
            
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1043/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1043/" class="post-title-link" itemprop="url">SpliceMap官网示例教程</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2019-03-21 18:21:55" itemprop="dateCreated datePublished" datetime="2019-03-21T18:21:55+08:00">2019-03-21</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>SpliceMap是一个从头开始发现和比对splice junction的工具。它提供高敏感度并且支持任意长度RNA-seq 序列片段read长度. SpliceMap将RNA-seq reads比对到参考基因组上用于发现splicing junctions. 它至少拥有与当前技术条件下其它分析工具同等的灵敏度和特异性。</p>
<p>官网    <a target="_blank" rel="noopener" href="http://web.stanford.edu/group/wonglab/SpliceMap/manual.html">http://web.stanford.edu/group/wonglab/SpliceMap/manual.html</a></p>
<p>下载    <a target="_blank" rel="noopener" href="http://web.stanford.edu/group/wonglab/SpliceMap/download.html">http://web.stanford.edu/group/wonglab/SpliceMap/download.html</a></p>
<p>示例教程    <a target="_blank" rel="noopener" href="https://web.stanford.edu/group/wonglab/SpliceMap/tutorial.html">https://web.stanford.edu/group/wonglab/SpliceMap/tutorial.html</a></p>
<p>以官网SpliceMap 3.3.5.2 example (Linux-x86 64bit)为例，介绍如何用来自21号染色体的100k 条100bp的RNA reads寻找junction。</p>
<p>测试是否正常./bin/runSpliceMap，如果报错的话，需要到src文件夹运行 ./install.sh ../bin来安装SpliceMap，运行./install-bowtie.sh ../bin来安装</p>
<h1 id="示例文件夹下面的结构"><a href="#示例文件夹下面的结构" class="headerlink" title="示例文件夹下面的结构"></a>示例文件夹下面的结构</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ls SpliceMap3352_example_linux-64</span><br><span class="line">all.gene.refFlat.txt  bin  data  genome  INSTALL  LICENSE  output  run.cfg  src  temp</span><br></pre></td></tr></table></figure>




<h1 id="run-cfg"><a href="#run-cfg" class="headerlink" title="run.cfg"></a>run.cfg</h1><p>最重要的文件，为文本文件，包含测序reads路径，基因组文件路径，配置设置等详见<br><a target="_blank" rel="noopener" href="https://web.stanford.edu/group/wonglab/SpliceMap/cfg.html%EF%BC%8C%E9%9C%80%E8%A6%81%E5%AF%B9%E6%AF%8F%E4%B8%AA%E6%95%B0%E6%8D%AE%E9%9B%86%E8%BF%9B%E8%A1%8C%E9%85%8D%E7%BD%AE%E3%80%82%E5%9C%A8cfg%E6%96%87%E4%BB%B6%E4%B8%AD%EF%BC%8C%E6%B3%A8%E9%87%8A%E4%BB%A5#%E5%BC%80%E5%A4%B4%EF%BC%8C%E8%B5%8B%E5%80%BC%E7%94%A8tag">https://web.stanford.edu/group/wonglab/SpliceMap/cfg.html，需要对每个数据集进行配置。在cfg文件中，注释以#开头，赋值用tag</a> = value表示，比如genome_dir = ../../genome/hg18，多值用如下表示</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">&gt; tag</span><br><span class="line">value_1</span><br><span class="line">value_2</span><br><span class="line">...</span><br><span class="line">&lt;</span><br></pre></td></tr></table></figure>

<h1 id="genome-directory"><a href="#genome-directory" class="headerlink" title="genome directory"></a>genome directory</h1><p>该路径下包括物种的染色体，bowtie的索引，测试示例中仅放了21号染色体和相关bowtie的索引。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">ls genome                                                      </span><br><span class="line">chr21.1.ebwt  chr21.2.ebwt  chr21.3.ebwt  chr21.4.ebwt  chr21.fa  chr21.rev.1.ebwt  chr21.rev.2.ebwt</span><br></pre></td></tr></table></figure>


<h1 id="data-directory"><a href="#data-directory" class="headerlink" title="data directory"></a>data directory</h1><p>样本的测序reads，真实使用中可以是其他位置。</p>
<h1 id="temp-directory"><a href="#temp-directory" class="headerlink" title="temp directory"></a>temp directory</h1><p>在SpliceMap运行过程中的临时文件，最初的序列比对结果存储在这里，所以这个文件夹可能会很大。</p>
<h1 id="output-directory"><a href="#output-directory" class="headerlink" title="output directory"></a>output directory</h1><p>SpliceMap的输出结果文件夹。</p>
<h1 id="all-gene-refFlat-txt"><a href="#all-gene-refFlat-txt" class="headerlink" title="all.gene.refFlat.txt"></a>all.gene.refFlat.txt</h1><p>包含来自ensembl，refseq，knowGene数据所有基因的注释，可以用来寻找新的junction。</p>
<h1 id="bin"><a href="#bin" class="headerlink" title="bin"></a>bin</h1><p>SpliceMap的运行程序，不需要安装，可以拷贝到其他地方。</p>
<h1 id="src"><a href="#src" class="headerlink" title="src"></a>src</h1><p>包含SpliceMap/Bowtie源程序的文件夹。</p>
<h1 id="运行SpliceMap"><a href="#运行SpliceMap" class="headerlink" title="运行SpliceMap"></a>运行SpliceMap</h1><p>在example文件夹中，运行大概两到三分钟可以结束</p>
<pre>./bin/runSpliceMap run.cfg</pre>

<h1 id="输出结果文件夹output"><a href="#输出结果文件夹output" class="headerlink" title="输出结果文件夹output"></a>输出结果文件夹output</h1><h1 id="junction-color-bed"><a href="#junction-color-bed" class="headerlink" title="junction_color.bed"></a>junction_color.bed</h1><p>包含发现的junctions，novel junction在bed文件中用红色表示，可以用UCSC基因组浏览器或cisGenome浏览器看。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">track   name=junctions  description=&quot;SpliceMap junctions&quot; itemRgb=&quot;On&quot;</span><br><span class="line">chr21   14669961        14672440        (2)[49_2](2/0)  50      -       14669961        14672440        96,96,96        2 50,50    0,2429</span><br><span class="line">chr21   16125769        16127587        (1)[1_1](1/0)   50      +       16125769        16127587        192,192,192     2 50,50    0,1768</span><br></pre></td></tr></table></figure>

<h1 id="junction-color-new-bed"><a href="#junction-color-new-bed" class="headerlink" title="junction_color.new.bed"></a>junction_color.new.bed</h1><p>不在all.gene.refFlat.txt文件中的新的junction。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">track   name=junctions  description=&quot;SpliceMap junctions&quot; itemRgb=&quot;On&quot;</span><br><span class="line">chr21   17186728        17208745        (2)[6_2](2/0)   50      +       17186728        17208745        255,0,0 2       50,50      0,21967</span><br><span class="line">chr21   23658336        23666569        (1)[1_1](1/0)   50      -       23658336        23666569        255,192,192     2 50,50    0,8183</span><br></pre></td></tr></table></figure>

<h1 id="junction-nUM-color-bed"><a href="#junction-nUM-color-bed" class="headerlink" title="junction_nUM_color.bed"></a>junction_nUM_color.bed</h1><p>junction_color.bed文件中的至少被一条唯一比对的read支持的junction。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">track   name=junctions  description=&quot;SpliceMap junctions&quot; itemRgb=&quot;On&quot;</span><br><span class="line">chr21   14669961        14672440        (2)[49_2](2/0)  50      -       14669961        14672440        96,96,96        2 50,50    0,2429</span><br><span class="line">chr21   16125769        16127587        (1)[1_1](1/0)   50      +       16125769        16127587        192,192,192     2 50,50    0,1768</span><br></pre></td></tr></table></figure>

<h1 id="junction-nUM-color-new-bed"><a href="#junction-nUM-color-new-bed" class="headerlink" title="junction_nUM_color.new.bed"></a>junction_nUM_color.new.bed</h1><p>在junction_nUM_color.bed中且不在all.gene.refFlat.txt中的junction。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">track   name=junctions  description=&quot;SpliceMap junctions&quot; itemRgb=&quot;On&quot;</span><br><span class="line">chr21   17186728        17208745        (2)[6_2](2/0)   50      +       17186728        17208745        255,0,0 2       50,50      0,21967</span><br><span class="line">chr21   23658336        23666569        (1)[1_1](1/0)   50      -       23658336        23666569        255,192,192     2 50,50    0,8183</span><br></pre></td></tr></table></figure>

<h1 id="coverage-up-wig"><a href="#coverage-up-wig" class="headerlink" title="coverage_up.wig"></a>coverage_up.wig</h1><p>染色体位置中唯一比对的reads的覆盖度</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">track   type=bedGraph   name=&quot;SpliceMap unique coverage&quot;</span><br><span class="line">chr21   9944337 9944421 1</span><br><span class="line">chr21   14070536        14070619        1</span><br></pre></td></tr></table></figure>

<h1 id="coverage-down-wig"><a href="#coverage-down-wig" class="headerlink" title="coverage_down.wig"></a>coverage_down.wig</h1><p>染色体位置中多重比对的reads的覆盖度</p>
<pre>track   type=bedGraph   name="SpliceMap non-unique coverage"
chr21   16276385        16276484        -0.5
chr21   16277073        16277137        -0.5</pre>

<h1 id="coverage-all-wig"><a href="#coverage-all-wig" class="headerlink" title="coverage_all.wig"></a>coverage_all.wig</h1><p>所以比对上的reads在染色体上的覆盖度</p>
<pre>track   type=bedGraph   name="SpliceMap all coverage"
chr21   9944337 9944421 1
chr21   14070536        14070619        1</pre>

<h1 id="good-hits-sam"><a href="#good-hits-sam" class="headerlink" title="good_hits.sam"></a>good_hits.sam</h1><p>SAM格式存储的比对好的reads</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">R[87976]        73      chr21   9944337 255     85M     *       0       0       CATGTATCAACACCAAGTGCCATTTCTCTTAAGGATGCAAGGATGTTTTTTCCAAGATGGCAGATTACAGGCTTTTAGCATGCCT      IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII      NM:i:0  XC:i:15</span><br><span class="line">R[28405]        177     chr21   14070536        255     100M    =       14070620        -15     GCAGAGGCTATTATTCATGGACTATCCAGTCTAACAGCTTGCCAGTTGAGAACGATATACATATGTCAGTTTTTGACAAGAATTTCAGCAGGAAAAACCC       IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII       MD:Z:40A43G15   NM:i:2  XC:i:0</span><br><span class="line">R[28405]        113     chr21   14070620        255     100M    =       14070536        -15     GCAGCAGGAAAAACCCTTGATGCACAGTTTGAAAATGATGAACGAATTACACCCTTGGAATCAGCCCTGATGATTTGGGGTTCAATTGAAAAGGAACATG       IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII       MD:Z:42T42C14   NM:i:2  XC:i:0</span><br></pre></td></tr></table></figure>

<h1 id="debug-logs"><a href="#debug-logs" class="headerlink" title="debug_logs"></a>debug_logs</h1><p>输出log</p>
<h1 id="用自己的数据"><a href="#用自己的数据" class="headerlink" title="用自己的数据"></a>用自己的数据</h1><p>重点修改cfg文件中genome_dir （参考基因组路径），reads_list1和2（fastq文件），annotations （基因注释文件），temp_path（中间数据存放路径），out_path（结果文件夹），bowtie_base_dir （bowtie2的index文件，如果没有，会在temp_path生成）</p>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p><a target="_blank" rel="noopener" href="https://web.stanford.edu/group/wonglab/SpliceMap/tutorial.html">https://web.stanford.edu/group/wonglab/SpliceMap/tutorial.html</a><br>大部分翻译这个tutorial，具体数据是自己跑tutorial得到的。</p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1042/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1042/" class="post-title-link" itemprop="url">Pacbio三代测序Primary Analysis Data文件夹</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2019-03-20 18:08:34" itemprop="dateCreated datePublished" datetime="2019-03-20T18:08:34+08:00">2019-03-20</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>三代测序很多年了，刚工作的时候在超算中心做过三代的拼接，没好好研究过之后就再也没接触过，现在要做三代的项目，从头学习，Primary Analysis Data为初步数据分析文件夹，类似下面的文件夹结构</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">/path/to/secondary/storage/2420294/0011</span><br><span class="line">├── Analysis_Results</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.1.bax.h5</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.1.log</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.1.subreads.fasta</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.1.subreads.fastq</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.2.bax.h5</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.2.log</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.2.subreads.fasta</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.2.subreads.fastq</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.3.bax.h5</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.3.log</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.3.subreads.fasta</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.3.subreads.fastq</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.bas.h5</span><br><span class="line">│   ├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.sts.csv</span><br><span class="line">│   └── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.sts.xml</span><br><span class="line">├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.1.xfer.xml</span><br><span class="line">├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.2.xfer.xml</span><br><span class="line">├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.3.xfer.xml</span><br><span class="line">├── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.mcd.h5</span><br><span class="line">└── m140415_143853_42175_c100635972550000001823121909121417_s1_p0.metadata.xml</span><br></pre></td></tr></table></figure>




<p><strong>主要文件有</strong></p>
<h1 id="bas-h5文件和bax-h5文件"><a href="#bas-h5文件和bax-h5文件" class="headerlink" title="bas.h5文件和bax.h5文件"></a>bas.h5文件和bax.h5文件</h1><p>bas.h5和相关的bax.h5文件是PacBio@RS II初级分析（primary analysis）的主要输出文件，这些文件由设备产生到本地存储位置，作为后续SMRT分析软件进行alignment、consensus和variant分析的输入文件。<br>PacBio@RS II之前，单个bas.h5文件包含了所有测序数据，随着PacBio@RS II升级，通量和read长度都在增加，现在包含一个bas.h5和3个bax.h5文件（1-3.bax.h5）。bax.h5文件包含测序的base call的信息，bas.h5是三个bax.h5的重要指针。<br>用h5dupm -n [movie name].bas.h5命令看一下文件</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">FILE_CONTENTS &#123;</span><br><span class="line"> group      /</span><br><span class="line"> group      /MultiPart</span><br><span class="line"> dataset    /MultiPart/HoleLookup</span><br><span class="line"> dataset    /MultiPart/Parts</span><br><span class="line"> &#125;</span><br></pre></td></tr></table></figure>



<p>可以看到有一个MultiPart的group组，下面有两个数据集，一个是HoleLookup，一个是Parts。<br>/MultiPart/Parts提供相关对应的的三个bax.h5的文件名<br>/MultiPart/HoleLookup对应ZMW零模波导孔编号</p>
<p>单个的bax.h5文件结构和原来一致 ，有两个组，分别是PulseData和ScanData。ScanData<br>包含了用于测试的仪器信息，一般不会用到。PulseData下面重点看/PulseData/BaseCalls组<br>用命令h5dump -d /PulseData/BaseCalls/Basecall [movie name].bax.h5 ‘ head -10 查看数据集basecall如下序列</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">DATASET &quot;/PulseData/BaseCalls/Basecall&quot; &#123;</span><br><span class="line">   DATATYPE  H5T_STD_U8LE</span><br><span class="line">   DATASPACE  SIMPLE &#123; ( 305612594 ) / ( H5S_UNLIMITED ) &#125;</span><br><span class="line">   DATA &#123;</span><br><span class="line">   (0): 84, 84, 67, 84, 84, 67, 84, 84, 84, 84, 67, 71, 84, 71, 71, 84, 84,</span><br><span class="line">   (17): 84, 84, 71, 84, 67, 84, 84, 84, 67, 84, 71, 84, 84, 84, 67, 84, 84,</span><br><span class="line">   (34): 84, 67, 84, 84, 71, 84, 84, 67, 84, 71, 67, 71, 71, 84, 84, 84, 84,</span><br><span class="line">   (51): 84, 84, 84, 84, 84, 84, 84, 84, 84, 84, 84, 84, 84, 84, 84, 84, 84,</span><br><span class="line">   (68): 84, 84, 84, 84, 71, 84, 84, 71, 84, 84, 84, 84, 84, 84, 84, 84, 84,</span><br></pre></td></tr></table></figure>




<p>其中A=&gt; 65, C=&gt; 67, G=&gt;71, T=&gt;84，与ASCII编码一致。</p>
<p>用h5dump -d /PulseData/BaseCalls/QualityValue bax.h5 ‘ head -10 查看数据集qualityvalue如下</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">DATASET &quot;/PulseData/BaseCalls/QualityValue&quot; &#123;</span><br><span class="line">   DATATYPE  H5T_STD_U8LE</span><br><span class="line">   DATASPACE  SIMPLE &#123; ( 305612594 ) / ( H5S_UNLIMITED ) &#125;</span><br><span class="line">   DATA &#123;</span><br><span class="line">   (0): 10, 11, 21, 12, 11, 9, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 14, 0, 0, 0, 10, 0, 1,</span><br><span class="line">   (23): 0, 0, 15, 1, 1, 1, 0, 1, 1, 1, 1, 0, 2, 10, 0, 0, 0, 0, 0, 0, 14, 20, 1,</span><br><span class="line">   (46): 1, 30, 1, 0, 0, 1, 22, 0, 19, 0, 1, 0, 0, 21, 1, 1, 1, 0, 0, 17, 0, 0, 1,</span><br><span class="line">   (69): 1, 1, 30, 2, 1, 0, 2, 1, 0, 10, 0, 1, 1, 0, 1, 1, 13, 0, 0, 0, 1, 0, 0,</span><br><span class="line">   (92): 1, 0, 5, 1, 0, 0, 1, 0, 2, 1, 0, 0, 21, 0, 1, 0, 0, 20, 0, 0, 0, 16, 1,</span><br></pre></td></tr></table></figure>




<p>表示测错的概率，用的是phred socre，和Illumina类似。当然，我看的这个文件可以看出测序质量并不是很好。</p>
<p>具体的组和数据集的说明，可以参考：</p>
<p><a target="_blank" rel="noopener" href="https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/bas.h5+Reference+Guide.pdf">https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/bas.h5+Reference+Guide.pdf</a></p>
<h1 id="metadata-xml"><a href="#metadata-xml" class="headerlink" title="metadata.xml"></a>metadata.xml</h1><p>包含测序仪ID，样本名，测序酶，试剂等元数据。<br>参考文件：<a target="_blank" rel="noopener" href="https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/Metadata+Output+Guide.pdf">https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/Metadata+Output+Guide.pdf</a></p>
<h1 id="sts-xml"><a href="#sts-xml" class="headerlink" title="sts.xml"></a>sts.xml</h1><p>包含了单个movie的概括统计<br>参考文件：<a target="_blank" rel="noopener" href="https://s3.amazonaws.com/files.pacb.com/software/instrument/1.3.1/Statistics+Output+Guide.pdf">https://s3.amazonaws.com/files.pacb.com/software/instrument/1.3.1/Statistics+Output+Guide.pdf</a></p>
<h1 id="用h5dump查看pacbio数据"><a href="#用h5dump查看pacbio数据" class="headerlink" title="用h5dump查看pacbio数据"></a>用h5dump查看pacbio数据</h1><p>查看h5格式的文件可以用hdf5下的h5dump工具查看,安装如下</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget -c https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.5/src/hdf5-1.10.5.tar.gz</span><br><span class="line">tar -xzvf hdf5-1.10.5.tar.gz</span><br><span class="line">cd hdf5-1.10.5  </span><br><span class="line">./configure &amp;&amp;make &amp;&amp; make install</span><br></pre></td></tr></table></figure>




<h1 id="参考："><a href="#参考：" class="headerlink" title="参考："></a>参考：</h1><p><a target="_blank" rel="noopener" href="https://www.cnblogs.com/xudongliang/p/6908953.html">https://www.cnblogs.com/xudongliang/p/6908953.html</a><br><a target="_blank" rel="noopener" href="https://smrt-analysis.readthedocs.io/en/latest/Data-files-you-received-from-your-service-provider/">https://smrt-analysis.readthedocs.io/en/latest/Data-files-you-received-from-your-service-provider/</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1032/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1032/" class="post-title-link" itemprop="url">EBI提供HLA序列BLAST</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-09-14 15:08:57" itemprop="dateCreated datePublished" datetime="2018-09-14T15:08:57+08:00">2018-09-14</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>基于我们有的HLA序列，可以和HLA序列的数据库比较，看与哪个HLA allele最相似。</p>
<p>HLA （human leukocyte antigen，人类白细胞抗原）是人类主要组织相容性复合体（major histocompatibility complex，MHC）的表达产物，根据HLA抗原结构、功能及组织分布的不同，分为I类，II类，III类分子，其中I类分子包括HLA-A，-B，-C系列抗原，广泛分布于各组织有核系统表面。</p>
<p>BLAST表示局部比对搜索工具，用来将新的序列与已有的数据库中的序列进行比较，可以发现区域的相似性，进而为功能和进化研究提供线索。<br>EBI（欧洲生物信息学中心）提供基于IPD-IMGT/HLA（IMGT国际免疫遗传学数据库）数据库的BLAST库。BLAST工具会搜索数据库中的HLA allele的核苷酸、蛋白质及相关对的序列。</p>
<p>HLA BLAST在线服务的链接如下：<br><a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/Tools/services/web_ncbiblast/toolform.ebi?tool=ncbiblast&context=nucleotide&database=imgthla">https://www.ebi.ac.uk/Tools/services/web_ncbiblast/toolform.ebi?tool=ncbiblast&amp;context=nucleotide&amp;database=imgthla</a></p>
<h2 id="1）选择数据库"><a href="#1）选择数据库" class="headerlink" title="1）选择数据库"></a>1）选择数据库</h2><p><img src="/wp/f4w/2020/2018-09-14-EBI-EMBL-HLA-BLAST1.png"> </p>
<p>选择IMGT下的IMGT/HLA (genomic)数据据，表示与HLA的基因组序列进行比对，如果需要和编码区序列进行比对，可以选择IMGT/HLA (cds)。</p>
<h2 id="2）输入待比对的序列"><a href="#2）输入待比对的序列" class="headerlink" title="2）输入待比对的序列"></a>2）输入待比对的序列</h2><p>BLAST支持简并碱基的比对，比如用R表示G或A (puRine)，用A表示M表示A或C。</p>
<h2 id="3）设置参数"><a href="#3）设置参数" class="headerlink" title="3）设置参数"></a>3）设置参数</h2><p><img src="/wp/f4w/2020/2018-09-14-EBI-EMBL-HLA-BLAST3.png"></p>
<p>选择blastn表示进行核苷酸序列比对，task可以用megablast也可以用blastn，其他参数可以选择默认。</p>
<h2 id="4）提交任务"><a href="#4）提交任务" class="headerlink" title="4）提交任务"></a>4）提交任务</h2><p>点击submit提交任务。</p>
<h2 id="5）等待结果"><a href="#5）等待结果" class="headerlink" title="5）等待结果"></a>5）等待结果</h2><p>一般情况下，会在10秒左右进入比对结果页面。结果页面包含相似性和显著性结果。</p>
<h2 id="测试序列"><a href="#测试序列" class="headerlink" title="测试序列"></a>测试序列</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">CGAACTTGGCGGGTCTCAGCCCTCCTSGCCCCAGGCTCCCACTCCATGAGGTATTTCTACACCGCCATGTCCCGGCCCGG</span><br><span class="line">CCGCGGGGAGCCCCGCTTCATCRCMGTGGGCTACGTGGACGACACSCAGTTCGTGMGGTTCGACAGCGACGCCRCGAGTC</span><br><span class="line">CGAGRRKGGMGCCSCGGGCGCCRTGGRTRGAGCAGGAGGGGCCGGAGTATTGGGACCGGAASACACAGATCTMCAAGGCC</span><br><span class="line">MASGCACAGACTGACCGAGAGAACCTGCGSAACCTGCTCGGCTACTACAACCAGAGCGAGGCCGGTGAGTGACCCCGGCC</span><br><span class="line">CGGGGCGCAGGTCACGACTCCCCATCCCCCACGKACGGCCCGGGTCGCCCCGAGTCTCCGGGTCCGAGATCCGCCTCCCT</span><br><span class="line">GAGGCCGCGGGACCCGCCCAGACCCTCGACCGGCGAGAGCCCCAGGCGCGTTTACCCGGTTTCATTTTCAGTTGAGGCCA</span><br><span class="line">AAATCCCCGCGGGTTGGTCGGGGCGGGGCGGGGCTCGGGGGACGGGGCTGACCGCGGGGCCGGGGCCAGGGTCTCACACT</span><br><span class="line">TGGCAGACGATGTATGGCTGCGACCTGGGGCCGGACGGGCGCCTCCTCCGCGGGCATAACCAGTTAGCTACGACGGCAGG</span><br></pre></td></tr></table></figure>




<p>该序列的测试结果，与本地blast的结果一致。</p>
<h2 id="参考："><a href="#参考：" class="headerlink" title="参考："></a>参考：</h2><p><a target="_blank" rel="noopener" href="https://www.ebi.ac.uk/ipd/imgt/hla/blast.html">https://www.ebi.ac.uk/ipd/imgt/hla/blast.html</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1030/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1030/" class="post-title-link" itemprop="url">将VCF文件中的突变拆分成SNP和INDEL</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-09-10 11:14:45" itemprop="dateCreated datePublished" datetime="2018-09-10T11:14:45+08:00">2018-09-10</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <h1 id="VCFTOOLS"><a href="#VCFTOOLS" class="headerlink" title="VCFTOOLS"></a>VCFTOOLS</h1><p>得到SNP</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">vcftools --vcf X.vcf --remove-indels --out X.snps --recode --recode-INFO-all</span><br></pre></td></tr></table></figure>

<p>得到INDEL</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">vcftools --vcf X.vcf --keep-only-indels --out X.indel --recode --recode-INFO-all</span><br></pre></td></tr></table></figure>



<h1 id="GATK"><a href="#GATK" class="headerlink" title="GATK"></a>GATK</h1><p>得到SNP</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">java -jar GenomeAnalysisTK.jar \</span><br><span class="line">    -T SelectVariants \</span><br><span class="line">    -R reference.fasta \</span><br><span class="line">    -V X.vcf   \</span><br><span class="line">    -selectType SNP \</span><br><span class="line">    -o X.snps.vcf</span><br></pre></td></tr></table></figure>




<p>得到INDEL</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">java -jar GenomeAnalysisTK.jar \</span><br><span class="line">    -T SelectVariants \</span><br><span class="line">    -R reference.fasta \</span><br><span class="line">    -V X.vcf  \</span><br><span class="line">    -selectType INDEL \</span><br><span class="line">    -o X.indel.vcf</span><br></pre></td></tr></table></figure>




<p>本着有轮子不造轮子的原则，可以用VCFTOOLS和GATK来实现，当然如果想自己拆分的话，可以根据VCF中是否有SNP和INDEL的tag标签，或者根据ALT和REF中的碱基长度是否一致来实现拆分。</p>
<p>参考：</p>
<p><a target="_blank" rel="noopener" href="https://www.biostars.org/p/48204/">https://www.biostars.org/p/48204/</a><br><a target="_blank" rel="noopener" href="https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.php">https://software.broadinstitute.org/gatk/documentation/tooldocs/3.8-0/org_broadinstitute_gatk_tools_walkers_variantutils_SelectVariants.php</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1026/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1026/" class="post-title-link" itemprop="url">分析带UMI标签的测序数据</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-07-31 12:50:00" itemprop="dateCreated datePublished" datetime="2018-07-31T12:50:00+08:00">2018-07-31</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>分析带UMI标签的测序数据</p>
<p>检测癌组织的低频突变，为了提高检测低频突变的灵敏度，往往进行高深度的测序。但样本之间存在交叉污染，测序有存在一定概率的错误，这些因素会导致高深度测序过程中将假阳性的信号放到，得到假阳性的结果。解决交叉污染的方法，有公司比如IDT采用唯一配对的样本index，只有配对的index中的reads才属于特定样本。解决测序错误的方法，研究人员在建库的时候，先对分子加上UMI碱基，unique molecular identifier -&gt; UMI，然后根据来源于同一个分子的测序数据进行测序错误修正，得到正确的分子序列。两种方法结合可以减少交叉污染提高准确性（<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5759201/%EF%BC%89%E3%80%82">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5759201/）。</a></p>
<p><img src="/wp/f4w/2020/2018-07-30-UMI-analysis.png"></p>
<p>如图中所示，左侧一个分子被测了5次，其中第二次有一个测序错误，但该错误并没有在每个测序数据中出现，所以在后续合成一个分子的时候，测序错误被修正，只保留了真正的突变。（<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5852328/%EF%BC%89">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5852328/）</a></p>
<p>常规的肿瘤配对测序分析，或者遗传性突变位点的分析，并不需要UMI信息，所以包含UMI的数据分析是需要不一样的分析流程来得到准确的分析结果，其中包括提取UMI分子标签，合并来自同一个分子的测序reads，低频突变检测而非胚系突变检测等。</p>
<p>大致流程为：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">-----prepare analysis ready BAM file------</span><br><span class="line">&#x27;         FASTQ</span><br><span class="line">&#x27;            ↓</span><br><span class="line">&#x27;          uBAM</span><br><span class="line">&#x27;            ↓  Extract UMI</span><br><span class="line">&#x27;          uBAM</span><br><span class="line">&#x27;             ↓  Align uBAM and merge </span><br><span class="line">&#x27;           BAM</span><br><span class="line">-----call consensus------</span><br><span class="line">&#x27;             ↓  Group Reads By Umi</span><br><span class="line">&#x27;           BAM</span><br><span class="line">&#x27;             ↓  Group Reads By Umi</span><br><span class="line">&#x27;           BAM</span><br><span class="line">&#x27;             ↓  Call Molecular Consensus Reads</span><br><span class="line">&#x27;           uBAM</span><br><span class="line">&#x27;             ↓  Align uBAM and merge</span><br><span class="line">&#x27;           BAM</span><br><span class="line">&#x27;             ↓  Filter Consensus Reads</span><br><span class="line">&#x27;           BAM</span><br><span class="line">&#x27;             ↓  Clip</span><br><span class="line">&#x27;           BAM</span><br><span class="line">-----Vardict------</span><br><span class="line">&#x27;             ↓  Call</span><br><span class="line">&#x27;            VCF</span><br></pre></td></tr></table></figure>




<h1 id="一，得到包含UMI分子标签信息的BAM文件"><a href="#一，得到包含UMI分子标签信息的BAM文件" class="headerlink" title="一，得到包含UMI分子标签信息的BAM文件"></a>一，得到包含UMI分子标签信息的BAM文件</h1><p>UMI信息，应该从fastq中的配对的reads中提取，但fastq不能存储更多的信息，所以需要先将fastq转成uBAM文件，提取uBAM文件中的UMI分子标签信息，将该信息通过RX标签写入uBAM文件中，通过uBAM和BAM文件合并，把RX信息合并到比对到的BAM文件中进行下一步分析。</p>
<h2 id="1）生成uBAM"><a href="#1）生成uBAM" class="headerlink" title="1）生成uBAM"></a>1）生成uBAM</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">java -Xmx8G -jar picard.jar FastqToSam \</span><br><span class="line">    FASTQ=$fq1    \</span><br><span class="line">    FASTQ2=$fq2   \</span><br><span class="line">    OUTPUT=test.uBAM \</span><br><span class="line">    READ_GROUP_NAME=test \</span><br><span class="line">    SAMPLE_NAME=test \</span><br><span class="line">    LIBRARY_NAME=test \</span><br><span class="line">    PLATFORM_UNIT=HiseqX10  PLATFORM=illumina \</span><br><span class="line">    RUN_DATE=`date --iso-8601=seconds`</span><br></pre></td></tr></table></figure>




<h2 id="2）提取UMI信息"><a href="#2）提取UMI信息" class="headerlink" title="2）提取UMI信息"></a>2）提取UMI信息</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">java -jar fgbio.jar ExtractUmisFromBAM \</span><br><span class="line">    --input=test.uBAM  \</span><br><span class="line">    --output=test.umi.uBAM \</span><br><span class="line">    --read-structure=2M148T 2M148T \</span><br><span class="line">    --single-tag=RX \</span><br><span class="line">    --molecular-index-tags=ZA ZB</span><br></pre></td></tr></table></figure>




<p>此时，uBAM文件中RX标签记录着UMI的信息，2M148T表示前两个碱基是UMI分子标签，148个template 测序序列，配对read在uBAM文件中有如下信息</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">R1----ZA:Z:TC ZB:Z:TA RG:Z:test       RX:Z:TC-TA</span><br><span class="line">R2----ZA:Z:TC ZB:Z:TA RG:Z:test       RX:Z:TC-TA</span><br></pre></td></tr></table></figure>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/1026/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1016/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1016/" class="post-title-link" itemprop="url">什么是ubam文件，为什么ubam文件比fastq文件好</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-07-12 15:00:46" itemprop="dateCreated datePublished" datetime="2018-07-12T15:00:46+08:00">2018-07-12</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>ubam是Unmapped BAM Format，是BAM文件的一个变种，里面的read是未经map的。大部分测序供应FASTQ文件，这是最常见的测序分析的起始文件。FASTQ文件的优势是，压缩比比bam文件好，解压速度快。</p>
<p>但与ubam文件相比，FASTQ并不是最理想的：</p>
<h1 id="1）单个文件更容易分析"><a href="#1）单个文件更容易分析" class="headerlink" title="1）单个文件更容易分析"></a>1）单个文件更容易分析</h1><p>在双端测序中，有些软件希望配对的reads放在一个文件中，有些希望配对的文件，有些软件直接根据read在文件中的位置判断read是否配对，当然现在FASTQ通过在文件中加入/1和/2来表明配对的read解决这一问题。在生信分析的时候，一个FASTQ文件往往和另一个FASTQ文件关联配对，比如R1和R2，往往会花费更多时间来验证read是否配对。但如果通过bam文件的话，会更简单。只需要在FLAG这个地方加入对应的值，比如77和141，就能指定。单个文件更加简单，能储存更多metadata的信息。</p>
<h1 id="2）BAM文件可以储存更多信息"><a href="#2）BAM文件可以储存更多信息" class="headerlink" title="2）BAM文件可以储存更多信息"></a>2）BAM文件可以储存更多信息</h1><p>实际上，正是这个原因，我才开始用ubam文件的。<br>因为FASTQ文件格式的问题，很多其他信息不能存在FASTQ文件中，而BAM/SAM文件可以储存很多信息，只需加tag或者header就行。如果你想放，可以放用了什么软件，平台，样本标签等。</p>
<p><img src="/wp/f4w/2020/2018-07-12-advantage-ubam-1.jpg"></p>
<p>上图是不利用ubam文件的流程，下图是利用ubam文件的流程。</p>
<p><img src="/wp/f4w/2020/2018-07-12-advantage-ubam-2.png"> </p>
<p>这个我体会很深，比如我在分析带分子标签UMI的数据时，需要在bam文件中添加read的UMI标签序列，具有相同的UMI标签的序列说明来自同一个分子。如果我将带UMI标签的reads直接比对基因组，UMI标签序列会影响比对质量，而且后续对bam文件的处理提取UMI时，需要考虑CIGAR标签，这个时候就头大了。实际上，我是先将fastq转成ubam文件（picard的FastqToSam工具），然后提取ubam文件中的UMI，生成新的ubam，将ubam转成fastq文件（reads都放在一个文件fastq文件中，bwa能识别其中的配对reads），与参考基因组进行比对。这个时候问题就来了，比对之后生成的bam文件，不包含我提取的UMI标签序列信息，因为这个信息在ubam文件中。不要着急，picard的另外一个工具MergeBamAlignment，可以将ubam文件和bam文件合并，这样ubam文件中的tag信息，比如UMI标签信息，就会补充到新的bam文件中。这样生成的bam文件供下一步数据分析。</p>
<h1 id="3）BAM能实现随机读写"><a href="#3）BAM能实现随机读写" class="headerlink" title="3）BAM能实现随机读写"></a>3）BAM能实现随机读写</h1><p>BAM文件采用的是BGZF压缩，可以通过随机读写快速读取文件特定位置的数据。对于未比对的reads（unaligned reads），这个功能不是很吸引人，因为后续序列比对都是顺序比对的，不可能跳着比对。</p>
<p>参考：<br><a target="_blank" rel="noopener" href="https://blastedbio.blogspot.com/2011/10/fastq-must-die-long-live-sambam.html">https://blastedbio.blogspot.com/2011/10/fastq-must-die-long-live-sambam.html</a></p>
<p><a target="_blank" rel="noopener" href="https://gatkforums.broadinstitute.org/gatk/discussion/5990/what-is-ubam-and-why-is-it-better-than-fastq-for-storing-unmapped-sequence-data">https://gatkforums.broadinstitute.org/gatk/discussion/5990/what-is-ubam-and-why-is-it-better-than-fastq-for-storing-unmapped-sequence-data</a></p>
<p><a target="_blank" rel="noopener" href="https://gatkforums.broadinstitute.org/gatk/discussion/11694/why-is-converting-from-fastq-to-ubam-nesessary-before-preprocessing">https://gatkforums.broadinstitute.org/gatk/discussion/11694/why-is-converting-from-fastq-to-ubam-nesessary-before-preprocessing</a></p>
<p><a target="_blank" rel="noopener" href="https://gatkforums.broadinstitute.org/gatk/discussion/6484/how-to-generate-an-unmapped-bam-from-fastq-or-aligned-bam">https://gatkforums.broadinstitute.org/gatk/discussion/6484/how-to-generate-an-unmapped-bam-from-fastq-or-aligned-bam</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1011/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1011/" class="post-title-link" itemprop="url">fastq压缩之后的gzip文件大小与样本数据量</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-06-20 14:10:10" itemprop="dateCreated datePublished" datetime="2018-06-20T14:10:10+08:00">2018-06-20</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>在测序的时候，我们先拿到的是样本fastq压缩后的gzip文件，这个时候可能最关心的是数据量够不够，那么fastq.gz文件大小和测序数据量有什么关系呢。</p>
<p>我用Miseq测序数据（gz文件200M左右），Hiseq panel（gz文件50M左右）和WES测序数据（gz文件4G左右）进行了简单的分析。有意思的地方是，虽然R1和R2的数据量是一样的，解压出来的文件大小是一样的，但R2的gzip文件总比R1大。不管是Miseq还是Hiseq的panel测序，压缩后的R2均大于R1文件，且文件越小，差异越大。</p>
<p><img src="/wp/f4w/2020/2018-06-20-different-ratio-R1R2.png"></p>
<p>因为R1和R2数据量相同的原因，我只看R1的真实文件和gz文件大小与数据量之间的关系。<br>数据量=FASTQ文件行数/4*151/1000/1000  单位为M<br>真实文件大小估计=FASTQ文件行数/4*357/1024/1024 单位为M，预测值，差别不大，因为FASTQ文件中每四行357个字符（和平台和设置有关系），每个字符1byte。<br>GZ文件大小通过ll -h查看</p>
<p>因为FASTQ文件是规范的，每四行字符基本一致，所以FASTQ真实文件大小和数据量成正比。比如我前面提到的每四行有357个字符，其中序列只占151个字符，也就是说FASTQ文件大小大概是测序量的357/151≈2.3倍多。但因为FASTQ文件为文本文件，占用空间较大，所以一般将FASTQ文件压缩成gzip格式文件。</p>
<p><img src="/wp/f4w/2020/2018-06-20-compression-ratio.png"> </p>
<p>用gzip文件大小除以估计的真实文件大小，得到压缩比，发现压缩比和测序平台有关系，Miseq的压缩比在0.35左右，Hiseq平台中的panel测序在0.24左右，WES在0.2左右。可以发现WES的数据的压缩效率最高，猜测不同平台在压缩过程中设置的压缩比例不同。Hiseq测序平台的gzip文件压缩性能大于Miseq平台。</p>
<p> <img src="/wp/f4w/2020/2018-06-20-gzip-file-size-correlation.png"></p>
<p>以Miseq的gzip文件大小和数据量作图，可以很直观的发现gzip文件大小和数据量之间的线性关系。如图，斜率是1.245，截距可以忽略不计，因为纵坐标很大。也就是说R1文件的数据量约等于gzip文件的1.245倍。</p>
<p><strong>从另外一个角度计算</strong>，R1文件的数据量=gzip文件大小/压缩效率/2.3，则因为Miseq的压缩效率在0.35，Hiseq在0.23左右，所以Miseq R1文件的数据量大概是gzip文件的1.242倍（1/0.35/2.3），和上图拟合的结果很相近，Hiseq约在1.89(1/0.23/2.3)倍。真实的数据量其实是R1文件数据量的2倍（R2文件中的数据量和R1文件相同）。</p>
<p><strong>结论：</strong>不同平台和不同测序方法之间的fastq文件压缩比例并不一致，但同一种方法和平台内的压缩比例是相近的，因而可以根据gzip文件的大小推测出测序的数据量。</p>
<p>我手头的数据中，Miseq的测序量约是R1 gzip文件大小的2.49倍，Hiseq的测序量约是R1 gzip文件大小的3.78倍。不同平台和方法之间，这个值是不固定的，可以根据手头的数据计算一下这个比例，就能迅速的估算出样本的数据量。</p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1008/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1008/" class="post-title-link" itemprop="url">突变注释后的过滤-variant filter</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-06-08 14:49:53" itemprop="dateCreated datePublished" datetime="2018-06-08T14:49:53+08:00">2018-06-08</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>一个全基因组或者全外显子组会产生几万、几十万乃至百万数量的突变。而目标突变往往只有几个或者几十个。我们需要大海捞针似得把我们想要分析的突变找到。突变过滤就是尽可能的将无关突变过滤掉，尽可能的保留目标突变，找出最可能致病的突变。关联性分析则不需要，因为关联性分析根据的是突变在对照和研究人群中的分布。</p>
<p>突变的过滤需要根据分析需要，比如需要寻找一个完全没有研究过的突变，则需要根据dbSNP来过滤等。如果需要寻找新生突变，要同时用到父母双方的突变数据。</p>
<p><img src="/wp/f4w/2020/2018-06-08-filter-var.png"></p>
<p>常用的过滤条件可以分为一下几类：<br>1）根据突变频率进行过滤：每个位点的基因型在人群中频率是不一样的，一般来说，在人群中频率高的基因型往往没有致病性，而是背景突变，只是人群中的多态性。通常用到的频率数据库有gnomAD，ESP，ExAC，1000 Genome等，这些项目检测或搜集了从上千到成万的样本数据，并把数据开放给学术界和产业界。一般过滤的阈值有0.05，0.01，0.001等。<br>2）根据突变位置进行过滤：不同的分析目的，关注的突变并不一样。有人关于调控区域的突变，有人关注编码区的突变，有人关注剪切位点突变等。根据分析目的不同，需要将非关注的区域内的突变过滤掉。如果关注编码区突变，则需要将基因间、内含子、非编码区的突变过滤掉。<br>3）根据突变类型进行过滤：编码区的突变，有同义突变、错义突变、终止密码子突变等。同义突变虽然可能会导致疾病的发生，导致调控紊乱，但一般分析会将同意突变过滤掉（和分析目的有关，如果想要研究同义突变，则不能过滤）。往往移码突变和终止密码子突变是需要关注的。<br>4）根据突变危害预测进行过滤：现在有很多算法在预测位点导致的氨基酸变化的危害性，通常这些算法会分析致病的突变所处的区域、氨基酸变化类型等特征，比如PolyPhen、SIFT等，进而预测出检测出的突变危害性。有些算法会分析序列的保守型，预测突变所处区域是否保守，比如PhastCon，保守区域的突变较非保守区域的突变危害性大。此外，还可以根据氨基酸类型的带电性变化进行过滤，或者根据BLOSUM等打分矩阵分值进行过滤。不同的算法预测的危害性不一定一致。一般统计多个算法预测中，预测危害性较大的比例。<br>5）根据先验知识库进行过滤：注释软件可以注释关联性分析的信息，也可以注释Clinvar相关数据，这些信息在后续的突变位点解读和分析中有很重要的作用。这些信息可以告诉分析人员位点和哪些疾病有关，或者在Clinvar中的致病性评级。可以根据显著性、是否为良性突变、疾病信息等内容对突变 进行过滤。<br>6）根据数据库标识进行过滤：注释软件会注释突变在dbSNP、COSMIC等数据库中的编号。如果研究的疾病非常非常罕见，以往没有研究过，可以尝试寻找非dbSNP数据库中的突变。如果研究和癌症相关的突变，可以寻找COSMIC数据库中出现的突变。<br>7）根据遗传方式进行过滤：如果疾病呈现家族性分布，可以根据系谱图推断出的遗传类型进行过滤。比如如果是隐形突变，则只考虑病人中的纯合突变。如果有线索指示突变为新生突变，则需要根据父母双方的突变数据，将后代中同样位点的突变过滤掉。<br>8）根据其他条件进行过滤：突变过滤还可以有其他的方式，最终还是要根据分析目的进行选择。比如有四个散发样本的突变数据，同样的表型，往往寻找四个样本在相同基因内是否都发生突变，或者是否有相同的高危害突变（比如移码突变）。</p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1001/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1001/" class="post-title-link" itemprop="url">小试shell中的神器zsh</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-05-28 22:20:37" itemprop="dateCreated datePublished" datetime="2018-05-28T22:20:37+08:00">2018-05-28</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>很早就知道神器zsh的存在，一直没有用，今天手痒装了一下，哈哈，用起来还不错。</p>
<p>zsh是bash的增强版，可以兼容bash，可谓是shell中的战斗机，非常方便的命令提示、补全、忽略大小写等功能，一个字爽，可参见</p>
<p><a target="_blank" rel="noopener" href="https://www.zhihu.com/question/21418449">https://www.zhihu.com/question/21418449</a></p>
<p>不过通常zsh配置起来比较复杂麻烦，于是诞生了oh-my-zsh，专门为简化zsh的配置而开发，<a target="_blank" rel="noopener" href="http://ohmyz.sh,极大的简化了zsh的配置,还是一个字,爽./">http://ohmyz.sh，极大的简化了zsh的配置，还是一个字，爽。</a></p>
<h1 id="安装zsh"><a href="#安装zsh" class="headerlink" title="安装zsh"></a>安装zsh</h1><p>首先查看系统是否安装了zsh，发现系统没有安装zsh<br>cat /etc/shells 或者chsh -l</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">/bin/sh</span><br><span class="line">/bin/bash</span><br><span class="line">/sbin/nologin</span><br><span class="line">/bin/tcsh</span><br><span class="line">/bin/csh</span><br><span class="line">/bin/dash</span><br><span class="line">/bin/ksh</span><br></pre></td></tr></table></figure>

<p>如果没有zsh，直接安装oh my zsh会报错Zsh is not installed! Please install zsh first!，所以第一步应该是先安装zsh<br>如果你有sudo权限或者root权限，可以执行</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">apt-get install zsh</span><br></pre></td></tr></table></figure>

<p>如果没有的话（如果有的话，忽略这一步），比如只能安装在自己的HOME目录下，可以通过下载源码自己编译的方式，实现安装zsh</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">wget -c https://jaist.dl.sourceforge.net/project/zsh/zsh/5.5.1/zsh-5.5.1.tar.gz</span><br><span class="line">tar -xvzf zsh-5.5.1.tar.gz</span><br><span class="line">cd zsh-5.5.1</span><br><span class="line">./configure --prefix=/PATH</span><br><span class="line">make</span><br><span class="line">make install</span><br><span class="line"></span><br><span class="line"># 在.bashrc文件中设置启动</span><br><span class="line">exec /share/apps/zsh/bin/zsh -l</span><br><span class="line"># 此时还要确保zsh在环境变量中，如果没有，请设置。</span><br></pre></td></tr></table></figure>

<h1 id="安装oh-my-zsh"><a href="#安装oh-my-zsh" class="headerlink" title="安装oh my zsh"></a>安装oh my zsh</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sh -c &quot;$(curl -fsSL https://raw.githubusercontent.com/robbyrussell/oh-my-zsh/master/tools/install.sh)&quot;</span><br></pre></td></tr></table></figure>



<p><img src="/wp/f4w/2020/2018-05-28-install-oh-my-zsh.png"> </p>
<p>既然用了zsh，我也尝试了一下新的虚拟终端MobaXterm。<br><a target="_blank" rel="noopener" href="https://mobaxterm.mobatek.net/">https://mobaxterm.mobatek.net</a></p>
<p>安利的文章可以看这个，<a target="_blank" rel="noopener" href="https://www.isharebest.com/mobaxterm.htm">https://www.isharebest.com/mobaxterm.htm</a></p>
<p>这个虚拟终端的功能非常丰富，但用起来有时候莫名其妙的卡。或许zsh和Xshell更搭。</p>
<p>尝试了下命令提示的功能，如图，还是很方便的，后续以后会经常用zsh了。</p>
<p><img src="/wp/f4w/2020/2018-05-28-autopromote.png"> </p>
<p>oh my zsh的主题列表<a target="_blank" rel="noopener" href="https://github.com/robbyrussell/oh-my-zsh/wiki/themes">https://github.com/robbyrussell/oh-my-zsh/wiki/themes</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/996/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/996/" class="post-title-link" itemprop="url">BCFTOOLS支持的表达式</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-05-24 15:20:52" itemprop="dateCreated datePublished" datetime="2018-05-24T15:20:52+08:00">2018-05-24</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>翻译<a target="_blank" rel="noopener" href="https://samtools.github.io/bcftools/bcftools-man.html#expressions">https://samtools.github.io/bcftools/bcftools-man.html#expressions</a></p>
<h1 id="有效的表达式可能包含"><a href="#有效的表达式可能包含" class="headerlink" title="有效的表达式可能包含"></a>有效的表达式可能包含</h1><ul>
<li><p>数字常量，字符常量，文件名</p>
<p>  1, 1.0, 1e-4<br>  “String”<br>  @file_name</p>
</li>
<li><p>算数运算符：</p>
<p>  +,*,-,/</p>
</li>
<li><p>比较运算符：</p>
<p>  == (同=), &gt;, &gt;=, &lt;=, &lt;, !=</p>
</li>
<li><p>正则运算符”<del>“及其非值”!</del>“。大小写敏感，加”/i”这大小写不敏感：</p>
<p>  INFO/HAYSTACK ~ “needle”<br>  INFO/HAYSTACK ~ “NEEDless/i”</p>
</li>
<li><p>括号：</p>
<p>  (, )</p>
</li>
<li><p>逻辑运算符：</p>
<p>  &amp;&amp; (同&amp;), ||,  |</p>
</li>
<li><p>VCF/BCF中的INFO tags, FORMAT tags, column names：</p>
<p>  INFO tags: INFO/DP or DP<br>  FORMAT tags: FORMAT/DV, FMT/DV, or DV<br>  column names:  FILTER, QUAL, ID, POS, REF, ALT[0]</p>
</li>
<li><p>1 (或0) 检测flag的存在 (或不存在) ：</p>
<p>  FlagA=1 &amp;&amp; FlagB=0</p>
</li>
<li><p>“.” 检测缺失值：</p>
<p>  DP=”.”, DP!=”.”, ALT=”.”</p>
</li>
<li><p>不论分型phase或者倍体ploid(“.|.”, “./.”, “.”) ，缺失的基因型genotype可以通过以下表达式匹配：</p>
<p>  GT<del>“\.”, GT!</del>“\.”</p>
</li>
<li><p>包括分型phase或者倍体ploid(“.|.”, “./.”, “.”) ，可以通过以下表达式匹配：</p>
<p>  GT=”.|.”, GT=”./.”, GT=”.”</p>
</li>
<li><p>样本基因型genotype：参考</p>
<p>  参考基因型（单倍体或二倍体）:  GT=”ref”<br>  突变基因型 （纯合或杂合，单倍体或二倍体）： GT=”alt”<br>  缺失基因型：GT=”mis”<br>  纯合：GT=”hom”<br>  杂合：GT=”het”<br>  单倍体： GT=”hap”<br>  纯合参考：GT=”RR”<br>  纯合突变： GT=”AA”<br>  ref-alt杂合： GT=”RA” or GT=”AR”<br>  alt-alt杂合：GT=”Aa” or GT=”aA”<br>  haploid ref： GT=”R”<br>  haploid alt：GT=”A”<br>  大小写不敏感</p>
</li>
<li><p>在REF和ALT列中的突变类型：（indel，snp，mnp，ref，bnd（断点breakend？），other），”\~” 表示至少一个等位基因是这种类型，”=” 要求所有等位基因是这个类型：</p>
<p>  TYPE=”snp”<br>  TYPE<del>“snp”<br>  TYPE!=”snp”<br>  TYPE!</del>“snp”</p>
</li>
<li><p>数组下标（0开始）， “*” 任意元素， “-“ 表示范围。在检索FORMAT向量时，冒号”:” 可以用来选择样本和元素，如下：</p>
<p>  INFO/AF[0] &gt; 0.3             ..   第一个AF值大于0.3<br>  FORMAT/AD[0:0] &gt; 30          ..   第一个样本的第一个AD值大于30<br>  FORMAT/AD[0:1]               ..   第一个样本的第二个AD值<br>  FORMAT/AD[1:0]               ..   第二个样本的第一个AD值<br>  DP4[*] == 0                  ..   任意DP4的值<br>  FORMAT/DP[0]   &gt; 30         ..   第一个样本的DP值大于30<br>  FORMAT/DP[1-3] &gt; 10         ..  第二到四的样本<br>  FORMAT/DP[1-]  &lt; 7          ..    除了第一个样本的其他所有样本<br>  FORMAT/DP[0,2-4] &gt; 20       ..   第一，三到五个样本<br>  FORMAT/AD[0:1]              ..   第一个样本的第二个AD field<br>  FORMAT/AD[0:<em>], AD[0:] or AD[0] ..  第一个样本得到所有AD field<br>  FORMAT/AD[</em>:1] or AD[:1]        ..   所有样本的AD field<br>  (DP4[0]+DP4[1])/(DP4[2]+DP4[3]) &gt; 0.3<br>  CSQ[*] ~ “missense_variant.*deleterious”</p>
</li>
<li><p>基于FORMAT tags和INFO tags的函数：</p>
<p>  MAX, MIN, AVG, SUM, STRLEN, ABS, COUNT</p>
</li>
<li><p>变量可以快速计算得到：</p>
<p>  alternate allele的数目：N_ALT<br>  样本数目：N_SAMPLES<br>  alternate allele的的计数：AC<br>  次等位基因计数，与AC相似，但统计的是小于0.5的等位基因）：MAC,<br>  alternate等位基因的频率（AF=AC/AN）：AF<br>  次等位基因频率（MAF=MAC/AN）： MAF<br>  在called genotypes中等位基因的数目： AN<br>  有missing genotype的样本数目：N_MISSING<br>  有missing genotype的样本比例：F_MISSING</p>
</li>
<li><p>配置perl过滤。注意这个命令默认是没有编译的，在安装的时候可以查看INSTALL文件的Optional Compilation with Perl部分，misc/demo-flt.pl 是一个示例，这个示例中定义了一个名为severity的perl子程序，并被以下命令行调用：</p>
<p>  perl:path/to/script.pl; perl.severity(INFO/CSQ) &gt; 3</p>
</li>
</ul>
<h1 id="注意"><a href="#注意" class="headerlink" title="注意"></a>注意</h1><p>字符和正则表达式都是不区分大小写的，除了tag名外，变量和函数名也是不区分大小写的。例如”qual” 等同 “QUAL”，”strlen()”等同”STRLEN()” ，但是”dp” 不等同 “DP”。 当检索多个值时，所有的元素都会被测试，通过或逻辑（OR）得到结果。比如检索”TAG=1,2,3,4”时：</p>
<pre><code>-i &#39;TAG\[*\]=1&#39;   .. true, 真，该条记录会被打印
-i &#39;TAG\[*\]!=1&#39;  .. true
-e &#39;TAG\[*\]=1&#39;   .. false,否，该条打印会被丢弃
-e &#39;TAG\[*\]!=1&#39;  .. false
-i &#39;TAG\[0\]=1&#39;   .. true
-i &#39;TAG\[0\]!=1&#39;  .. false
-e &#39;TAG\[0\]=1&#39;   .. false
-e &#39;TAG\[0\]!=1&#39;  .. true
</code></pre>
<h1 id="示例"><a href="#示例" class="headerlink" title="示例"></a>示例</h1><pre><code>MIN(DV)&gt;5
MIN(DV/DP)&gt;0.3
MIN(DP)&gt;10 &amp; MIN(DV)&gt;3
FMT/DP&gt;10  &amp; FMT/GQ&gt;10 ..   同一个样本需满足两个条件 
FMT/DP&gt;10 &amp;&amp; FMT/GQ&gt;10 ..  允许不同样本间可以满足这两个条件
QUAL&gt;10 |  FMT/GQ&gt;10   ..   对QUAL&gt;10 的位点或者 FMT/GQ&gt;10的样本为真，但只选择FMT/GQ&gt;10的样本
QUAL&gt;10 || FMT/GQ&gt;10   .. 对QUAL&gt;10 的位点或者 FMT/GQ&gt;10的样本为真，选择该位点的所有样本
TYPE=&quot;snp&quot; &amp;&amp; QUAL&gt;=10 &amp;&amp; (DP4\[2\]+DP4\[3\] &gt; 2)
COUNT(GT=&quot;hom&quot;)=0
MIN(DP)&gt;35 &amp;&amp; AVG(GQ)&gt;50
ID=@file       ..   选择出现在文件file中的ID
ID!=@~/file    ..   忽略出现在文件file中的ID
MAF\[0\]&lt;0.05    ..   以5%为阈值，选择rare variant
POS&gt;=100   ..   限制检索范围，比如20:100-200内的突变
</code></pre>
<h1 id="Shell-扩展"><a href="#Shell-扩展" class="headerlink" title="Shell 扩展"></a>Shell 扩展</h1><p>表达式需要用单引号引起来，因为一些字符可能在shell中有特殊含义。比如</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bcftools view -i <span class="string">&#x27;%ID!=&quot;.&quot; &amp; MAF\[0\]&lt;0.01&#x27;</span> </span><br></pre></td></tr></table></figure>

<h1 id="用bcftools进行突变筛选的示例"><a href="#用bcftools进行突变筛选的示例" class="headerlink" title="用bcftools进行突变筛选的示例"></a>用bcftools进行突变筛选的示例</h1><p><strong>摘自公众号生信开发者</strong> </p>
<p>#选出位于bed文件中的所有区域的突变位点</p>
<p><code>bcftools view -R target.bed sample.filter.vcf.bgz </code></p>
<p>#选出INFO中SAO=1的所有位点</p>
<p><code>bcftools view -i &#39;SAO=1&#39; sample.filter..b37.vcf </code></p>
<p>#选出INFO中AC&gt;2的所有位点</p>
<p><code>bcftools view -i &quot;AC&gt;=2&quot; ExAC.r1.sites.vep.vcf.gz </code></p>
<p>#选出遗传方式是AR或XR的位点（需INFO字段中已有INHERITANCE注释）</p>
<p><code>bcftools view -i &#39;INHERITANCE\[*\] = &quot;AR&quot; || INHERITANCE\[*\] = &quot;XR&quot;&#39; sample.filter.annovar.hg19_multianno.Fun.ar.vcf </code></p>
<p>#选出位于区域X:31136335-33358725的所有位点</p>
<p><code>bcftools view sample.filter.vcf.bgz X:31136335-33358725</code></p>
<p>#选出除了INFO字段CLINSIG匹配”Benign”以外的所有</p>
<p><code>bcftools view -e &#39;CLINSIG~&quot;Benign&quot;&#39; sample.filter.hg19_multianno.Fun.vcf </code></p>
<p>#选出最小的depth&gt;500而且，肿瘤样品的VAF&gt;0.05的所有位点</p>
<p><code>bcftools view -i &quot; MIN(FMT/DP)&gt;500 &amp;&amp; FORMAT/AD\[1:1\]/FORMAT/DP\[1\]&gt;0.05 &quot; annovar.exdbsnp.exCOSMIC.exTSG.cds.vcf | sed &#39;/^#/d&#39; </code></p>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p>表达式手册翻译自<a target="_blank" rel="noopener" href="https://samtools.github.io/bcftools/bcftools-man.html#expressions">https://samtools.github.io/bcftools/bcftools-man.html#expressions</a> </p>
<p>bcftools突变筛选摘自<strong>公众号生信开发者</strong></p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  


  
  <nav class="pagination">
    <a class="extend prev" rel="prev" href="/page/5/"><i class="fa fa-angle-left" aria-label="Previous page"></i></a><a class="page-number" href="/">1</a><span class="space">&hellip;</span><a class="page-number" href="/page/5/">5</a><span class="page-number current">6</span><a class="page-number" href="/page/7/">7</a><span class="space">&hellip;</span><a class="page-number" href="/page/17/">17</a><a class="extend next" rel="next" href="/page/7/"><i class="fa fa-angle-right" aria-label="Next page"></i></a>
  </nav>



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Jason</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">163</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">167</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>


<!--google_adsense-->

  <div class="links-of-blogroll-title">
    <i class="fa fa-google"></i> AdSense
    <ins class="adsbygoogle"
        style="display:block"
        data-ad-client="ca-pub-9370593480269435"
        data-ad-slot="8824957305"
        data-ad-format="auto"
        data-full-width-responsive="true"></ins>
    <script>
      (adsbygoogle = window.adsbygoogle || []).push({});
    </script>
  </div>

<!--/google_adsense-->



    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Jason</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

<script data-ad-client="ca-pub-9370593480269435" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<!-- PGAD -->
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-9370593480269435"
     data-ad-slot="8824957305"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
