<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"md.zxzyl.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>


  <meta property="og:type" content="website">
<meta property="og:title" content="Zhongxu&#39;s blog">
<meta property="og:url" content="https://md.zxzyl.com/page/8/index.html">
<meta property="og:site_name" content="Zhongxu&#39;s blog">
<meta property="og:locale" content="en_US">
<meta property="article:author" content="Jason">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://md.zxzyl.com/page/8/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : true,
    isPost : false,
    lang   : 'en'
  };
</script>

  <title>Zhongxu's blog</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a3b5d3f466edb3e3f15e8238d8f04be8";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Zhongxu's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content index posts-expand">
            
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/848/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/848/" class="post-title-link" itemprop="url">思考：是否升级参考基因组版本</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-02-27 14:27:45" itemprop="dateCreated datePublished" datetime="2018-02-27T14:27:45+08:00">2018-02-27</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>Should I switch to a newer reference？</p>
<p>GRCh38 consists of several components: chromosomal assembly, unlocalized contigs (chromosome known but location unknown), unplaced contigs (chromosome unknown) and ALT contigs (long clustered variations). The combination of the first three components is called the primary assembly. It is recommended to use the complete primary assembly for all analyses. </p>
<p><strong>参考基因组包括chromosomal assembly, unlocalized contigs (chromosome known but location unknown), unplaced contigs (chromosome unknown)和ALT contigs (long clustered variations)，前三个是primary assembly,alt contigs代表的是部分区域单体型的多样性，这些区域过于复杂不能用一条序列表示。</strong></p>
<p>In addition to adding many alternate contigs, GRCh38 corrects thousands of SNPs and indels in the GRCh37 assembly that are absent in the population and are likely sequencing artifacts. It also includes synthetic centromeric sequence and updates non-nuclear genomic sequence.</p>
<p><strong>除了添加了很多alternative contigs，GRCh38更正了数以千计的GRCh37版本中的SNPs和indels，这些SNPs和indels在人群中没有出现过、可能因为测序错误导致。GRCh38版本也包含了人工的着丝粒序列和更新了非核基因组序列。</strong></p>
<p>The ability to recognize alternate haplotypes for loci is a drastic improvement that GRCh38 makes possible. Going forward, expanding genomics data will help identify variants for alternate haplotypes, improve existing and add additional alternate haplotypes and give us a better accounting of alternate haplotypes within populations. We are already seeing improvements and additions in the patch releases to reference genomes, e.g. the seven minor releases of GRCh38 available at the time of this writing.<br><strong>GRCh38大幅提高了识别alternate haplotype的能力，进一步提高了识别alternate haplotype的突变的能力。</strong></p>
<p>BWA的作者Li Heng推荐GRCh37 primary assembly+ALT+decoy组成的参考基因组hs38DH，可以通过bwa下载，见<a target="_blank" rel="noopener" href="https://github.com/lh3/bwa/blob/master/README-alt.md">https://github.com/lh3/bwa/blob/master/README-alt.md</a> ：</p>
<pre>bwa.kit/run-gen-ref hs38DH</pre>

<p>作者的用NA12878做测试的比对结果如下，</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Assembly	hs37	hs38	hs38DH</span><br><span class="line">FP	255706	168068	142516</span><br><span class="line">TP	2142260	2163113	2150844</span><br></pre></td></tr></table></figure>




<p><strong>PS：</strong></p>
<p>从改进来看，这些改进都将提高比对和突变检测的质量，<br>从结果看，38版本提高了TP的灵敏度，并降低了FP，<br>从趋势来看，37升级成38是大趋势，so请及时升级。</p>
<p><strong>参考</strong></p>
<p><a target="_blank" rel="noopener" href="https://software.broadinstitute.org/gatk/documentation/article.php?id=7857">https://software.broadinstitute.org/gatk/documentation/article.php?id=7857</a></p>
<p><a target="_blank" rel="noopener" href="https://gatkforums.broadinstitute.org/gatk/discussion/8180/9-takeaways-to-help-you-get-started-with-grch38">https://gatkforums.broadinstitute.org/gatk/discussion/8180/9-takeaways-to-help-you-get-started-with-grch38</a><a target="_blank" rel="noopener" href="https://software.broadinstitute.org/gatk/documentation/article.php?id=8017">https://software.broadinstitute.org/gatk/documentation/article.php?id=8017</a></p>
<p><a target="_blank" rel="noopener" href="https://github.com/lh3/bwa/blob/master/README-alt.md">https://github.com/lh3/bwa/blob/master/README-alt.md</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/838/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/838/" class="post-title-link" itemprop="url">liftover,crossmap进行坐标转换时用到的chain文件介绍</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-02-12 21:46:41" itemprop="dateCreated datePublished" datetime="2018-02-12T21:46:41+08:00">2018-02-12</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p><img src="/wp/f4w/2020/2018-02-11-liftover-chain-description.png" alt="chain description"></p>
<p>在做基因组坐标转换的时候，用crossmap和liftover的时候，会用到chain file。大家都在讲坐标转换会用到这个文件，却没有讲过chain文件的具体内容（百度和谷歌搜索结果都没有中文介绍，本文应该是第一个）。本文的内容都翻译自UCSC网站，原文<a target="_blank" rel="noopener" href="https://genome.ucsc.edu/goldenpath/help/chain.html%EF%BC%8C%E5%B8%8C%E6%9C%9B%E8%83%BD%E5%B8%AE%E5%88%B0%E5%A4%A7%E5%AE%B6%E4%BA%86%E8%A7%A3%E8%BF%99%E4%B8%AA%E6%96%87%E4%BB%B6%E3%80%82">https://genome.ucsc.edu/goldenpath/help/chain.html，希望能帮到大家了解这个文件。</a></p>
<p>UCSC chain文件 <a target="_blank" rel="noopener" href="http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/">http://hgdownload.soe.ucsc.edu/goldenPath/hg38/liftOver/</a><br>Ensembl chain文件 <a target="_blank" rel="noopener" href="https://sourceforge.net/projects/crossmap/files/Ensembl_chain_files/">https://sourceforge.net/projects/crossmap/files/Ensembl_chain_files/</a></p>
<p>chain file里面包含许多块alignment的信息（个人觉得可以理解为同源的地方，chain），其中每一块有一个header，记录alignment在两个版本中坐标，以及许多行alignment data line记录具体比对情况。</p>
<p>即每一块有 Header Line 和 Alignment Data Lines组成。形式如下，有两个chain</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">chain 4900 chrY 58368225 + 25985403 25985638 chr5 151006098 - 43257292 43257528 1</span><br><span class="line"> 9       1       0</span><br><span class="line"> 10      0       5</span><br><span class="line"> 61      4       0</span><br><span class="line"> 16      0       4</span><br><span class="line"> 42      3       0</span><br><span class="line"> 16      0       8</span><br><span class="line"> 14      1       0</span><br><span class="line"> 3       7       0</span><br><span class="line"> 48</span><br><span class="line"></span><br><span class="line"> chain 4900 chrY 58368225 + 25985406 25985566 chr5 151006098 - 43549808 43549970 2</span><br><span class="line"> 16      0       2</span><br><span class="line"> 60      4       0</span><br><span class="line"> 10      0       4</span><br></pre></td></tr></table></figure>



<h1 id="Header-Lines"><a href="#Header-Lines" class="headerlink" title="Header Lines"></a>Header Lines</h1><p>记录两个版本的序列对应区域</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">chain score tName tSize tStrand tStart tEnd qName qSize qStrand qStart qEnd id</span><br></pre></td></tr></table></figure>

<p>header line以chain开头<br>score – chain 值<br>tName – 染色体 (reference sequence)<br>tSize – 染色体大小 (reference sequence)<br>tStrand – 链strand (reference sequence)<br>tStart – 比对开始 (reference sequence)<br>tEnd – 比对结束 (reference sequence)<br>qName – 检索染色体 (query sequence，对应的另外一个基因组版本 )<br>qSize – 检索染色体大小 (query sequence)<br>qStrand – 链strand (query sequence)<br>qStart – 比对开始 (query sequence)<br>qEnd – 比对结束 (query sequence)<br>id – chain ID</p>
<p>位置从0开始，前100bp的表示方法为0到100，接下的100bp为100到200，strand的值为-号时，对应的是反向互补序列。</p>
<h1 id="Alignment-Data-Lines"><a href="#Alignment-Data-Lines" class="headerlink" title="Alignment Data Lines"></a>Alignment Data Lines</h1><p>Alignment Data Line有三列，记录区域内的具体对应信息，最后一行只有ungapped的序列长度。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">size dt dq</span><br></pre></td></tr></table></figure>

<p>size – ungapped（一致的区域）的长度<br>dt – 该block块（一致的区域）的结束位置，到下一个block快（下一个一致的区域）起始位置的长度，其实就是gapped block不一致的区域的长度 (reference sequence)<br>dq – 该block块（一致的区域）的结束位置，到下一个block快（下一个一致的区域）起始位置的长度，其实就是gapped block不一致的区域的长度(query sequence)</p>
<p>以chain id为2的chain为例</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">chain 4900 chrY 58368225 + 25985406 25985566 chr5 151006098 - 43549808 43549970 2</span><br><span class="line">16      0       2</span><br><span class="line">60      4       0</span><br><span class="line">10      0       4</span><br><span class="line">70</span><br></pre></td></tr></table></figure>




<p>我个人理解的，不一定正确，欢迎指正</p>
<p><img src="/wp/f4w/2020/2018-02-11-liftover-chain-description.png" alt="chain description"></p>
<p>坐标转换在reference和query的位置为：<br>reference: start 25985406 end 25985566<br>query: start 43549808 end 43549970</p>
<p>区域长度分别为：<br>reference region length = 25985566 - 25985406 = 160 bp<br>query region length = 43549970 - 43549808 = 162 bp</p>
<p>一致的区域（ungapped）长度为<br>reference = query = 16 + 60 +10 + 70 = 156 bp</p>
<p>不一致的区域（gapped）长度为<br>reference = 0 + 4 +0 =4<br>query = 2 + 0 + 4 =6</p>
<p>区域长度也可以通过ungapped和gapped区域计算，分别为：<br>reference = ungapped + gapped = 156 + 4 =160 bp<br>query = ungapped + gapped = 156 + 6 =162 bp</p>
<p>更多信息详见：</p>
<p><a target="_blank" rel="noopener" href="http://genomewiki.ucsc.edu/index.php/LiftOver_Howto">http://genomewiki.ucsc.edu/index.php/LiftOver_Howto</a></p>
<p><a target="_blank" rel="noopener" href="https://genome.ucsc.edu/goldenpath/help/chain.html">https://genome.ucsc.edu/goldenpath/help/chain.html</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/834/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/834/" class="post-title-link" itemprop="url">基因组坐标转换工具-以BED文件为例，从hg19转换到hg38坐标</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-02-11 11:39:11" itemprop="dateCreated datePublished" datetime="2018-02-11T11:39:11+08:00">2018-02-11</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>分析时使用的基因组版本，可能会与其他来源数据所使用的基因组版本不一致，需要统一成同一个版本的坐标，才能方便下一步的分析。</p>
<p>常用的有NCBI的Remap在线服务和UCSC的liftover，其实还有很多，本文暂时总结部分工具的用法。以将APOA1的编码区坐标（利用UCSC的genome browser下载，或者下载该文件<a href="/wp/f4w/2020/FileAttach/2018-02-11-APOA1.bed">APOA1.bed</a>）转换为例，从hg19转到hg38版本坐标上。需要注意的是，在使用的时候，需要注意是否支持对应的格式。</p>
<table>
<thead>
<tr>
<th></th>
<th>类型</th>
<th>支持格式</th>
<th>地址</th>
<th>推荐指数</th>
</tr>
</thead>
<tbody><tr>
<td>Liftover</td>
<td>在线</td>
<td>bed</td>
<td><a target="_blank" rel="noopener" href="http://genome.ucsc.edu/cgi-bin/hgLiftOver">http://genome.ucsc.edu/cgi-bin/hgLiftOver</a></td>
<td>一般</td>
</tr>
<tr>
<td>Liftover</td>
<td>本地</td>
<td>bed和gff</td>
<td><a target="_blank" rel="noopener" href="http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver">http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver</a></td>
<td>推荐</td>
</tr>
<tr>
<td>Remap</td>
<td>在线</td>
<td>hgvs，bed，gvf，gff，gtf，Text ASN.1，Binary ASN.1，UCSC Region和VCF</td>
<td><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genome/tools/remap">https://www.ncbi.nlm.nih.gov/genome/tools/remap</a></td>
<td>推荐</td>
</tr>
<tr>
<td>CrossMap</td>
<td>本地</td>
<td>SAM/BAM,，Wiggle/BigWig， bed， gff/gtf，VCF</td>
<td><a target="_blank" rel="noopener" href="http://crossmap.sourceforge.net/">http://crossmap.sourceforge.net/</a></td>
<td>推荐</td>
</tr>
<tr>
<td>picard</td>
<td>本地</td>
<td>interval和VCF</td>
<td><a target="_blank" rel="noopener" href="http://broadinstitute.github.io/picard/">http://broadinstitute.github.io/picard/</a></td>
<td></td>
</tr>
</tbody></table>
<h1 id="在线版"><a href="#在线版" class="headerlink" title="在线版"></a>在线版</h1><h2 id="Liftover"><a href="#Liftover" class="headerlink" title="Liftover"></a>Liftover</h2><p><img src="/wp/f4w/2020/2018-02-11-screencapture-UCSC-hgLiftOver.png" alt="hg-liftover-ucsc"> </p>
<p>服务地址： <a target="_blank" rel="noopener" href="http://genome.ucsc.edu/cgi-bin/hgLiftOver">http://genome.ucsc.edu/cgi-bin/hgLiftOver</a></p>
<ul>
<li><ul>
<li>  1，首先选择检索和目标基因组</li>
</ul>
</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">Original Genome:	Human</span><br><span class="line">Original Assembly:	Feb. 2009 (GRCh37/hg19)</span><br><span class="line">New Genome:  	Human</span><br><span class="line">New Assembly:	Dec. 2013 (GRCh38/hg38)  </span><br></pre></td></tr></table></figure>


<ul>
<li>  2，将bed格式内容复制或者上传，如果是复制内容的话，点击submit；如果是上传文件的 话，点击submit</li>
<li>  3，在Result部分下载结果，Result部分会显示结果说明，点击View Conversions，即能得到转换后的bed文件。</li>
</ul>
<h2 id="Remap"><a href="#Remap" class="headerlink" title="Remap"></a>Remap</h2><p><img src="/wp/f4w/2020/2018-02-11-screencapture-NCBI-remap.png" alt="NCBI-remap"></p>
<p>支持hgvs，bed，gvf，gff，gtf，Text ASN.1，Binary ASN.1，UCSC Region和VCF，如果数据量较少，可以考虑该方法。</p>
<p>服务地址：<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/genome/tools/remap">https://www.ncbi.nlm.nih.gov/genome/tools/remap</a></p>
<ul>
<li>  1，在Assembly-Assembly中选择</li>
</ul>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">Source Organism *：Homo sapiens</span><br><span class="line">Source Assembly *:   GRCh37(hg19)</span><br><span class="line">Target Assembly *:    GRCh38(hg38)</span><br></pre></td></tr></table></figure>




<ul>
<li>  2，在Data处，可以选择复制文件内容还是上传文件，还可以指定文件格式。</li>
<li>  3，点击submit，页面会跳转到结果页面，Summary Data告诉你有多少条匹配，Mapping Report告诉你对应关系。这些都可以下载，NCBI remap的结果文件前几列是原来的内容，后几列是转换后的坐标。</li>
</ul>
<h2 id="Ensembl-Assembly-Converter"><a href="#Ensembl-Assembly-Converter" class="headerlink" title="Ensembl Assembly Converter"></a>Ensembl Assembly Converter</h2><p><a target="_blank" rel="noopener" href="http://asia.ensembl.org/Homo_sapiens/Tools/AssemblyConverter?db=core">http://asia.ensembl.org/Homo_sapiens/Tools/AssemblyConverter?db=core</a></p>
<h1 id="本地版"><a href="#本地版" class="headerlink" title="本地版"></a>本地版</h1><p>本地工具需要利用chain file，才能知道两个版本的坐标对应关系。chain文件可以从UCSC下载。我们需要的是hg19Thg38.over.chain.gz<br><a target="_blank" rel="noopener" href="http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz">http://hgdownload.soe.ucsc.edu/goldenPath/hg19/liftOver/hg19ToHg38.over.chain.gz</a><br>chain文件后续会有介绍。</p>
<h2 id="CrossMap"><a href="#CrossMap" class="headerlink" title="CrossMap"></a>CrossMap</h2><p>CrossMap支持SAM/BAM, Wiggle/BigWig, BED, GFF/GTF, VCF格式。支持的格式较多，用起来方便。</p>
<p>软件网址：<a target="_blank" rel="noopener" href="http://crossmap.sourceforge.net/">http://crossmap.sourceforge.net/</a></p>
<p>安装：pip install CrossMap</p>
<p>当然也可以从源码编译安装，官网有说明（<a target="_blank" rel="noopener" href="http://crossmap.sourceforge.net/#install-crossmap-from-source-code%EF%BC%89%E3%80%82">http://crossmap.sourceforge.net/#install-crossmap-from-source-code）。</a></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python CrossMap.py bed    hg19ToHg38.over.chain      APOA1.bed      APOA1.hg38.bed</span><br></pre></td></tr></table></figure>



<h2 id="Liftover-1"><a href="#Liftover-1" class="headerlink" title="Liftover"></a>Liftover</h2><p>支持bed和gff格式</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">wget -c http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/liftOver</span><br><span class="line">chmod 755 liftOver</span><br><span class="line"># liftOver oldFile map.chain newFile unMapped</span><br><span class="line">liftOver   APOA1.bed    hg19ToHg38.over.chain    APOA1.hg38.bed   unMapped.txt</span><br></pre></td></tr></table></figure>




<h2 id="Picard"><a href="#Picard" class="headerlink" title="Picard"></a>Picard</h2><p>地址： <a target="_blank" rel="noopener" href="http://broadinstitute.github.io/picard/">http://broadinstitute.github.io/picard/</a></p>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/834/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/829/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/829/" class="post-title-link" itemprop="url">hg19、GRCH37、b37、hs37d5介绍和区别</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-02-09 15:37:02" itemprop="dateCreated datePublished" datetime="2018-02-09T15:37:02+08:00">2018-02-09</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>大家经常用UCSC的hg19和NCBI的GRCh37版本的，但还有其他的版本，比如b37，hg37d5，比如在分析NIST的genome in a bottle（GIAB）提供的bam数据时，就遇到了hg37d5的版本，在用GATK的时候会遇到b37版本。</p>
<h1 id="GRCh37"><a href="#GRCh37" class="headerlink" title="GRCh37"></a>GRCh37</h1><p>Genome Reference Consortium(基因组参照序列联盟)，由英国Wellcome Trust Sanger研究中心（the Wellcome Trust Sanger Center）、华盛顿大学基因组中心（The Washington University Genome Center）、欧洲生物信息研究所（the European Bioinformatics Institute）和美国国家生物技术信息中心（NCBI）联合组成。</p>
<p>GRCH37版本发布之后，也会有小的更新，比如GRCh37.p2，大的更新比如由GRCh37升级到GRCh38，填补gap，修改部分序列，其目的是提供一个完整的基因组序列assemble。GRCh38已经在2013年发布，多数基因组数据库正在兼容或者更新到该版本。</p>
<p>该版本包含人类chr1到chr22，chrX，chrY，MT染色体以及</p>
<ul>
<li>  “unlocalized sequences”：知道来自哪条染色体但不知道具体位置的序列</li>
<li>  “unplaced sequences”：知道来自人类基因组序列，但不知道与染色体的关系</li>
<li>  “alternate loci”：来自基因组特定区域，代表该区域序列的多样性<br>“1” to “22”, “X”, “Y” and “MT”命名比较规范，ENSEMBL， genome browser， the NCBI dbSNP (in VCF files), the Sanger COSMIC (in VCF files),都依照该规范。</li>
</ul>
<p>下载地址：<a href="ftp://ftp.ncbi.nih.gov/genomes/Homo_sapiens">ftp://ftp.ncbi.nih.gov/genomes/Homo_sapiens</a></p>
<h1 id="GRCh37lite"><a href="#GRCh37lite" class="headerlink" title="GRCh37lite"></a>GRCh37lite</h1><p>只包含GRCh37版本中的chr1到chr22，chrX，chrY，MT染色体</p>
<p>下载地址：<a href="ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh37/special_requests/">ftp://ftp.ncbi.nih.gov/genbank/genomes/Eukaryotes/vertebrates_mammals/Homo_sapiens/GRCh37/special_requests/</a></p>
<h1 id="hg19"><a href="#hg19" class="headerlink" title="hg19"></a>hg19</h1><p>UCSC提供，容易下载，因为UCSC方便下载各种坐标文件（bed，gtf等），该版本可以与这些坐标对应。与GRCh38对应的是hg38版本。</p>
<p>该版本序列包括chr1到chr22，chrX，chrY序列与GRCh37完全一致（完全一致，完全一致），线粒体序列稍微不一样，以及</p>
<ul>
<li>“chr*_random sequences” 知道来自哪条染色体但不知道具体位置的序列</li>
<li>“chrUn_* sequences” 知道来自人类基因组序列，但不知道与染色体的关系</li>
</ul>
<p>UCSC与GRCh不同的地方有：</p>
<ul>
<li>在重复区域repeat region有小写来表示，这点和GRCh不同</li>
<li>此外染色体有chr前缀，而GRCh没有chr前缀。</li>
<li>线粒体序列版本不一样</li>
</ul>
<p>下载地址：<a href="ftp://hgdownload.cse.ucsc.edu/goldenPath/hg19/chromosomes">ftp://hgdownload.cse.ucsc.edu/goldenPath/hg19/chromosomes</a></p>
<h1 id="b37"><a href="#b37" class="headerlink" title="b37"></a>b37</h1><p>来自1000 Genome第一阶段，被称为b37（GATK和IGV社区小组中经常使用），包含了GRCh37, the rCRS mitochondrial sequence（MT序列）<br>unlocalized sequences和unplaced sequences以他们的检索号命名，比如”GL000191.1”, “GL000194.1”, etc.<br>但是不（不不不）包含 alternate loci</p>
<p>下载地址：<a href="ftp://ftp.broadinstitute.org/pub/seq/references">ftp://ftp.broadinstitute.org/pub/seq/references</a></p>
<h1 id="hs37d5"><a href="#hs37d5" class="headerlink" title="hs37d5"></a>hs37d5</h1><p>可以理解成b37的升级版，在1000 Genome第二阶段使用。该版本包含了b37数据，以及</p>
<ul>
<li>human herpesvirus 4 type 1 sequence 人类疱疹病毒序列(“NC_007605”) </li>
<li>“decoy” sequence诱饵序列（名为hs37d5） 来自HuRef、BAC或者质粒克隆和NA12878，可以提高序列比对的正确率 </li>
<li>此外在Y染色体上的性染色体同源区域PAR标为N碱基，这样对应的X染色体的区域当作二倍体 </li>
</ul>
<p>这些更改有利于序列比对和突变检测，降低假阳性与b37兼容。</p>
<p>PS：正如以前讲到的（文章《思考在比对时，关于是否将chr*<em>random和chrUn</em>*序列放在参考基因组中的思考》），序列比对不要只放chr1到chr22，chrX，chrY和MT（我见过国内知名测序公司有这么干的），其他序列和诱饵序列非常重要，可以提高比对的准确率，降低假阳性。</p>
<p>详情见：<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.slides.pdf">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.slides.pdf</a><br>下载地址：<a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence</a></p>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p><a target="_blank" rel="noopener" href="http://googlegenomics.readthedocs.io/en/latest/use_cases/discover_public_data/reference_genomes.html">http://googlegenomics.readthedocs.io/en/latest/use_cases/discover_public_data/reference_genomes.html</a><br><a target="_blank" rel="noopener" href="https://wiki.dnanexus.com/Scientific-Notes/human-genome">https://wiki.dnanexus.com/Scientific-Notes/human-genome</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/815/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/815/" class="post-title-link" itemprop="url">BCL文件与BCL2FAFSTQ程序简介</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-02-07 11:34:40" itemprop="dateCreated datePublished" datetime="2018-02-07T11:34:40+08:00">2018-02-07</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p><img src="/wp/f4w/2020/2016-04-0301.jpg"></p>
<h1 id="BCL文件"><a href="#BCL文件" class="headerlink" title="BCL文件"></a>BCL文件</h1><p>测序产生的原始文件是BCL（binary base call）文件，测序仪在测序的时候，每个cycle都会测量编码不同颜色的通道强度，并确定最有可能的碱基类型。Real Time Analysis (RTA) 软件会将碱基类型和可信度（一个质量分数）。与FASTQ文件不同的是，BCL文件是实时产生，每个cycle的每个tile都会有一个对应文件，文件放在</p>
<p><code>&lt;run directory&gt;/Data/Intensities/BaseCalls/L&lt;lane&gt;/C&lt;cycle&gt;.1</code></p>
<p>文件的命名</p>
<p> <code>s_&lt;lane&gt;_&lt;tile&gt;.bcl</code></p>
<h1 id="bcl2fastq"><a href="#bcl2fastq" class="headerlink" title="bcl2fastq"></a>bcl2fastq</h1><p>该文件需要通过Illunima的软件或者第三方分析工具将BCL文件转成FASTQ文件。一般而言，数据下机之后，Illumina测序仪会自动将BCL转成FASTQ文件。有时候，根据实验需要，需要自己手工将BCL文件转成FASTQ文件，比如自己设计的index中含有简并碱基，或者需要调整一下转换的参数等。 Illumina提供bcl2fastq的程序包，共离线处理BCL文件，生成FASTQ文件。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">bcl2fastq  -i /paht/to/run/Data/Intensities/BaseCalls/ \</span><br><span class="line">       -o /output/dir --sample-sheet /paht/to/run/SampleSheet.csv \</span><br><span class="line">       -R /paht/to/run/</span><br></pre></td></tr></table></figure>

<p>bcl2fastq文件有很多参数可调，比如在FASTQ中记录read的index(fastq文件中会记录配对的index具体序列，以及会生成额外对应的index文件)，可以添加–create-fastq-for-index-reads选项。 如果允许index有mismatch的话，可以通过–barcode-mismatches设置。 –fastq-compression-level可以设置生成的FASTQ.GZ文件的压缩比例 安装如下</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">unzip bcl2fastq2-v2.17.1.14.tar.zip</span><br><span class="line">tar -xvzf bcl2fastq2-v2.17.1.14.tar.gz</span><br><span class="line">./bcl2fastq/src/configure --prefix=/path/to/install/</span><br><span class="line">make</span><br><span class="line">make install</span><br></pre></td></tr></table></figure>

<p>安装过程中，如果遇到一下问题，请更新gcc版本 问题1 cc1plus: error: unrecognized command line option “-std=c++11” 问题2 undefined reference to `boost::re_detail::perl_matcher collect2: error: ld returned 1 exit status 软件选项说明如下：</p>
<figure class="highlight java"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br></pre></td><td class="code"><pre><span class="line">BCL to FASTQ file converter</span><br><span class="line">bcl2fastq v2<span class="number">.17</span><span class="number">.1</span><span class="number">.14</span></span><br><span class="line">Copyright (c) <span class="number">2007</span>-<span class="number">2015</span> Illumina, Inc.</span><br><span class="line"></span><br><span class="line">Command-line options:</span><br><span class="line">  -h [ --help ]                                     produce help message and exit</span><br><span class="line">  -v [ --version ]                                print program version information</span><br><span class="line">  -l [ --min-log-level ] arg (=INFO)              minimum log level</span><br><span class="line">                                                          recognized values: NONE, FATAL, ERROR, WARNING, INFO, DEBUG, TRACE</span><br><span class="line">  -i [ --input-dir ] arg (=&lt;runfolder-dir&gt;/Data/Intensities/BaseCalls/)</span><br><span class="line">                                                          path to input directory</span><br><span class="line">  -R [ --runfolder-dir ] arg (=./)         path to runfolder directory</span><br><span class="line">  --intensities-<span class="function">dir <span class="title">arg</span> <span class="params">(=&lt;input-dir&gt;/../)</span>        path to intensities directory</span></span><br><span class="line"><span class="function">                                                          <span class="keyword">if</span> intensities directory is specified, also input directory must be specified</span></span><br><span class="line"><span class="function">  -o [ --output-dir ] <span class="title">arg</span> <span class="params">(=&lt;input-dir&gt;)</span>                path to demultiplexed output</span></span><br><span class="line"><span class="function">  --interop-dir <span class="title">arg</span> <span class="params">(=&lt;runfolder-dir&gt;/InterOp/)</span>   path to demultiplexing statistics directory</span></span><br><span class="line"><span class="function">  --stats-dir <span class="title">arg</span> <span class="params">(=&lt;output-dir&gt;/Stats/)</span>                path to human-readable demultiplexing statistics directory</span></span><br><span class="line"><span class="function">  --reports-dir <span class="title">arg</span> <span class="params">(=&lt;output-dir&gt;/Reports/)</span>       path to reporting directory</span></span><br><span class="line"><span class="function">  --sample-sheet <span class="title">arg</span> <span class="params">(=&lt;runfolder-dir&gt;/SampleSheet.csv)</span></span></span><br><span class="line"><span class="function">                                                                              path to the sample sheet</span></span><br><span class="line"><span class="function">  --aggregated-tiles <span class="title">arg</span> <span class="params">(=AUTO)</span>                  tiles aggregation flag determining structure of input files</span></span><br><span class="line"><span class="function">                                                   recognized values:</span></span><br><span class="line"><span class="function">                                                      AUTO Try to detect correct setting</span></span><br><span class="line"><span class="function">                                                      YES  Tiles are aggregated into single input file</span></span><br><span class="line"><span class="function">                                                      NO   There are separate input files <span class="keyword">for</span> individual tiles</span></span><br><span class="line"><span class="function">                                                  </span></span><br><span class="line"><span class="function">  -r [ --loading-threads ] <span class="title">arg</span> <span class="params">(=<span class="number">4</span>)</span>               number of threads used <span class="keyword">for</span> loading BCL data</span></span><br><span class="line"><span class="function">  -d [ --demultiplexing-threads ] arg          number of threads used <span class="keyword">for</span> demultiplexing</span></span><br><span class="line"><span class="function">  -p [ --processing-threads ] arg                 number of threads used <span class="keyword">for</span> processing demultiplexed data</span></span><br><span class="line"><span class="function">  -w [ --writing-threads ] <span class="title">arg</span> <span class="params">(=<span class="number">4</span>)</span>               number of threads used <span class="keyword">for</span> writing FASTQ data</span></span><br><span class="line"><span class="function">                                                                    <span class="keyword">this</span> must not be higher than number of samples</span></span><br><span class="line"><span class="function">  --tiles arg                                 Comma-separated list of regular expressions to select only a subset of the tiles </span></span><br><span class="line"><span class="function">                                                  available in the flow-cell. Multiple entries allowed, each applies to the corresponding base-calls.</span></span><br><span class="line"><span class="function">                                                  For example:</span></span><br><span class="line"><span class="function">                                                   * to select all the tiles ending with &#x27;5&#x27; in all lanes:</span></span><br><span class="line"><span class="function">                                                       --tiles [0-9][0-9][0-9]5</span></span><br><span class="line"><span class="function">                                                   * to select tile 2 in lane 1 and all the tiles in the other lanes:</span></span><br><span class="line"><span class="function">                                                       --tiles s_1_0002,s_[2-8]</span></span><br><span class="line"><span class="function">  --minimum-trimmed-read-length <span class="title">arg</span> <span class="params">(=<span class="number">35</span>)</span>         minimum read length after adapter trimming</span></span><br><span class="line"><span class="function">  --use-bases-mask arg                                             Specifies how to use each cycle.</span></span><br><span class="line"><span class="function">  --mask-<span class="keyword">short</span>-adapter-reads <span class="title">arg</span> <span class="params">(=<span class="number">22</span>)</span>            smallest number of remaining <span class="title">bases</span> <span class="params">(after masking bases below the minimum </span></span></span><br><span class="line"><span class="params"><span class="function">                                                                           trimmed read  length)</span> below which whole read is masked</span></span><br><span class="line"><span class="function">  --adapter-stringency <span class="title">arg</span> <span class="params">(=<span class="number">0.9</span>)</span>                  adapter stringency</span></span><br><span class="line"><span class="function">  --ignore-missing-bcls                                 assume &#x27;N&#x27;/&#x27;#&#x27; <span class="keyword">for</span> missing calls</span></span><br><span class="line"><span class="function">  --ignore-missing-filter                                assume &#x27;<span class="keyword">true</span>&#x27; <span class="keyword">for</span> missing filters</span></span><br><span class="line"><span class="function">  --ignore-missing-positions                         assume \[0,i\] <span class="keyword">for</span> missing positions, where i is incremented starting from 0</span></span><br><span class="line"><span class="function">  --ignore-missing-controls                           assume 0 <span class="keyword">for</span> missing controls</span></span><br><span class="line"><span class="function">  --write-fastq-reverse-complement              Generate FASTQs containing reverse complements of actual data</span></span><br><span class="line"><span class="function">  --with-failed-reads                                        include non-PF clusters</span></span><br><span class="line"><span class="function">  --create-fastq-<span class="keyword">for</span>-index-reads                     create FASTQ files also <span class="keyword">for</span> index reads</span></span><br><span class="line"><span class="function">  --find-adapters-with-sliding-window           find adapters with simple sliding window algorithm</span></span><br><span class="line"><span class="function">  --no-bgzf-compression                                Turn off BGZF compression <span class="keyword">for</span> FASTQ files</span></span><br><span class="line"><span class="function">  --fastq-compression-level <span class="title">arg</span> <span class="params">(=<span class="number">4</span>)</span>              Zlib compression <span class="title">level</span> <span class="params">(<span class="number">1</span>-<span class="number">9</span>)</span> used <span class="keyword">for</span> FASTQ files</span></span><br><span class="line"><span class="function">  --barcode-mismatches <span class="title">arg</span> <span class="params">(=<span class="number">1</span>)</span>                   number of allowed mismatches per index</span></span><br><span class="line"><span class="function">                                                                        multiple entries, comma delimited entries, allowed</span>; each entry is applied to the </span><br><span class="line">                                                                        corresponding index;last entry applies to all remaining indices</span><br><span class="line">  --no-lane-splitting                             Do not split fastq files by lane.</span><br></pre></td></tr></table></figure>

<p>软件地址 <a target="_blank" rel="noopener" href="http://support.illumina.com.cn/downloads/bcl2fastq-conversion-software-v217.html">http://support.illumina.com.cn/downloads/bcl2fastq-conversion-software-v217.html</a> </p>
<p>文档手册 <a target="_blank" rel="noopener" href="http://support.illumina.com/content/dam/illumina-support/documents/documentation/software/_documentation/bcl2fastq/bcl2fastq/_letterbooklet_15038058brpmi.pdf">http://support.illumina.com/content/dam/illumina-support/documents/documentation/software\_documentation/bcl2fastq/bcl2fastq\_letterbooklet_15038058brpmi.pdf</a></p>
<p>##################################################################### #版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/810/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/810/" class="post-title-link" itemprop="url">PLOT | 目标区域的测序覆盖度作图</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-02-01 15:56:04" itemprop="dateCreated datePublished" datetime="2018-02-01T15:56:04+08:00">2018-02-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>不论是目标区域测序还是外显子组测序还是全基因组测序之后，我们会关注目标区域在特定深度下的覆盖度。比如20X的比例，50X的比例。如果只计算特定深度的覆盖度，我们了解不到其他深度下的覆盖度情况。如果每个都列出来，又不直观，这个时候用图片来表示就非常直观了。</p>
<h1 id="获取统计数据"><a href="#获取统计数据" class="headerlink" title="获取统计数据"></a>获取统计数据</h1><p>我们需要知道在每个深度下碱基的比例，这个时候强大的bedtools就出场了。</p>
<pre>bedtools coverage -sorted -hist -g genome.file  -b  samp.bam -a target_regions.bed ' grep ^all > samp.hist.all.txt</pre>

<p>-hist是为了获取目标区域的总结信息，以all开头，输出每个每个深度下碱基的比例，所以后续用grep ^all来过滤。</p>
<p>-sorted可以节省内存，但需要通过-g跟genome file来指定染色体的顺序。</p>
<p>genome file可以通过awk -v OFS=’\t’ {‘print $1,$2’} genome.fa.fai &gt; genome.file 得到。</p>
<p>假设我们得到了smp1.hist.all.txt、smp2.hist.all.txt、smp3.hist.all.txt、smp4.hist.all.txt、smp5.hist.all.txt、smp6.hist.all.txt、smp7.hist.all.txt、</p>
<h1 id="作图"><a href="#作图" class="headerlink" title="作图"></a>作图</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Get a list of the bedtools output files you&#x27;d like to read in</span></span><br><span class="line"></span><br><span class="line">print(labs &lt;- paste(<span class="string">&quot;&quot;</span>, gsub(<span class="string">&quot;.hist.all.txt&quot;</span>, <span class="string">&quot;&quot;</span>, files, perl=<span class="literal">TRUE</span>), sep=<span class="string">&quot;&quot;</span>))</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Create lists to hold coverage and cumulative coverage for each alignment,</span></span><br><span class="line"><span class="comment"># and read the data into these lists.</span></span><br><span class="line">cov &lt;- <span class="built_in">list</span>()</span><br><span class="line">cov_cumul &lt;- <span class="built_in">list</span>()</span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:<span class="built_in">length</span>(files)) &#123;</span><br><span class="line">    cov[[i]] &lt;- read.table(files[i])</span><br><span class="line">    <span class="comment"># The value should be 1 at 0X.</span></span><br><span class="line">    cov_cumul[[i]] &lt;- <span class="number">1</span>-<span class="built_in">cumsum</span>(<span class="built_in">c</span>(<span class="number">0</span>,cov[[i]][,<span class="number">5</span>]))</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">library(RColorBrewer)</span><br><span class="line">cols &lt;- brewer.pal(<span class="built_in">length</span>(cov), <span class="string">&quot;Dark2&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Save the graph to a file</span></span><br><span class="line">png(<span class="string">&quot;target-coverage-plots.png&quot;</span>, h=<span class="number">1000</span>, w=<span class="number">1000</span>, pointsize=<span class="number">20</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Create plot area, but do not plot anything. Add gridlines and axis labels.</span></span><br><span class="line">plot(cov[[<span class="number">1</span>]][<span class="number">2</span>:<span class="number">401</span>, <span class="number">2</span>], cov_cumul[[<span class="number">1</span>]][<span class="number">1</span>:<span class="number">400</span>], type=<span class="string">&#x27;n&#x27;</span>, xlab=<span class="string">&quot;Depth&quot;</span>, ylab=<span class="string">&quot;Fraction of capture target bases \u2265 depth&quot;</span>, ylim=<span class="built_in">c</span>(<span class="number">0</span>,<span class="number">1.0</span>), main=<span class="string">&quot;Target Region Coverage&quot;</span>)</span><br><span class="line">abline(v = <span class="number">20</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(v = <span class="number">50</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(v = <span class="number">80</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(v = <span class="number">100</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(h = <span class="number">0.50</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(h = <span class="number">0.90</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">axis(<span class="number">1</span>, at=<span class="built_in">c</span>(<span class="number">20</span>,<span class="number">50</span>,<span class="number">80</span>), labels=<span class="built_in">c</span>(<span class="number">20</span>,<span class="number">50</span>,<span class="number">80</span>))</span><br><span class="line">axis(<span class="number">2</span>, at=<span class="built_in">c</span>(<span class="number">0.90</span>), labels=<span class="built_in">c</span>(<span class="number">0.90</span>))</span><br><span class="line">axis(<span class="number">2</span>, at=<span class="built_in">c</span>(<span class="number">0.50</span>), labels=<span class="built_in">c</span>(<span class="number">0.50</span>))</span><br><span class="line"></span><br><span class="line"><span class="comment"># Actually plot the data for each of the alignments (stored in the lists).</span></span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:<span class="built_in">length</span>(cov)) points(cov[[i]][<span class="number">2</span>:<span class="number">401</span>, <span class="number">2</span>], cov_cumul[[i]][<span class="number">1</span>:<span class="number">400</span>], type=<span class="string">&#x27;l&#x27;</span>, lwd=<span class="number">3</span>, col=cols[i])</span><br><span class="line"></span><br><span class="line"><span class="comment"># Add a legend using the nice sample labeles rather than the full filenames.</span></span><br><span class="line">legend(<span class="string">&quot;topright&quot;</span>, legend=labs, col=cols, lty=<span class="number">1</span>, lwd=<span class="number">4</span>)</span><br><span class="line"></span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>

<p><img src="/wp/f4w/2020/2018-02-01-target-region-coverage.png"></p>
<p>图片也可以用ggplot2来做。横坐标是DP，纵坐标是大于等于&gt;=该DP的碱基比例。该图还有一个好处，是可以多个样本同时放在一张图上，便于进行样本间比较。</p>
<p>可以看出smp2的覆盖度最好，随着横左边的增加，下降最缓慢。smp1和smp3最不好，在20X以后，覆盖度显著下降。如果样本smp来自不同的测序实验，则smp2所用的方法是最有的。</p>
<p>最后还是那句话，图片表达的信息量多而且直观。让我们用图片说话。</p>
<p>主要CLONE参考：</p>
<p><a target="_blank" rel="noopener" href="http://www.gettinggeneticsdone.com/2014/03/visualize-coverage-exome-targeted-ngs-bedtools.html">http://www.gettinggeneticsdone.com/2014/03/visualize-coverage-exome-targeted-ngs-bedtools.html</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/805/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/805/" class="post-title-link" itemprop="url">Illumina下机FASTQ文件命名规则</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-02-01 13:30:16" itemprop="dateCreated datePublished" datetime="2018-02-01T13:30:16+08:00">2018-02-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>FASTQ文件在Illumina下机数据文件夹<em>*Data\Intensities\BaseCalls**中，类似**SampleName_S1_L001_R1_001.fastq.gz（比如NTC_S11_L001_R1_001.fastq.gz）</em>*</p>
<p>其中被下划线_分为了五个部分。</p>
<ul>
<li>  第一部分：SampleName，样本名，与上机时在Sample Sheet中填写的一致</li>
<li>  第二部分：S1，S***，S后跟的数字与样本在Sample Sheet中的顺序一致，从1开始。不能分配到确定样本的read会归到S0（Undetermined_S0）</li>
<li>  第三部分：L00*，泳道lane的编号</li>
<li>  第四部分：R*，R1表示read1，R2表示read2。R1和R2为paired end reads。同一个样本的配对的FASTQ，只有这个地方不同</li>
<li>  第五部分：001，通常为001</li>
<li><em>Naming</em>*<br>FASTQ files are named with the sample name and the sample number, which is a numeric assignment based on the order that the sample is listed in the sample sheet. </li>
<li><em>Example:</em>*<br>Data\Intensities\BaseCalls\SampleName_S1_L001_R1_001.fastq.gz<br>• SampleName-The sample name provided in the sample sheet. If a sample name is not provided, the file name includes the sample ID, which is a required field in the sample sheet and must be unique.<br>• S1-The sample number based on the order that samples are listed in the sample sheet starting with 1. In this example, S1 indicates that this sample is the first sample listed in the sample sheet.</li>
</ul>
<p><strong>NOTE</strong><br>Reads that cannot be assigned to any sample are written to a FASTQ file for sample number 0, and excluded from downstream analysis.<br>• L001-The lane number.<br>• R1-The read. In this example, R1 means Read 1. For a paired-end run, there is at least one file with R2 in the file name for Read 2. When generated, index reads are I1 or I2.<br>• 001-The last segment is always 001.</p>
<p><strong>参考：</strong></p>
<p><a target="_blank" rel="noopener" href="https://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/NamingConvention_FASTQ-files-swBS.htm">https://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/NamingConvention_FASTQ-files-swBS.htm</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/854/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/854/" class="post-title-link" itemprop="url">2018-01-09 GATK 4.0 正式发布</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-01-20 20:46:55" itemprop="dateCreated datePublished" datetime="2018-01-20T20:46:55+08:00">2018-01-20</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p><img src="https://theme.zdassets.com/theme_assets/2378360/df085f154321faac9159dda57f50103b87a4f743.png" alt="hhttps://theme.zdassets.com/theme_assets/2378360/df085f154321faac9159dda57f50103b87a4f743.png"> </p>
<p>GATK4正式版已经发布，快去体验啦。GATK4在上一年提出开源，并放出beta版本，现在终于姗姗来迟。</p>
<p>GATK4是业界第一次涵盖了胚细胞和体细胞基因型分析中的主要突变类型的基因组分析工具，且已经开源。新版本的GATK为了解决性能瓶颈近乎完全重构，提高了速度和扩展性有不失其过往的准确度。</p>
<p>GATK4包含了备受大家喜爱的pipeline和新工具，汲取了机器学习和神经网络算法的优点。</p>
<p>GATK早期版本关注检测胚细胞短突变（germline short variant），新版本将体细胞短突变（somatic short variant）检测工具Mutect2也整合在内，Mutect2整合广泛使用的算法Mutect和GATK卓越的胚细胞突变检测算法HaplotypeCaller。除了短突变检测外，GATK4增加了检测胚细胞和体细胞拷贝数变异的流程，增加了基于体细胞等位拷贝数（somatic allelic CNV (ACNV)）的肿瘤异质性评估。这些流程经过重构，无缝用于gene panel、外显子到全基因组的测序数据。</p>
<p>GATK4同时包含了早期可以使用现在依然在开发的结构变异（structural variant）检测，胚细胞拷贝数变异CNV检测使用了机器学习的方法，基于卷积神经网络（Convolutional Neural Networks，CNN）的胚细胞短突变过滤方法。</p>
<p>GATK4对性能、灵活性、速度、扩展性进行了广泛优化，并包含了可以在本地或者云设备上可以运行的点对点流程（Best practice）。</p>
<p>该版本得益于Broad研究所大规模运行基因组分析流程的科学和业务专家，来自业界Intel, Google Cloud, Cloudera, Microsoft Genmics, IBM Research, Amazon Web Services and Alibaba Cloud的工程师，都为GATK4的的开发做出来贡献。</p>
<p>2017成立的Intel-Broad基因组数据工程中心，Broad研究所的人员与Intel协同合作，对关键流程步骤做出了巨大优化。特别是Intel显著优化了基于GVCF的胚细胞joint-calling流程的扩展性，使得Broad团队可以在2个星期内完成15,000个全基因组样本的变异分析工作，而GATK3要处理3,000个全基因组样本需要至少6个星期。</p>
<p><a target="_blank" rel="noopener" href="https://software.broadinstitute.org/gatk/gatk4">https://software.broadinstitute.org/gatk/gatk4</a> <a target="_blank" rel="noopener" href="https://github.com/broadinstitute/gatk">https://github.com/broadinstitute/gatk</a> <a target="_blank" rel="noopener" href="http://jandan.net/2017/05/31/gatk4-open.html">http://jandan.net/2017/05/31/gatk4-open.html</a> <a target="_blank" rel="noopener" href="http://www.eeboard.com/news/gatk4/">http://www.eeboard.com/news/gatk4/</a> <a target="_blank" rel="noopener" href="http://ju.outofmemory.cn/entry/324861">http://ju.outofmemory.cn/entry/324861</a> <a target="_blank" rel="noopener" href="https://gatkforums.broadinstitute.org/gatk/discussion/10502/gatk-4-0-will-be-released-jan-9-2018#latest">https://gatkforums.broadinstitute.org/gatk/discussion/10502/gatk-4-0-will-be-released-jan-9-2018#latest</a></p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/798/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/798/" class="post-title-link" itemprop="url">Picard---RuntimeIOException: java.io.IOException: No space left on device</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-01-10 20:13:11" itemprop="dateCreated datePublished" datetime="2018-01-10T20:13:11+08:00">2018-01-10</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>在用Picard跑sortSam或者markDuplicate的时候，报错</p>
<p><strong>RuntimeIOException: java.io.IOException: No space left on device</strong></p>
<p>提示硬盘空间不足，实际上节点的硬盘空间是够的。这是因为Picard在做这两个处理的时候，会生成临时文件，临时文件默认储存在系统的tmp文件夹。这个文件夹内的文件存储一些程序和软件的临时文件，在RHEL6中，系统自动清理/tmp文件夹的默认时限是30天。可以通过/etc/cron.daily/tmpwatch配置,在Ubuntu中，系统自动清理/tmp文件夹的时限默认每次启动。而全基因组和全外显子组的数据非常大，现在分给系统的空间都很小，导致tmp文件夹写满而报错。</p>
<h2 id="处理方法："><a href="#处理方法：" class="headerlink" title="处理方法："></a><strong>处理方法：</strong></h2><p>在个人home文件夹下，新建一个tmp文件夹，每次运行picard的时候，指定IO临时文件夹为这个文件夹。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mkdir $HOME/tmp</span><br><span class="line">java -Xmx2g -Djava.io.tmpdir=$HOME/tmp -jar SortSam.jar SORT_ORDER=coordinate INPUT=input.bam OUTPUT=output.sort TMP_DIR=$HOME/tmp</span><br></pre></td></tr></table></figure>


<p>不能使用<del>，picard会在工作目录下创建’</del>‘文件夹。</p>
<p>参考：</p>
<p><a target="_blank" rel="noopener" href="https://www.biostars.org/p/42613/">https://www.biostars.org/p/42613/</a></p>
<p><a target="_blank" rel="noopener" href="https://www.cnblogs.com/hnrainll/archive/2011/06/08/2074976.html">https://www.cnblogs.com/hnrainll/archive/2011/06/08/2074976.html</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/793/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/793/" class="post-title-link" itemprop="url">合并bed文件中的区域</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-01-06 23:04:51" itemprop="dateCreated datePublished" datetime="2018-01-06T23:04:51+08:00">2018-01-06</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>数据分析中会经常用到bed文件，有时候bed文件中的区域有重叠时会影响统计结果，或者没有按照顺序排序时会影响代码的逻辑。合并bed文件或者文件中的区域是经常进行的。</p>
<p>我常用的工具是bedtools的merge功能，官方示例如下：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">$ cat A.bed</span><br><span class="line">chr1  100  200</span><br><span class="line">chr1  180  250</span><br><span class="line">chr1  250  500</span><br><span class="line">chr1  501  1000</span><br><span class="line">$ bedtools merge -i A.bed</span><br><span class="line">chr1  100  500</span><br><span class="line">chr1  501  1000</span><br></pre></td></tr></table></figure>

<p>bedtools的merge功能强大的地方，更在于在合并的时候，可以进行一下操作。如下操作，通过-c指定要操作的目标列，-o指定操作动作。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">$ cat A.bed</span><br><span class="line">chr1  100  200   a1  1 +</span><br><span class="line">chr1  180  250   a2  2 +</span><br><span class="line">chr1  250  500   a3  3 -</span><br><span class="line">chr1  501  1000  a4  4 +</span><br><span class="line"></span><br><span class="line">$ bedtools merge -i A.bed -S +</span><br><span class="line"># 合并正链的区域feature或者interval</span><br><span class="line">chr1  100 250</span><br><span class="line">chr1  501 1000</span><br><span class="line"></span><br><span class="line">$  bedtools merge -i A.bed -c 5,5,5,6 -o mean,min,max,collapse</span><br><span class="line"># 计算重叠区域第五列的平均值、最小值、最大值</span><br><span class="line"># 将重叠区域的第六列按顺序列出</span><br><span class="line">chr1  100 500 2 1 3 +,+,-</span><br><span class="line">chr1  501 1000  4 4 4 +</span><br></pre></td></tr></table></figure>



<p>我个人写的bash脚本如下：</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">#! /bin/bash</span><br><span class="line"></span><br><span class="line"># 1）合并所有传入脚本中的bed文件，并生成一个中间文件</span><br><span class="line">cat $@ | sort -k1,1 -k2,2n &gt; bedtools.in.sorted.bed</span><br><span class="line"># 2）合并中间文件，并输出到标准输出流中，供下一步处理</span><br><span class="line">bedtools merge -i bedtools.in.sorted.bed</span><br><span class="line"># 3）删除中间文件</span><br><span class="line">rm bedtools.in.sorted.bed</span><br></pre></td></tr></table></figure>



<h1 id="参考："><a href="#参考：" class="headerlink" title="参考："></a>参考：</h1><p><a target="_blank" rel="noopener" href="http://bedtools.readthedocs.io/en/latest/content/tools/merge.html">http://bedtools.readthedocs.io/en/latest/content/tools/merge.html</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  


  
  <nav class="pagination">
    <a class="extend prev" rel="prev" href="/page/7/"><i class="fa fa-angle-left" aria-label="Previous page"></i></a><a class="page-number" href="/">1</a><span class="space">&hellip;</span><a class="page-number" href="/page/7/">7</a><span class="page-number current">8</span><a class="page-number" href="/page/9/">9</a><span class="space">&hellip;</span><a class="page-number" href="/page/17/">17</a><a class="extend next" rel="next" href="/page/9/"><i class="fa fa-angle-right" aria-label="Next page"></i></a>
  </nav>



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Jason</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">163</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">167</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>


<!--google_adsense-->

  <div class="links-of-blogroll-title">
    <i class="fa fa-google"></i> AdSense
    <ins class="adsbygoogle"
        style="display:block"
        data-ad-client="ca-pub-9370593480269435"
        data-ad-slot="8824957305"
        data-ad-format="auto"
        data-full-width-responsive="true"></ins>
    <script>
      (adsbygoogle = window.adsbygoogle || []).push({});
    </script>
  </div>

<!--/google_adsense-->



    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Jason</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

<script data-ad-client="ca-pub-9370593480269435" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<!-- PGAD -->
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-9370593480269435"
     data-ad-slot="8824957305"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
