<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"md.zxzyl.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>


  <meta name="description" content="MAF格式Mutation Annotation Format (MAF) ，是TCGA组织对突变进行注释的格式。一些和癌症分析相关的软件，经常要求MAF格式文件作为输入。而现在经过GATK或samtools检测出突变的格式一般为VCF格式，的注释软件，即使经过SNPEff和annovar注释（当然还有VEP），结果依然为VCF格式或者tab分割的文件等。 MAF中每一列是一种注释信息，由于包含的">
<meta property="og:type" content="article">
<meta property="og:title" content="oncotator对VCF进行注释，并生成MAF格式文件">
<meta property="og:url" content="https://md.zxzyl.com/archives/769/index.html">
<meta property="og:site_name" content="Zhongxu&#39;s blog">
<meta property="og:description" content="MAF格式Mutation Annotation Format (MAF) ，是TCGA组织对突变进行注释的格式。一些和癌症分析相关的软件，经常要求MAF格式文件作为输入。而现在经过GATK或samtools检测出突变的格式一般为VCF格式，的注释软件，即使经过SNPEff和annovar注释（当然还有VEP），结果依然为VCF格式或者tab分割的文件等。 MAF中每一列是一种注释信息，由于包含的">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://md.zxzyl.com/wp/f4w/2020/2016-09-06-oncotator-annotation-maf.png">
<meta property="article:published_time" content="2016-09-06T08:00:01.000Z">
<meta property="article:modified_time" content="2016-09-06T08:00:01.000Z">
<meta property="article:author" content="Jason">
<meta property="article:tag" content="VCF">
<meta property="article:tag" content="Annotation">
<meta property="article:tag" content="MAF">
<meta property="article:tag" content="oncotator">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://md.zxzyl.com/wp/f4w/2020/2016-09-06-oncotator-annotation-maf.png">

<link rel="canonical" href="https://md.zxzyl.com/archives/769/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>oncotator对VCF进行注释，并生成MAF格式文件 | Zhongxu's blog</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a3b5d3f466edb3e3f15e8238d8f04be8";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Zhongxu's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/769/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          oncotator对VCF进行注释，并生成MAF格式文件
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-09-06 16:00:01" itemprop="dateCreated datePublished" datetime="2016-09-06T16:00:01+08:00">2016-09-06</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>MAF格式Mutation Annotation Format (MAF) ，是TCGA组织对突变进行注释的格式。一些和癌症分析相关的软件，经常要求MAF格式文件作为输入。而现在经过GATK或samtools检测出突变的格式一般为VCF格式，的注释软件，即使经过SNPEff和annovar注释（当然还有VEP），结果依然为VCF格式或者tab分割的文件等。</p>
<p>MAF中每一列是一种注释信息，由于包含的注释信息太多（<a target="_blank" rel="noopener" href="https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification;WIKISESSIONID=52395E56D6223DF1AB5840164882D34A">详见格式</a>），单纯的通过写脚本转换SNPEff或者annovar的注释文件，会变得非常麻烦而且考虑的问题可能不完全（有人实现过，通过Ensembl的VEP对VCF注释，然后转换，可以在github上搜索到）。</p>
<p>这里介绍注释软件oncotator，可以注释VCF文件，并直接生成MAF格式，相当于将VCF格式转换成MAF格式。Broad institute开发的，用起来放心哈。</p>
<h2 id="首先安装python-virtualenv"><a href="#首先安装python-virtualenv" class="headerlink" title="首先安装python virtualenv"></a>首先安装python virtualenv</h2><p>virtualenv可以创建一个虚拟的python环境，在虚拟环境中安装的包，不影响机器已经安装的python包。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">curl -O https://pypi.python.org/packages/<span class="built_in">source</span>/v/virtualenv/virtualenv-14.0.6.tar.gz</span><br><span class="line">tar -xvzf virtualenv-14.0.6.tar.gz </span><br><span class="line"><span class="built_in">cd</span> virtualenv-14.0.6/</span><br><span class="line">python setup.py install --prefix=~/pypackages/</span><br><span class="line"><span class="built_in">cd</span> ..</span><br><span class="line">rm -rf virtualenv-14.0.6*</span><br></pre></td></tr></table></figure>




<h2 id="安装oncotator依赖"><a href="#安装oncotator依赖" class="headerlink" title="安装oncotator依赖"></a>安装oncotator依赖</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">git clone  https://github.com/broadinstitute/oncotator.git</span><br><span class="line">cd oncotator</span><br><span class="line">mkdir env</span><br><span class="line">bash scripts/create_oncotator_venv.sh -e ./env</span><br></pre></td></tr></table></figure>


<h2 id="安装oncotator"><a href="#安装oncotator" class="headerlink" title="安装oncotator"></a>安装oncotator</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">source ./env/bin/activate</span><br><span class="line">python setup.py install</span><br><span class="line">deactivate</span><br></pre></td></tr></table></figure>




<h2 id="下载注释数据文件"><a href="#下载注释数据文件" class="headerlink" title="下载注释数据文件"></a>下载注释数据文件</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">wget -c http://www.broadinstitute.org/~lichtens/oncobeta/oncotator_v1_ds_Jan262015.tar.gz</span><br><span class="line">tar -xvzf  oncotator_v1_ds_Jan262015.tar.gz</span><br></pre></td></tr></table></figure>

<p>注释内容如下，包括基因组注释，蛋白注释，癌症突变注释等，用的基因组是GENCODE的hg19版本。<br><strong>Genomic Annotations</strong></p>
<p>Gene, transcript, and functional consequence annotations using GENCODE for hg19.<br>Reference sequence around a variant.<br>GC content around a variant.<br>Human DNA Repair Gene annotations from Wood et al.</p>
<p><strong>Protein Annotations</strong></p>
<p>Site-specific protein annotations from UniProt.<br>Functional impact predictions from dbNSFP.</p>
<p><strong>Cancer Variant Annotations</strong></p>
<p>Observed cancer mutation frequency annotations from COSMIC.<br>Cancer gene and mutation annotations from the Cancer GenCensus.<br>Overlapping mutations from the Cancer Cell Line Encyclopedia.<br>Cancer gene annotations from the Familial Cancer Database.<br>Cancer variant annotations from ClinVar.</p>
<p><strong>Non-Cancer Variant Annotations</strong></p>
<p>Common SNP annotations from dbSNP.<br>Variant annotations from 1000 Genomes.<br>Variant annotations from NHLBI GO Exome Sequencing Project (ESP).</p>
<h1 id="运行oncotator"><a href="#运行oncotator" class="headerlink" title="运行oncotator"></a>运行oncotator</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">#进入虚拟环境env</span><br><span class="line">source ./env/bin/activate</span><br><span class="line"># WGS = whole genome sequencing, WXS = whole exome sequencing</span><br><span class="line"># 通过annotate-manual设置MAF注释，比如样本barcode，样本来源等，至于氨基酸突变等，会自动进行注释</span><br><span class="line">oncotator --input_format=VCF --db-dir=./oncotator_v1_ds_Jan262014 --output_format=TCGAMAF --tx-mode=EFFECT --collapse-number-annotations --annotate-default=&#x27;Sequencer:Illumina HiSeq&#x27; --annotate-manual=&#x27;Sequence_Source:WGS&#x27; --annotate-manual=&#x27;Tumor_Sample_Barcode:Tumor&#x27;  --annotate-manual=&#x27;Matched_Norm_Sample_Barcode:normal&#x27;  input.vcf output.maf hg19</span><br><span class="line">deactivate</span><br></pre></td></tr></table></figure>




<h1 id="结果"><a href="#结果" class="headerlink" title="结果"></a>结果</h1><p><img src="/wp/f4w/2020/2016-09-06-oncotator-annotation-maf.png"></p>
<span id="more"></span>

<p>下面的内容如果在浏览器中有换行，可以直接复制到文本编辑器中看。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">#version 2.4</span><br><span class="line">## Oncotator v1.10.0.0dev &#x27; Flat File Reference hg19 &#x27; GENCODE v19 EFFECT &#x27; UniProt_AAxform 2014_12 &#x27; dbSNP build 142 &#x27; 1000gp3 20130502 &#x27; ESP 6500SI-V2 &#x27; ESP 6500SI-V2 &#x27; dbNSFP v2.4 &#x27; COSMIC v62_291112 &#x27; UniProt_AA 2014_12 &#x27; ORegAnno UCSC Track &#x27; CCLE_By_GP 09292010 &#x27; ClinVar 12.03.20 &#x27; Ensembl ICGC MUCOPA &#x27; ACHILLES_Lineage_Results 110303 &#x27; COSMIC_Tissue 291112 &#x27; CGC full_2012-03-15 &#x27; TCGAScape 110405 &#x27; HGNC Sept172014 &#x27; Familial_Cancer_Genes 20110905 &#x27; CCLE_By_Gene 09292010 &#x27; MutSig Published Results 20110905 &#x27; UniProt 2014_12 &#x27; COSMIC_FusionGenes v62_291112 &#x27; HumanDNARepairGenes 20110905 &#x27; TUMORScape 20100104 &#x27; gencode_xref_refseq metadata_v19 </span><br><span class="line">Hugo_Symbol     Entrez_Gene_Id  Center  NCBI_Build      Chromosome      Start_position  End_position    Strand  Variant_Classification     Variant_Type    Reference_Allele        Tumor_Seq_Allele1       Tumor_Seq_Allele2       dbSNP_RS        dbSNP_Val_Status  Tumor_Sample_Barcode     Matched_Norm_Sample_Barcode     Match_Norm_Seq_Allele1  Match_Norm_Seq_Allele2  Tumor_Validation_Allele1  Tumor_Validation_Allele2 Match_Norm_Validation_Allele1   Match_Norm_Validation_Allele2   Verification_Status     Validation_Status Mutation_Status  Sequencing_Phase        Sequence_Source Validation_Method       Score   BAM_file        Sequencer       Tumor_Sample_UUID  Matched_Norm_Sample_UUID        Genome_Change   Annotation_Transcript   Transcript_Strand       Transcript_Exon Transcript_Position        cDNA_Change     Codon_Change    Protein_Change  Other_Transcripts       Refseq_mRNA_Id  Refseq_prot_Id  SwissProt_acc_Id   SwissProt_entry_Id      Description     UniProt_AApos   UniProt_Region  UniProt_Site    UniProt_Natural_Variations      UniProt_Experimental_Info  GO_Biological_Process   GO_Cellular_Component   GO_Molecular_Function   COSMIC_overlapping_mutations    COSMIC_fusion_genes        COSMIC_tissue_types_affected    COSMIC_total_alterations_in_gene        Tumorscape_Amplification_Peaks  Tumorscape_Deletion_Peaks  TCGAscape_Amplification_Peaks   TCGAscape_Deletion_Peaks        DrugBank        ref_context     gc_contentCCLE_ONCOMAP_overlapping_mutations       CCLE_ONCOMAP_total_mutations_in_gene    CGC_Mutation_Type       CGC_Translocation_Partner CGC_Tumor_Types_Somatic  CGC_Tumor_Types_Germline        CGC_Other_Diseases      DNARepairGenes_Role     FamilialCancerDatabase_Syndromes   MUTSIG_Published_Results        OREGANNO_ID     OREGANNO_Values 1000gp3_AA      1000gp3_AC      1000gp3_AF      1000gp3_AFR_AF     1000gp3_AMR_AF  1000gp3_AN      1000gp3_CIEND   1000gp3_CIPOS   1000gp3_CS      1000gp3_DP      1000gp3_EAS_AF  1000gp3_EN1000gp3_EUR_AF   1000gp3_IMPRECISE       1000gp3_MC      1000gp3_MEINFO  1000gp3_MEND    1000gp3_MLEN    1000gp3_MSTART  1000gp3_NS1000gp3_SAS_AF   1000gp3_SVLEN   1000gp3_SVTYPE  1000gp3_TSD     ACHILLES_Lineage_Results_Top_Genes      CGC_Cancer Germline Mut CGC_Cancer Molecular Genetics      CGC_Cancer Somatic Mut  CGC_Cancer Syndrome     CGC_Chr CGC_Chr Band    CGC_GeneID      CGC_Name  CGC_Other Germline Mut   CGC_Tissue Type COSMIC_n_overlapping_mutations  COSMIC_overlapping_mutation_descriptions        COSMIC_overlapping_primary_sites   ClinVar_ASSEMBLY        ClinVar_HGMD_ID ClinVar_SYM     ClinVar_TYPE    ClinVar_rs      ESP_AA  ESP_AAC ESP_AA_AC  ESP_AA_AGE      ESP_AA_GTC      ESP_AvgAAsampleReadDepth        ESP_AvgEAsampleReadDepth        ESP_AvgSampleReadDepth  ESP_CA     ESP_CDP ESP_CG  ESP_CP  ESP_Chromosome  ESP_DBSNP       ESP_DP  ESP_EA_AC       ESP_EA_AGE      ESP_EA_GTC      ESP_EXOME_CHIP     ESP_FG  ESP_GL  ESP_GM  ESP_GS  ESP_GTC ESP_GTS ESP_GWAS_PUBMED ESP_MAF ESP_PH  ESP_PP  ESP_Position    ESP_TAC ESP_TotalAAsamplesCovered  ESP_TotalEAsamplesCovered       ESP_TotalSamplesCovered Ensembl_so_accession    Ensembl_so_term Familial_Cancer_Genes_Reference    Familial_Cancer_Genes_Synonym   HGNC_Accession Numbers  HGNC_CCDS IDs   HGNC_Chromosome HGNC_Date Modified      HGNC_Date Name Changed     HGNC_Date Symbol Changed        HGNC_Ensembl Gene ID    HGNC_Ensembl ID(supplied by Ensembl)    HGNC_Enzyme IDs    HGNC_Gene family description    HGNC_HGNC ID    HGNC_Locus Group        HGNC_Locus Type HGNC_Name Synonyms      HGNC_OMIM ID(supplied by NCBI)     HGNC_Previous Names     HGNC_Previous Symbols   HGNC_Primary IDs        HGNC_Pubmed IDs HGNC_Record Type  HGNC_RefSeq(supplied by NCBI)    HGNC_Secondary IDs      HGNC_Status     HGNC_Synonyms   HGNC_UCSC ID(supplied by UCSC)  HGNC_UniProt ID(supplied by UniProt)       HGNC_VEGA IDs   HGVS_coding_DNA_change  HGVS_genomic_change     HGVS_protein_change     ORegAnno_bin       UniProt_alt_uniprot_accessions  alt_allele_seen annotation_transcript   build   ccds_id dbNSFP_1000Gp1_AC       dbNSFP_1000Gp1_AF  dbNSFP_1000Gp1_AFR_AC   dbNSFP_1000Gp1_AFR_AF   dbNSFP_1000Gp1_AMR_AC   dbNSFP_1000Gp1_AMR_AF   dbNSFP_1000Gp1_ASN_AC   dbNSFP_1000Gp1_ASN_AF      dbNSFP_1000Gp1_EUR_AC   dbNSFP_1000Gp1_EUR_AF   dbNSFP_Ancestral_allele dbNSFP_CADD_phred       dbNSFP_CADD_raw    dbNSFP_CADD_raw_rankscore       dbNSFP_ESP6500_AA_AF    dbNSFP_ESP6500_EA_AF    dbNSFP_Ensembl_geneid   dbNSFP_Ensembl_transcriptid        dbNSFP_FATHMM_pred      dbNSFP_FATHMM_rankscore dbNSFP_FATHMM_score     dbNSFP_GERP++_NR        dbNSFP_GERP++_RS  dbNSFP_GERP++_RS_rankscore       dbNSFP_Interpro_domain  dbNSFP_LRT_Omega        dbNSFP_LRT_converted_rankscore  dbNSFP_LRT_pred dbNSFP_LRT_score   dbNSFP_LR_pred  dbNSFP_LR_rankscore     dbNSFP_LR_score dbNSFP_MutationAssessor_pred    dbNSFP_MutationAssessor_rankscore  dbNSFP_MutationAssessor_score   dbNSFP_MutationTaster_converted_rankscore       dbNSFP_MutationTaster_pred      dbNSFP_MutationTaster_score        dbNSFP_Polyphen2_HDIV_pred      dbNSFP_Polyphen2_HDIV_rankscore dbNSFP_Polyphen2_HDIV_score     dbNSFP_Polyphen2_HVAR_pred dbNSFP_Polyphen2_HVAR_rankscore dbNSFP_Polyphen2_HVAR_score     dbNSFP_RadialSVM_pred   dbNSFP_RadialSVM_rankscoredbNSFP_RadialSVM_score   dbNSFP_Reliability_index        dbNSFP_SIFT_converted_rankscore dbNSFP_SIFT_pred        dbNSFP_SIFT_score dbNSFP_SLR_test_statistic        dbNSFP_SiPhy_29way_logOdds      dbNSFP_SiPhy_29way_logOdds_rankscore    dbNSFP_SiPhy_29way_pi   dbNSFP_UniSNP_ids  dbNSFP_Uniprot_aapos    dbNSFP_Uniprot_acc      dbNSFP_Uniprot_id       dbNSFP_aaalt    dbNSFP_aapos    dbNSFP_aapos_FATHMM        dbNSFP_aapos_SIFT       dbNSFP_aaref    dbNSFP_cds_strand       dbNSFP_codonpos dbNSFP_fold-degenerate  dbNSFP_genename    dbNSFP_hg18_pos(1-coor) dbNSFP_phastCons100way_vertebrate       dbNSFP_phastCons100way_vertebrate_rankscore     dbNSFP_phastCons46way_placental    dbNSFP_phastCons46way_placental_rankscore       dbNSFP_phastCons46way_primate   dbNSFP_phastCons46way_primate_rankscore    dbNSFP_phyloP100way_vertebrate  dbNSFP_phyloP100way_vertebrate_rankscore        dbNSFP_phyloP46way_placental    dbNSFP_phyloP46way_placental_rankscore     dbNSFP_phyloP46way_primate      dbNSFP_phyloP46way_primate_rankscore    dbNSFP_refcodon entrez_gene_id     gc_content_full gencode_transcript_name gencode_transcript_status       gencode_transcript_tags gencode_transcript_type    gene_type       havana_transcript       id      qual    refseq_mrna_id  secondary_variant_classification</span><br><span class="line">MT-ND6  4541    __UNKNOWN__     __UNKNOWN__     M       16322   16322   __UNKNOWN__     5&#x27;Flank SNP     T       T       C         Tumore   normal  __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__        __UNKNOWN__     __UNKNOWN__     WGS     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     Illumina HiSeq  Tumore  N63_S1.variant     g.chrM:16322T&gt;C ENST00000361681.2       -       0       0                               MT-TP_ENST00000387461.2_RNA&#x27;MT-TE_ENST00000387459.1_RNA&#x27;MT-TT_ENST00000387460.2_RNA                        P03923  NU6M_HUMAN      mitochondrially encoded NADH dehydrogenase 6                                               cellular metabolic process (GO:0044237)&#x27;respiratory electron transport chain (GO:0022904)&#x27;small molecule metabolic process (GO:0044281)    integral component of membrane (GO:0016021)&#x27;mitochondrial inner membrane (GO:0005743)&#x27;respiratory chain (GO:0070469)       NADH dehydrogenase (ubiquinone) activity (GO:0008137)             breast(1)&#x27;endometrium(10)&#x27;kidney(17)&#x27;urinary_tract(1)    29                                              GTACATAAAGTCATTTACCGT   0.478                                                                                                                                SO:0001631       upstream_gene_variant                                   mitochondria    2011-07-04      2005-02-15      2005-02-16ENSG00000198695  ENSG00000198695 1.6.5.3 &quot;&quot;&quot;Mitochondrial respiratory chain complex / Complex I&quot;&quot;&quot;       7462    protein-coding gengene with protein product        &quot;&quot;&quot;complex I ND6 subunit&quot;&quot;, &quot;&quot;NADH-ubiquinone oxidoreductase chain 6&quot;&quot;&quot; 516006  &quot;&quot;&quot;NADH dehydrogenase 6&quot;&quot;&quot; MTND6                   Standard                        Approved        NAD6, ND6               P03923                  chrM.hg19:g.16322T&gt;C       Exception_encountered           Q34774&#x27;Q8HG30   True    ENST00000361681.2       hg19                      0.478    MT-ND6-201      KNOWN   basic&#x27;appris_principal  protein_coding  protein_coding                  185.84  YP_003024037</span><br><span class="line">Unknown 0       __UNKNOWN__     __UNKNOWN__     1       672209  672209  __UNKNOWN__     IGR     SNP     G       G       A         Tumore   normal  __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__        __UNKNOWN__     __UNKNOWN__     WGS     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     Illumina HiSeq  Tumore  N63_S1.variant     g.chr1:672209G&gt;A                                                                RP5-857K21.4 (16629 upstream) : RP11-206L10.3 (4983 downstream)                                                                                                      taaggacagcgcaaccttgat    0.493                                                                                                     SO:0001628       intergenic_variant                                                                                                chr1.hg19:g.672209G&gt;A                            True            hg19                                                              0.493                                                    rs79279787      56.77</span><br><span class="line">FAM87B  400728  __UNKNOWN__     __UNKNOWN__     1       752721  752721  __UNKNOWN__     lincRNA SNP     A       A       G         Tumore   normal  __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__        __UNKNOWN__     __UNKNOWN__     WGS     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     Illumina HiSeq  Tumore  N63_S1.variant     g.chr1:752721A&gt;G        ENST00000326734.1       +       0       0                               RP11-206L10.10_ENST00000435300.1_RNA       NR_103536.1                             family with sequence similarity 87, member B                      AGAAAGGAAAAGGGAAGAATG    0.488                                                                                                   .&#x27;&#x27;&#x27;       3272    0.653355        0.2905  0.7363  5008    ,       ,               22729   0.7659          0.839   False           ,,2504     0.7781                                                                                                                  0 AK097327         1p36.33 2013-02-01                      ENSG00000177757 ENSG00000177757         &quot;&quot;&quot;Long non-coding RNAs&quot;&quot;&quot;      32236      non-coding RNA  RNA, long non-coding                                                    Standard        NR_103536         Approved FLJ40008                Q8N852  OTTHUMG00000002471              chr1.hg19:g.752721A&gt;G                           True    ENST00000326734.1  hg19                                                                                                              0.488    FAM87B-001      KNOWN   basic   lincRNA lincRNA OTTHUMT00000007025.1    rs3131972       951.77</span><br><span class="line">FAM41C  284593  __UNKNOWN__     __UNKNOWN__     1       804115  804115  __UNKNOWN__     lincRNA SNP     G       G       A         Tumore   normal  __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__        __UNKNOWN__     __UNKNOWN__     WGS     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     Illumina HiSeq  Tumore  N63_S1.variant     g.chr1:804115G&gt;A        ENST00000446136.1       -       0       1202                                    NR_027055.family with sequence similarity 41, member C                                                                                       AATCCCTAGAGACAACCTACC    0.368                                                                                                   .&#x27;&#x27;&#x27;       4587    0.915935        0.8699  0.9236  5008    ,       ,               17607   0.8423          0.993   False           ,,2504     0.9693                                                                                                                  0 BC047940         1p36.33 2013-01-16      2011-08-31      2011-08-31      ENSG00000230368 ENSG00000230368         &quot;&quot;&quot;Long non-coding RNAs&quot;&quot;&quot; 27635   non-coding RNA  RNA, long non-coding                                            12477932        Standard        NR_027055          Approved                uc001abt.4              OTTHUMG00000002469              chr1.hg19:g.804115G&gt;A             True     ENST00000446136.1       hg19                                                                                              0.368    FAM41C-001      KNOWN   basic   lincRNA lincRNA OTTHUMT00000007021.1    rs9725068       68.77   NR_027055</span><br><span class="line">FAM41C  284593  __UNKNOWN__     __UNKNOWN__     1       808631  808631  __UNKNOWN__     lincRNA SNP     G       G       A         Tumore   normal  __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__        __UNKNOWN__     __UNKNOWN__     WGS     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     Illumina HiSeq  Tumore  N63_S1.variant     g.chr1:808631G&gt;A        ENST00000446136.1       -       0       1202                                    NR_027055.family with sequence similarity 41, member C                                                                                       ACTGATGACCGTATACCTGGC    0.547                                                                                                   .&#x27;&#x27;&#x27;       2271    0.453474        0.115   0.6527  5008    ,       ,               21503   0.4167          0.7724  False           ,,2504     0.4796                                                                                                                  0 BC047940         1p36.33 2013-01-16      2011-08-31      2011-08-31      ENSG00000230368 ENSG00000230368         &quot;&quot;&quot;Long non-coding RNAs&quot;&quot;&quot; 27635   non-coding RNA  RNA, long non-coding                                            12477932        Standard        NR_027055          Approved                uc001abt.4              OTTHUMG00000002469              chr1.hg19:g.808631G&gt;A             True     ENST00000446136.1       hg19                                                                                              0.547    FAM41C-001      KNOWN   basic   lincRNA lincRNA OTTHUMT00000007021.1    rs11240779      412.77  NR_027055</span><br><span class="line">FAM41C  284593  __UNKNOWN__     __UNKNOWN__     1       808922  808922  __UNKNOWN__     lincRNA SNP     G       G       A         Tumore   normal  __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__        __UNKNOWN__     __UNKNOWN__     WGS     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     Illumina HiSeq  Tumore  N63_S1.variant     g.chr1:808922G&gt;A        ENST00000446136.1       -       0       1202                                    NR_027055.family with sequence similarity 41, member C                                                                                       CCAAGGTGTCGGACACCGTGG    0.532                                                                                                   .&#x27;&#x27;&#x27;       4921    0.982628        0.9554  0.9942  5008    ,       ,               21602   0.9841          1.0     False           ,,2504     0.9918                                                                                                                  0 BC047940         1p36.33 2013-01-16      2011-08-31      2011-08-31      ENSG00000230368 ENSG00000230368         &quot;&quot;&quot;Long non-coding RNAs&quot;&quot;&quot; 27635   non-coding RNA  RNA, long non-coding                                            12477932        Standard        NR_027055          Approved                uc001abt.4              OTTHUMG00000002469              chr1.hg19:g.808922G&gt;A             True     ENST00000446136.1       hg19                                                                                              0.532    FAM41C-001      KNOWN   basic   lincRNA lincRNA OTTHUMT00000007021.1    rs6594027       815.77  NR_027055</span><br><span class="line">FAM41C  284593  __UNKNOWN__     __UNKNOWN__     1       808928  808928  __UNKNOWN__     lincRNA SNP     C       C       T         Tumore   normal  __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__        __UNKNOWN__     __UNKNOWN__     WGS     __UNKNOWN__     __UNKNOWN__     __UNKNOWN__     Illumina HiSeq  Tumore  N63_S1.variant     g.chr1:808928C&gt;T        ENST00000446136.1       -       0       1202                                    NR_027055.family with sequence similarity 41, member C                                                                                       TGTCGGACACCGTGGTGGAGC    0.517                                                                                                   .&#x27;&#x27;&#x27;       2265    0.452276        0.1172  0.6527  5008    ,       ,               22141   0.4157          0.7734  False           ,,2504     0.4703                                                                                                                  0 BC047940         1p36.33 2013-01-16      2011-08-31      2011-08-31      ENSG00000230368 ENSG00000230368         &quot;&quot;&quot;Long non-coding RNAs&quot;&quot;&quot; 27635   non-coding RNA  RNA, long non-coding                                            12477932        Standard        NR_027055          Approved                uc001abt.4              OTTHUMG00000002469              chr1.hg19:g.808928C&gt;T             True     ENST00000446136.1       hg19                                                                                              0.517    FAM41C-001      KNOWN   basic   lincRNA lincRNA OTTHUMT00000007021.1    rs11240780      259.77  NR_027055</span><br></pre></td></tr></table></figure>




<h2 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h2><p><a target="_blank" rel="noopener" href="http://dx.doi.org/10.1002/humu.22771">http://dx.doi.org/10.1002/humu.22771</a></p>
<p><a target="_blank" rel="noopener" href="http://portals.broadinstitute.org/oncotator/">http://portals.broadinstitute.org/oncotator/</a><br>这个网页里面包含一个自动安装依赖的脚本</p>
<p><a target="_blank" rel="noopener" href="http://gatkforums.broadinstitute.org/gatk/discussion/4153/what-are-the-platform-requirements-and-software-dependencies-for-running-oncotator">http://gatkforums.broadinstitute.org/gatk/discussion/4153/what-are-the-platform-requirements-and-software-dependencies-for-running-oncotator</a></p>
<p>If you want to create indexed VCF files for use with Oncotator, you will need to have Samtools and Tabix installed and on your path.</p>
<p><a target="_blank" rel="noopener" href="http://gatkforums.broadinstitute.org/gatk/discussion/4154/howto-install-and-run-oncotator-for-the-first-time">http://gatkforums.broadinstitute.org/gatk/discussion/4154/howto-install-and-run-oncotator-for-the-first-time</a></p>
<p><a target="_blank" rel="noopener" href="http://gatkforums.broadinstitute.org/gatk/discussion/4162/howto-set-up-a-virtual-environment-and-install-all-dependencies-for-oncotator">http://gatkforums.broadinstitute.org/gatk/discussion/4162/howto-set-up-a-virtual-environment-and-install-all-dependencies-for-oncotator</a></p>
<p><a target="_blank" rel="noopener" href="https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification;WIKISESSIONID=52395E56D6223DF1AB5840164882D34A">https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification;WIKISESSIONID=52395E56D6223DF1AB5840164882D34A</a></p>
<p><a target="_blank" rel="noopener" href="https://github.com/broadinstitute/oncotator">https://github.com/broadinstitute/oncotator</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>####################################################################</p>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/VCF/" rel="tag"># VCF</a>
              <a href="/tags/Annotation/" rel="tag"># Annotation</a>
              <a href="/tags/MAF/" rel="tag"># MAF</a>
              <a href="/tags/oncotator/" rel="tag"># oncotator</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/archives/768/" rel="prev" title="Convert gtf to bed12 format --- gtf2bed">
      <i class="fa fa-chevron-left"></i> Convert gtf to bed12 format --- gtf2bed
    </a></div>
      <div class="post-nav-item">
    <a href="/archives/770/" rel="next" title="利用genome music分析癌症样本中显著突变的基因和相关通路">
      利用genome music分析癌症样本中显著突变的基因和相关通路 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E9%A6%96%E5%85%88%E5%AE%89%E8%A3%85python-virtualenv"><span class="nav-number">1.</span> <span class="nav-text">首先安装python virtualenv</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%AE%89%E8%A3%85oncotator%E4%BE%9D%E8%B5%96"><span class="nav-number">2.</span> <span class="nav-text">安装oncotator依赖</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%AE%89%E8%A3%85oncotator"><span class="nav-number">3.</span> <span class="nav-text">安装oncotator</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#%E4%B8%8B%E8%BD%BD%E6%B3%A8%E9%87%8A%E6%95%B0%E6%8D%AE%E6%96%87%E4%BB%B6"><span class="nav-number">4.</span> <span class="nav-text">下载注释数据文件</span></a></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E8%BF%90%E8%A1%8Concotator"><span class="nav-number"></span> <span class="nav-text">运行oncotator</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E7%BB%93%E6%9E%9C"><span class="nav-number"></span> <span class="nav-text">结果</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#%E5%8F%82%E8%80%83"><span class="nav-number">1.</span> <span class="nav-text">参考</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Jason</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">163</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">167</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>


<!--google_adsense-->

  <div class="links-of-blogroll-title">
    <i class="fa fa-google"></i> AdSense
    <ins class="adsbygoogle"
        style="display:block"
        data-ad-client="ca-pub-9370593480269435"
        data-ad-slot="8824957305"
        data-ad-format="auto"
        data-full-width-responsive="true"></ins>
    <script>
      (adsbygoogle = window.adsbygoogle || []).push({});
    </script>
  </div>

<!--/google_adsense-->



    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Jason</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

<script data-ad-client="ca-pub-9370593480269435" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<!-- PGAD -->
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-9370593480269435"
     data-ad-slot="8824957305"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
