---
title: Fixation index (FST)
tags:
  - Population
url: archives/1186/index.html
id: 1186
categories:
  - Default Category
date: 2020-11-02 21:03:23
---

# Fixation index (FST)

整理来源 http://www.uwyo.edu/dbmcd/popecol/maylects/fst.html，把这个课程的步骤用表格来表示

|                |                                                   |                                  | Subpopulation 1                                              | Subpopulation 2                                       | Subpopulation 3                                          | Total |
| -------------- | ------------------------------------------------- | -------------------------------- | ------------------------------------------------------------ | ----------------------------------------------------- | -------------------------------------------------------- | ----- |
|                | Genotype                                          | AA                               | 125                                                          | 50                                                    | 100                                                      |       |
|                |                                                   | Aa                               | 250                                                          | 30                                                    | 500                                                      |       |
|                |                                                   | aa                               | 125                                                          | 20                                                    | 400                                                      |       |
|                | Number of individual                              |                                  | 500                                                          | 100                                                   | 1000                                                     | 1600  |
|                | Number of alleles                                 |                                  | 1000                                                         | 200                                                   | 2000                                                     | 3200  |
|                |                                                   |                                  |                                                              |                                                       |                                                          |       |
| Step 1.        |                                                   |                                  | Calculate the gene   (allele) frequencies                    |                                                       |                                                          |       |
|                | Observed allele frequency                         | A (p)                            | (125*2+250)/1000=0.5                                         | (50*2+30)/200=0.65                                    | (2*100+500)/2000=0.35                                    |       |
|                |                                                   | a (q)                            | 0.5                                                          | 0.35                                                  | 0.65                                                     |       |
|                |                                                   |                                  |                                                              |                                                       |                                                          |       |
| Step 2.        |                                                   |                                  | Calculate the expected   genotypic counts under Hardy-Weinberg Equilibrium, and then calculate the   excess or deficiency of homozygotes in each subpopulation.      Summary of homozygote deficiency or excess relative to HWE:          Pop. 1.  Observed = Expected: perfect fit          Pop. 2.  Excess of 15.5 homozygotes: some inbreeding          Pop. 3.  Deficiency of 45 homozygotes: outbred or   experiencing a Wahlund effect (isolate breaking). |                                                       |                                                          |       |
|                | Expected allele frequency                         | AA                               | 500*0.5^2 = 125 (= observed)                                 | 100*0.65^2 = 42.25 (observed has excess of 7.75)      | 1,000*0.35^2 = 122.5   (observed has deficiency of 22.5) |       |
|                |                                                   | Aa                               | 500*2*0.5*0.5 = 250  (=   observed)                          | 100*2*0.65*0.35 = 45.5 (observed has deficit of 15.5) | 1,000*2*0.65*0.35 = 455   (observed has excess of 45)    |       |
|                |                                                   | aa                               | 500*0.5^2 = 125 (= observed)                                 | 100*0.35^2 = 12.25 (observed has excess of 7.75)      | 1,000*0.35^2 = 422.5   (observed has deficiency of 22.5) |       |
|                |                                                   |                                  |                                                              |                                                       |                                                          |       |
| Step 3.        |                                                   |                                  | Calculate the local   observed heterozygosities of each subpopulation (we will call them Hobs s,   where the s subscript refers to the sth of n populations -- 3 in this   example). |                                                       |                                                          |       |
|                | Local observed   heterozygosities                 |                                  | 250/500 = 0.5 (Hobs 1)                                       | 30/100 = 0.3 (Hobs 2)                                 | 500/1000 = 0.5(Hobs 3)                                   |       |
|                |                                                   |                                  |                                                              |                                                       |                                                          |       |
| Step 4.        |                                                   |                                  | Calculate the local   expected heterozygosity, or gene diversity, of each subpopulation       Hexp = 2pq |                                                       |                                                          |       |
|                | Local expected   heterozygosity                   |                                  | 2*0.5*0.5=0.5 (Hexp 1)                                       | 2*0.65*3.5=0.455 (Hexp 2)                             | 2*0.35*0.65=0.455 (Hexp 2)                               |       |
|                |                                                   |                                  |                                                              |                                                       |                                                          |       |
| Step 5.        |                                                   |                                  | Calculate the local   inbreeding coefficient of each subpopulation      F = (Hexps -Hobs)/Hexp      [positive F means fewer heterozygotes than expected indicates   inbreeding]       [negative F means more heterozygotes   than expected means excess outbreeding] |                                                       |                                                          |       |
|                |                                                   |                                  | F1=(0.5—0.5)/0.5=0                                           | F2=(0.455—0.3)/0.455=0.341                            | F3=(0.455—0.5)/0.455=-0.099                              |       |
|                |                                                   |                                  |                                                              |                                                       |                                                          |       |
| Step 6. and 7. |                                                   |                                  | Calculate p-bar   (p-bar, the frequency of allele A) over the total population.      Calculate q-bar (q-bar, the frequency of allele a) over the total   population.      Check: p-bar + q-bar =  1.0 |                                                       |                                                          |       |
|                | the frequency of allele over the total population | p-bar                            | (0.5*1000+0.65*200+0.35*2000)/3200=0.4156                    |                                                       |                                                          |       |
|                |                                                   | q-bar                            | (0.5*1000+0.35*200+0.65*2000)/3200=0.5844                    |                                                       |                                                          |       |
|                |                                                   |                                  |                                                              |                                                       |                                                          |       |
| Step 8.        |                                                   |                                  | Calculate the global   heterozygosity indices (over Individuals, Subpopulations and Total   population)      HI based on observed heterozygosities in individuals in subpopulations         HS based on expected heterozygosities in subpopulations      HT based on expected heterozygosities for overall total population |                                                       |                                                          |       |
|                |                                                   | HI (observed)                    | (0.5*500+0.3*100+0.5*1000)/1600=0.4875                       |                                                       |                                                          |       |
|                |                                                   | HS (expected)                    | (0.5*500+0.455*100+0.455*1000)/1600=0.4691                   |                                                       |                                                          |       |
|                |                                                   | HT (in overall total population) | 2*p-bar *q-bar     = 2 * 0.4156 * 0.5844  = 0.4858           |                                                       |                                                          |       |
|                |                                                   |                                  |                                                              |                                                       |                                                          |       |
| Step 9.        |                                                   |                                  | Calculate the global   F-statistics      Compare and contrast the global FISbelow with the "local inbreeding   coefficient" Fs of Step 5.      Here we are using a weighted average of the individual heterozygosities   over all the subpopulations.      Both FIS and  Fs are, however, based   on the observed heterozygosities,              whereas FST and FIT are based   on expected heterozygosities. |                                                       |                                                          |       |
|                |                                                   | FIS                              | (Hs-Hi)/Hs=(0.4691-0.4875)/0.4691=-0.0393                    |                                                       |                                                          |       |
|                |                                                   | FST                              | (Ht-Hs)/Ht=(0.4858-0.4691)/0.4858=-0.0344                    |                                                       |                                                          |       |
|                |                                                   | FIT                              | (Ht-Hi)/Ht=(0.4858-0.4875)/0.4858=-0.0036                    |                                                       |                                                          |       |
| Step 10        |                                                   | conclusions                      |                                                              |                                                       |                                                          |       |

Finally, draw some   conclusions about the genetic structure of the population and its   subpopulations.      1)  One of the possible HWE   conclusions we could make:      Pop. 1 is consistent with HWE (results of Step 

2) Two of the possible "local   inbreeding" conclusions we could make from Step 5:      

Pop. 2 is inbred (results of Step 5), and    

 Pop. 3 may have disassortative mating or be experiencing a Wahlund effect   (more heterozygotes than expected).      

3)  Conclusion concerning overall   degree of genetic differentiation (FST)      

Subdivision of populations, possibly due to genetic drift, accounts for approx. 3.4% of the total genetic variation (result of Eqn FST.8 FST   calculation in Step 9),      

4)  No excess or deficiency of   heterozygotes over the total population (FIT    is nearly zero).