<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"md.zxzyl.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>


  <meta property="og:type" content="website">
<meta property="og:title" content="Zhongxu&#39;s blog">
<meta property="og:url" content="https://md.zxzyl.com/page/2/index.html">
<meta property="og:site_name" content="Zhongxu&#39;s blog">
<meta property="og:locale" content="en_US">
<meta property="article:author" content="Jason">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://md.zxzyl.com/page/2/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : true,
    isPost : false,
    lang   : 'en'
  };
</script>

  <title>Zhongxu's blog</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a3b5d3f466edb3e3f15e8238d8f04be8";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Zhongxu's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content index posts-expand">
            
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1287/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1287/" class="post-title-link" itemprop="url">Take only memories, leave only footprints</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-07-19 09:58:20" itemprop="dateCreated datePublished" datetime="2021-07-19T09:58:20+08:00">2021-07-19</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p><img src="/wp/f4w/2021/2021-09-09-Longbeach.jpg"></p>
<p><img src="/wp/f4w/2021/2021-09-09-Sequoia1.jpg"></p>
<p><img src="/wp/f4w/2021/2021-09-09-generalSherman.jpg"></p>
<p><img src="/wp/f4w/2021/2021-09-09-Sequoia-HumeLake.jpg"></p>
<p><img src="/wp/f4w/2021/2021-09-09-Zion1.jpg"></p>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/1287/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1269/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1269/" class="post-title-link" itemprop="url">用SRA-Explorer辅助下载测序数据</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-06-01 10:35:20" itemprop="dateCreated datePublished" datetime="2021-06-01T10:35:20+08:00">2021-06-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>下载数据的时候，偶然碰到了SRA-Explorer，感觉挺好用的，地址：<a target="_blank" rel="noopener" href="https://sra-explorer.info/">https://sra-explorer.info/</a></p>
<p>这个页面本身非常小，见<a target="_blank" rel="noopener" href="https://github.com/ewels/sra-explorer">https://github.com/ewels/sra-explorer</a>，利用的是SRA API。</p>
<p>检索好之后，选择你想下载的样本，点击Add ** to collection，然后点击右上角saved datasets，页面下方就可以原始的fastq的链接，用curl下载fastq的命令，用aspera下载fastq的命令，还有下载SRA的命令，以及样本的metadata。非常好用。</p>
<p><img src="/wp/f4w/2021/2021-06-01-SRA-Explorer.png"></p>
<p>和SRA的Run Selector类似，<a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/Traces/study/">https://www.ncbi.nlm.nih.gov/Traces/study/?</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1256/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1256/" class="post-title-link" itemprop="url">预测模型校准曲线 Calibration curve</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-05-21 19:34:20" itemprop="dateCreated datePublished" datetime="2021-05-21T19:34:20+08:00">2021-05-21</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p><img src="/wp/f4w/2021/2021-05-21-Calibration-Curve.png"></p>
<p>我们常用ROC曲线来衡量模型的预测能力，但很少关注模型的校准度calibration。</p>
<p>Calibration curve的<strong>横坐标</strong>是我们用模型预测的probability，比如我预测的是可能是肿瘤患者的概率，risk probability，<strong>纵坐标</strong>是真实的事件event的概率或者事件的proportion。</p>
<p>我们画图的时候，通常是点，需要进行平滑smooting，常用的是 <strong>loess</strong>方法。</p>
<p>如果我们预测的概率和真实的事件概率完全匹配，那么calibration plot则是一条斜率为1的直线（perfect calibration）。如果我们画出的先在perfect calibration上面，则表示underestimated，就是我们预测的风险或者概率低，真实情况却比较高；反之如果在perfect calibration在下，则是overestimated ，我们预测的风险比真实的风险要高。</p>
<p>根据吻合的程度，又分为poor calibration或者strong calibration，也可以做Hosmer–Lemeshow (HL) goodness-of-fit test。</p>
<p>参考：<a target="_blank" rel="noopener" href="https://bmcmedicine.biomedcentral.com/articles/10.1186/s12916-019-1466-7">https://bmcmedicine.biomedcentral.com/articles/10.1186/s12916-019-1466-7</a></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"># rms包可以画calibration plot</span><br><span class="line">library(rms)</span><br><span class="line">val.prob(predicted.probability, TRUE.Label)</span><br><span class="line"></span><br><span class="line">如果模型用rms建立的话，还可以用</span><br><span class="line">rms::calibrate的方法</span><br><span class="line"></span><br><span class="line">#我自己也参考别人写了一个函数,结果如下图，欢迎使用</span><br><span class="line">library(loonR)</span><br><span class="line">riskCalibrationPlot(TRUE.Label, predicted.probability)</span><br></pre></td></tr></table></figure>


<p><img src="/wp/f4w/2021/2021-05-21-Calibration-Curve-1.png"></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1235/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1235/" class="post-title-link" itemprop="url">特征选择</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-05-10 13:36:20" itemprop="dateCreated datePublished" datetime="2021-05-10T13:36:20+08:00">2021-05-10</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>特征选择是机器学习中的一个重要步骤，通过特征选择挑选出对预测起重要作用的变量，既可以减少数据的维度，也可以减少计算的消耗，同时也有助于我们对自己的数据的理解。有许多方法都可以应用到特征选择，比如大家常用的LASSO。我在用R做数据分析的时候，看到过这个帖子，进而了解了很多算法，所以对这个帖子进行了翻译，方便自己复习，也方便大家学习。</p>
<p>原文参考： <a target="_blank" rel="noopener" href="https://www.machinelearningplus.com/machine-learning/feature-selection/">https://www.machinelearningplus.com/machine-learning/feature-selection/</a></p>
<ol>
<li>Boruta</li>
<li>Variable Importance from Machine Learning Algorithms</li>
<li>Lasso Regression</li>
<li>Step wise Forward and Backward Selection</li>
<li>Relative Importance from Linear Regression</li>
<li>Recursive Feature Elimination (RFE)</li>
<li>Genetic Algorithm</li>
<li>Simulated Annealing</li>
<li>Information Value and Weights of Evidence</li>
<li>DALEX Package</li>
</ol>
<h2 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction"></a>Introduction</h2><p>真实的数据中，有些变量可能仅是噪声，并没有多少重要意义。</p>
<p>这类变量占用内存空间、消耗计算资源，我们最好去除这类变量，特别是在很大的数据集中。</p>
<p>有时候，我们有一个具有业务意义的变量，但不确定它是否确实有助于预测Y。还有一个事实：在一个机器学习算法中有用的特征（例如决策树） 可能其他算法中（例如回归模型）不被选用或者低估。</p>
<p>同时，有些变量单独预测Y的性能不好，但与其他预测变量/特征组合的情况下却非常显著。比如说有些变量与预测指标的相关性很低，但在其他变量参与的情况下，它可以帮助解释某些其他变量无法解释的模式/现象。</p>
<p>在这些情况下，很难决定包含还是去掉这些变量/特征。</p>
<p>这里讨论的策略可以解决这些问题，同时可以帮助理解对于一个模型而言，变量的重要性与否importance，以及对模型有多少贡献。</p>
<p>重要的一点是，我们最希望使用的变量是既具有业务意义同时也有重要性方面的指标。</p>
<p>我们这里导入Glaucoma 数据集，此数据集的目标是通过63个不同的生理测量指标来预测青光眼的与否。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Load Packages and prepare dataset</span></span><br><span class="line">library(TH.data)</span><br><span class="line">library(caret)</span><br><span class="line">data(<span class="string">&quot;GlaucomaM&quot;</span>, package = <span class="string">&quot;TH.data&quot;</span>)</span><br><span class="line">trainData &lt;- GlaucomaM</span><br><span class="line">head(trainData)</span><br></pre></td></tr></table></figure>



<p><img src="/wp/f4w/2021/2021-05-10-glaucoma-dataset.webp"></p>
<p>Glaucoma数据集</p>
<h2 id="1-Boruta"><a href="#1-Boruta" class="headerlink" title="1. Boruta"></a>1. Boruta</h2><p>Boruta 是一个基于随机森林对特征进行排名和筛选的算法。</p>
<p>Boruta 的优势是它可以明确的决定变量是否重要，并且帮助选择那些统计显著的变量。此外，可以通过调整p值和maxRuns来调整算法的严格性。</p>
<p>maxRuns是算法运行的次数，该参数值越大，会选择出更多的变量。默认是100.</p>
<p>在决定特征是否重要的过程中，一些特征可能会被标为Tentative。有时增加maxRuns或许可以解决特征的暂定行Tentativeness。</p>
<p>以TH.data的Glaucoma数据集为例子。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># install.packages(&#x27;Boruta&#x27;)</span></span><br><span class="line">library(Boruta)</span><br></pre></td></tr></table></figure>

<p>Boruta的方程用的公式同其他预测模型类似，响应变量response在左边，预测变量在右边。</p>
<p>如果预测变量处输入的是点，则意味着所有变量都会纳入评价中。</p>
<p><code>doTrace</code>参数控制打印到终端的数目。该值越高，打印的log信息越多。为了节省空间，这里设置0，你可以设置1和2试试。</p>
<p>结果输出在<code>boruta_output</code>.</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Perform Boruta search</span></span><br><span class="line">boruta_output &lt;- Boruta(Class ~ ., data=na.omit(trainData), doTrace=<span class="number">0</span>)  </span><br></pre></td></tr></table></figure>

<p>看看里面包含哪些内容</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">names</span>(boruta_output)</span><br><span class="line"><span class="number">1.</span> ‘finalDecision’</span><br><span class="line"><span class="number">2.</span> ‘ImpHistory’</span><br><span class="line"><span class="number">3.</span> ‘pValue’</span><br><span class="line"><span class="number">4.</span> ‘maxRuns’</span><br><span class="line"><span class="number">5.</span> ‘light’</span><br><span class="line"><span class="number">6.</span> ‘mcAdj’</span><br><span class="line"><span class="number">7.</span> ‘timeTaken’</span><br><span class="line"><span class="number">8.</span> ‘roughfixed’</span><br><span class="line"><span class="number">9.</span> ‘<span class="built_in">call</span>’</span><br><span class="line"><span class="number">10.</span> ‘impSource’</span><br></pre></td></tr></table></figure>

<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 得到显著的或者潜在的变量</span></span><br><span class="line">boruta_signif &lt;- getSelectedAttributes(boruta_output, withTentative = <span class="literal">TRUE</span>)</span><br><span class="line">print(boruta_signif)  </span><br><span class="line"> [<span class="number">1</span>] <span class="string">&quot;as&quot;</span>   <span class="string">&quot;ean&quot;</span>  <span class="string">&quot;abrg&quot;</span> <span class="string">&quot;abrs&quot;</span> <span class="string">&quot;abrn&quot;</span> <span class="string">&quot;abri&quot;</span> <span class="string">&quot;hic&quot;</span>  <span class="string">&quot;mhcg&quot;</span> <span class="string">&quot;mhcn&quot;</span> <span class="string">&quot;mhci&quot;</span></span><br><span class="line">[<span class="number">11</span>] <span class="string">&quot;phcg&quot;</span> <span class="string">&quot;phcn&quot;</span> <span class="string">&quot;phci&quot;</span> <span class="string">&quot;hvc&quot;</span>  <span class="string">&quot;vbss&quot;</span> <span class="string">&quot;vbsn&quot;</span> <span class="string">&quot;vbsi&quot;</span> <span class="string">&quot;vasg&quot;</span> <span class="string">&quot;vass&quot;</span> <span class="string">&quot;vasi&quot;</span></span><br><span class="line">[<span class="number">21</span>] <span class="string">&quot;vbrg&quot;</span> <span class="string">&quot;vbrs&quot;</span> <span class="string">&quot;vbrn&quot;</span> <span class="string">&quot;vbri&quot;</span> <span class="string">&quot;varg&quot;</span> <span class="string">&quot;vart&quot;</span> <span class="string">&quot;vars&quot;</span> <span class="string">&quot;varn&quot;</span> <span class="string">&quot;vari&quot;</span> <span class="string">&quot;mdn&quot;</span> </span><br><span class="line">[<span class="number">31</span>] <span class="string">&quot;tmg&quot;</span>  <span class="string">&quot;tmt&quot;</span>  <span class="string">&quot;tms&quot;</span>  <span class="string">&quot;tmn&quot;</span>  <span class="string">&quot;tmi&quot;</span>  <span class="string">&quot;rnf&quot;</span>  <span class="string">&quot;mdic&quot;</span> <span class="string">&quot;emd&quot;</span> </span><br></pre></td></tr></table></figure>

<p>如果不确定潜在的变量是否保留，可以对<code>boruta_output</code>进行<code>TentativeRoughFix</code> </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Do a tentative rough fix</span></span><br><span class="line">roughFixMod &lt;- TentativeRoughFix(boruta_output)</span><br><span class="line">boruta_signif &lt;- getSelectedAttributes(roughFixMod)</span><br><span class="line">print(boruta_signif)</span><br><span class="line"> [<span class="number">1</span>] <span class="string">&quot;abrg&quot;</span> <span class="string">&quot;abrs&quot;</span> <span class="string">&quot;abrn&quot;</span> <span class="string">&quot;abri&quot;</span> <span class="string">&quot;hic&quot;</span>  <span class="string">&quot;mhcg&quot;</span> <span class="string">&quot;mhcn&quot;</span> <span class="string">&quot;mhci&quot;</span> <span class="string">&quot;phcg&quot;</span> <span class="string">&quot;phcn&quot;</span></span><br><span class="line">[<span class="number">11</span>] <span class="string">&quot;phci&quot;</span> <span class="string">&quot;hvc&quot;</span>  <span class="string">&quot;vbsn&quot;</span> <span class="string">&quot;vbsi&quot;</span> <span class="string">&quot;vasg&quot;</span> <span class="string">&quot;vbrg&quot;</span> <span class="string">&quot;vbrs&quot;</span> <span class="string">&quot;vbrn&quot;</span> <span class="string">&quot;vbri&quot;</span> <span class="string">&quot;varg&quot;</span></span><br><span class="line">[<span class="number">21</span>] <span class="string">&quot;vart&quot;</span> <span class="string">&quot;vars&quot;</span> <span class="string">&quot;varn&quot;</span> <span class="string">&quot;vari&quot;</span> <span class="string">&quot;tmg&quot;</span>  <span class="string">&quot;tms&quot;</span>  <span class="string">&quot;tmi&quot;</span>  <span class="string">&quot;rnf&quot;</span>  <span class="string">&quot;mdic&quot;</span> <span class="string">&quot;emd&quot;</span> </span><br></pre></td></tr></table></figure>

<p>这样Boruta就代表我们来决定是否保留Tentative变量。查看这些变量的重要性分值。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Variable Importance Scores</span></span><br><span class="line">imps &lt;- attStats(roughFixMod)</span><br><span class="line">imps2 = imps[imps$decision != <span class="string">&#x27;Rejected&#x27;</span>, <span class="built_in">c</span>(<span class="string">&#x27;meanImp&#x27;</span>, <span class="string">&#x27;decision&#x27;</span>)]</span><br><span class="line">head(imps2[order(-imps2$meanImp), ])  <span class="comment"># descending sort</span></span><br></pre></td></tr></table></figure>

<table>
<thead>
<tr>
<th align="left"></th>
<th align="left">meanImp</th>
<th align="left">decision</th>
</tr>
</thead>
<tbody><tr>
<td align="left">varg</td>
<td align="left">10.279747</td>
<td align="left">Confirmed</td>
</tr>
<tr>
<td align="left">vari</td>
<td align="left">10.245936</td>
<td align="left">Confirmed</td>
</tr>
<tr>
<td align="left">tmi</td>
<td align="left">9.067300</td>
<td align="left">Confirmed</td>
</tr>
<tr>
<td align="left">vars</td>
<td align="left">8.690654</td>
<td align="left">Confirmed</td>
</tr>
<tr>
<td align="left">hic</td>
<td align="left">8.324252</td>
<td align="left">Confirmed</td>
</tr>
<tr>
<td align="left">varn</td>
<td align="left">7.327045</td>
<td align="left">Confirmed</td>
</tr>
</tbody></table>
<p>用画图的形式来展示变量的重要性</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Plot variable importance</span></span><br><span class="line">plot(boruta_output, cex.axis=<span class="number">.7</span>, las=<span class="number">2</span>, xlab=<span class="string">&quot;&quot;</span>, main=<span class="string">&quot;Variable Importance&quot;</span>)  </span><br></pre></td></tr></table></figure>

<p><img src="/wp/f4w/2021/2021-05-10-Variable_Importance_Boruta.webp">Variable Importance Boruta</p>
<p>这幅图展示了每个变量的重要性。</p>
<p>绿色表示的是筛选出的变量confirmed，红色则是需要提出的变量，蓝色不是真实的特征，待变的是<code>ShadowMax</code> 和 <code>ShadowMin</code>，用来决定变量重要性与否。</p>
<h2 id="2-Variable-Importance-from-Machine-Learning-Algorithms"><a href="#2-Variable-Importance-from-Machine-Learning-Algorithms" class="headerlink" title="2. Variable Importance from Machine Learning Algorithms"></a>2. Variable Importance from Machine Learning Algorithms</h2><p>另外一种特征选择的方法是将各种ML算法最常用的变量视为最重要的变量。</p>
<p>根据机器学习算法学习X与Y之间关系的方式，不同的机器学习算法选出不同的变量（大多数是重叠的），但赋予的权重并不相同。</p>
<p>例如，在基于树的算法（如“ rpart”）中被证明有用的变量在基于回归的模型中可能没那么有用。 因此，不同算法没必要用相同的变量。</p>
<p>那么对于一个特定的机器学算法，如何找到变量的重要性？</p>
<ol>
<li><p>利用<a target="_blank" rel="noopener" href="https://www.machinelearningplus.com/machine-learning/caret-package/">caret 包</a>中的train()一个特定的模型</p>
</li>
<li><p>使用 <code>varImp()</code> 来决定变量的重要性。</p>
</li>
</ol>
<p>可能尝试多种算法，以了解各种算法之间的重要变量。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 训练rpart模型，计算变量重要性</span></span><br><span class="line">library(caret)</span><br><span class="line">set.seed(<span class="number">100</span>)</span><br><span class="line">rPartMod &lt;- train(Class ~ ., data=trainData, method=<span class="string">&quot;rpart&quot;</span>)</span><br><span class="line">rpartImp &lt;- varImp(rPartMod)</span><br><span class="line">print(rpartImp)</span><br><span class="line">rpart variable importance</span><br><span class="line"></span><br><span class="line">  only <span class="number">20</span> most important variables shown (out of <span class="number">62</span>)</span><br><span class="line"></span><br><span class="line">     Overall</span><br><span class="line">varg  <span class="number">100.00</span></span><br><span class="line">vari   <span class="number">93.19</span></span><br><span class="line">vars   <span class="number">85.20</span></span><br><span class="line">varn   <span class="number">76.86</span></span><br><span class="line">tmi    <span class="number">72.31</span></span><br><span class="line">vbss    <span class="number">0.00</span></span><br><span class="line">eai     <span class="number">0.00</span></span><br><span class="line">tmg     <span class="number">0.00</span></span><br><span class="line">tmt     <span class="number">0.00</span></span><br><span class="line">vbst    <span class="number">0.00</span></span><br><span class="line">vasg    <span class="number">0.00</span></span><br><span class="line">at      <span class="number">0.00</span></span><br><span class="line">abrg    <span class="number">0.00</span></span><br><span class="line">vbsg    <span class="number">0.00</span></span><br><span class="line">eag     <span class="number">0.00</span></span><br><span class="line">phcs    <span class="number">0.00</span></span><br><span class="line">abrs    <span class="number">0.00</span></span><br><span class="line">mdic    <span class="number">0.00</span></span><br><span class="line">abrt    <span class="number">0.00</span></span><br><span class="line">ean     <span class="number">0.00</span></span><br></pre></td></tr></table></figure>



<p> rpart仅使用了63个功能中的5个，如果仔细观察，这5个变量位于boruta选择的前6个中。</p>
<p>让我们再做一件事：正则随机森林（Regularized Random Forest ，RRF）算法中的变量重要性。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Train an RRF model and compute variable importance.</span></span><br><span class="line">set.seed(<span class="number">100</span>)</span><br><span class="line">rrfMod &lt;- train(Class ~ ., data=trainData, method=<span class="string">&quot;RRF&quot;</span>)</span><br><span class="line">rrfImp &lt;- varImp(rrfMod, scale=<span class="built_in">F</span>)</span><br><span class="line">rrfImp</span><br><span class="line">RRF variable importance</span><br><span class="line"></span><br><span class="line">  only <span class="number">20</span> most important variables shown (out of <span class="number">62</span>)</span><br><span class="line"></span><br><span class="line">     Overall</span><br><span class="line">varg <span class="number">24.0013</span></span><br><span class="line">vari <span class="number">18.5349</span></span><br><span class="line">vars  <span class="number">6.0483</span></span><br><span class="line">tmi   <span class="number">3.8699</span></span><br><span class="line">hic   <span class="number">3.3926</span></span><br><span class="line">mhci  <span class="number">3.1856</span></span><br><span class="line">mhcg  <span class="number">3.0383</span></span><br><span class="line">mv    <span class="number">2.1570</span></span><br><span class="line">hvc   <span class="number">2.1357</span></span><br><span class="line">phci  <span class="number">1.8830</span></span><br><span class="line">vasg  <span class="number">1.8570</span></span><br><span class="line">tms   <span class="number">1.5705</span></span><br><span class="line">phcn  <span class="number">1.4475</span></span><br><span class="line">phct  <span class="number">1.4473</span></span><br><span class="line">vass  <span class="number">1.3097</span></span><br><span class="line">tmt   <span class="number">1.2485</span></span><br><span class="line">phcg  <span class="number">1.1992</span></span><br><span class="line">mdn   <span class="number">1.1737</span></span><br><span class="line">tmg   <span class="number">1.0988</span></span><br><span class="line">abrs  <span class="number">0.9537</span></span><br><span class="line">plot(rrfImp, top = <span class="number">20</span>, main=<span class="string">&#x27;Variable Importance&#x27;</span>)</span><br></pre></td></tr></table></figure>

<p><img src="/wp/f4w/2021/2021-05-10-RRF-Variable-Importance.webp">正则随机森林中的变量重要性。</p>
<p>最前面的重要变量也于Boruta选择的吻合</p>
<p>其它的在train()中可以用的算法也可以用varImp计算变量的重要性：包括</p>
<p><em>ada, AdaBag, AdaBoost.M1, adaboost, bagEarth, bagEarthGCV, bagFDA, bagFDAGCV, bartMachine, blasso, BstLm, bstSm, C5.0, C5.0Cost, C5.0Rules, C5.0Tree, cforest, chaid, ctree, ctree2, cubist, deepboost, earth, enet, evtree, extraTrees, fda, gamboost, gbm_h2o, gbm, gcvEarth, glmnet_h2o, glmnet, glmStepAIC, J48, JRip, lars, lars2, lasso, LMT, LogitBoost, M5, M5Rules, msaenet, nodeHarvest, OneR, ordinalNet, ORFlog, ORFpls, ORFridge, ORFsvm, pam, parRF, PART, penalized, PenalizedLDA, qrf, ranger, Rborist, relaxo, rf, rFerns, rfRules, rotationForest, rotationForestCp, rpart, rpart1SE, rpart2, rpartCost, rpartScore, rqlasso, rqnc, RRF, RRFglobal, sdwd, smda, sparseLDA, spikeslab, wsrf, xgbLinear, xgbTree.</em></p>
<h2 id="3-Lasso-Regression"><a href="#3-Lasso-Regression" class="headerlink" title="3. Lasso Regression"></a>3. Lasso Regression</h2><p>最小绝对收缩和选择算子（LASSO，Least Absolute Shrinkage and Selection Operator）回归是一种用L1范数惩罚的正则化方法。</p>
<p>从根本上来说，要增加权重（系数值）会带来成本。 它被称为L1正则化，增加的成本，与权重系数的绝对值成正比。</p>
<p>结果，在收缩系数的过程中，最终将某些不需要的特征的系数全部减小到零。 也就是说，它删除了不重要的变量。</p>
<p>LASSO回归也可以被视为有效的变量选择技术。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">library(glmnet)</span><br><span class="line">trainData &lt;- read.csv(<span class="string">&#x27;https://raw.githubusercontent.com/selva86/datasets/master/GlaucomaM.csv&#x27;</span>)</span><br><span class="line"></span><br><span class="line">x &lt;- as.matrix(trainData[,-<span class="number">63</span>]) <span class="comment"># all X vars</span></span><br><span class="line">y &lt;- <span class="built_in">as.double</span>(as.matrix(ifelse(trainData[, <span class="number">63</span>]==<span class="string">&#x27;normal&#x27;</span>, <span class="number">0</span>, <span class="number">1</span>))) <span class="comment"># Only Class</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># Fit the LASSO model (Lasso: Alpha = 1)</span></span><br><span class="line">set.seed(<span class="number">100</span>)</span><br><span class="line">cv.lasso &lt;- cv.glmnet(x, y, family=<span class="string">&#x27;binomial&#x27;</span>, alpha=<span class="number">1</span>, parallel=<span class="literal">TRUE</span>, standardize=<span class="literal">TRUE</span>, type.measure=<span class="string">&#x27;auc&#x27;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Results</span></span><br><span class="line">plot(cv.lasso)</span><br></pre></td></tr></table></figure>

<p><img src="/wp/f4w/2021/2021-05-10-Variable_Importance_LASSO.webp"></p>
<p>LASSO变量重要性</p>
<p>X轴是log之后的lambda，当是2的时候，lambda的真实值是100。</p>
<p>图的最上面显示了模型使用了多少个变量，相对应的红点Y值则是在这些变量使用的情况下，模型可以达到多少的AUC。</p>
<p>可以看到两条垂直虚线，左边的第一个指向具有最小均方误差的lambda。 右边的一个表示在1个标准偏差内偏差最大的变量的数量。</p>
<p>最优的lambda值存储在cv.lasso $ lambda.min中。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># plot(cv.lasso$glmnet.fit, xvar=&quot;lambda&quot;, label=TRUE)</span></span><br><span class="line">cat(<span class="string">&#x27;Min Lambda: &#x27;</span>, cv.lasso$lambda.min, <span class="string">&#x27;\n 1Sd Lambda: &#x27;</span>, cv.lasso$lambda.1se)</span><br><span class="line">df_coef &lt;- <span class="built_in">round</span>(as.matrix(coef(cv.lasso, s=cv.lasso$lambda.min)), <span class="number">2</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># See all contributing variables</span></span><br><span class="line">df_coef[df_coef[, <span class="number">1</span>] != <span class="number">0</span>, ]</span><br><span class="line">Min Lambda:  <span class="number">0.01166507</span> </span><br><span class="line"> <span class="number">1</span>Sd Lambda:  <span class="number">0.2513163</span></span><br><span class="line"></span><br><span class="line">Min Lambda:  <span class="number">0.01166507</span> </span><br><span class="line"><span class="number">1</span>Sd Lambda:  <span class="number">0.2513163</span></span><br><span class="line"></span><br><span class="line">(Intercept) <span class="number">3.65</span></span><br><span class="line">at         -<span class="number">0.17</span></span><br><span class="line">as         -<span class="number">2.05</span></span><br><span class="line">eat        -<span class="number">0.53</span></span><br><span class="line">mhci        <span class="number">6.22</span></span><br><span class="line">phcs       -<span class="number">0.83</span></span><br><span class="line">phci        <span class="number">6.03</span></span><br><span class="line">hvc        -<span class="number">4.15</span></span><br><span class="line">vass       -<span class="number">23.72</span></span><br><span class="line">vbrn       -<span class="number">0.26</span></span><br><span class="line">vars       -<span class="number">25.86</span></span><br><span class="line">mdt        -<span class="number">2.34</span></span><br><span class="line">mds         <span class="number">0.5</span></span><br><span class="line">mdn         <span class="number">0.83</span></span><br><span class="line">mdi         <span class="number">0.3</span></span><br><span class="line">tmg         <span class="number">0.01</span></span><br><span class="line">tms         <span class="number">3.02</span></span><br><span class="line">tmi         <span class="number">2.65</span></span><br><span class="line">mv          <span class="number">4.94</span></span><br></pre></td></tr></table></figure>

<p>上面的输出显示了LASSO认为重要的变量。绝对值越高表示该变量越重要。</p>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/1235/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1232/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1232/" class="post-title-link" itemprop="url">将ggplot导出成ppt的R包</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-05-09 15:16:30" itemprop="dateCreated datePublished" datetime="2021-05-09T15:16:30+08:00">2021-05-09</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <h2 id="1，export的graph2ppt函数"><a href="#1，export的graph2ppt函数" class="headerlink" title="1，export的graph2ppt函数"></a>1，export的graph2ppt函数</h2><p><a target="_blank" rel="noopener" href="https://github.com/tomwenseleers/export">https://github.com/tomwenseleers/export</a></p>
<p>export虽然从CRAN下架了，但依然可以通过github的库来安装，devtools::install_github(“tomwenseleers/export”)</p>
<h2 id="2，eoffice的topptx函数"><a href="#2，eoffice的topptx函数" class="headerlink" title="2，eoffice的topptx函数"></a>2，eoffice的topptx函数</h2><p><a target="_blank" rel="noopener" href="https://github.com/guokai8/eoffice">https://github.com/guokai8/eoffice</a></p>
<h2 id="3-officer"><a href="#3-officer" class="headerlink" title="3, officer"></a>3, officer</h2><p><a target="_blank" rel="noopener" href="https://github.com/davidgohel/officer">https://github.com/davidgohel/officer</a></p>
<p>用法可以参考：<a target="_blank" rel="noopener" href="https://www.brodrigues.co/blog/2018-10-05-ggplot2_purrr_officer/">https://www.brodrigues.co/blog/2018-10-05-ggplot2_purrr_officer/</a></p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1219/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1219/" class="post-title-link" itemprop="url">Continuous Long Read, subreads and scraps in Pacbio Raw data</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-04-01 20:25:23" itemprop="dateCreated datePublished" datetime="2021-04-01T20:25:23+08:00">2021-04-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>Pacbio的工具更新实在是太快了，<a target="_blank" rel="noopener" href="https://github.com/PacificBiosciences/pbbioconda">https://github.com/PacificBiosciences/pbbioconda</a></p>
<p>原来Isoseq的数据处理之后还叫high quality reads，现在叫HiFi reads，而且Pacbio的工具包有这么个倾向，把Isoseq分析过程中用的工具都囊括在内，比如后续的比对和collapse的工作。在tofu停了之后，cDNA_Cupcake也顶上来了。下面是几个名词解释，CLR, CCS, subreads, scrapes等，在处理原始的raw data时会碰到。</p>
<p>The contiguous sequence generated by the polymerase during sequencing is referred to as a “polymerase read” or a <strong>Continuous Long Read (CLR)</strong>.</p>
<p>This CLR read may include sequence from adapters and multiple copies of inserts, because it traverses the circular template many times. The CLRs are processed to remove adapter sequences and to retain only the insert sequence, called “<strong>subreads</strong>”.</p>
<p>All other sequences sequenced from the CLR are called “<strong>scraps</strong>”.</p>
<p>Multiple copies of subreads generated from the single SMRTBell can then be collapsed to a single, high-quality sequence, called the “read of insert” ROI or Circular Consensus Sequence (<strong>CCS</strong>).</p>
<p>The scraps file includes all the sequence not in the HQ (high quality) region, and the adapter sequences. The old bax format saved everything with indexes into the usable data. The subreads.bam contains all the usable data.</p>
<p>关于scrapes和subreads的bam文件的区别，只用subreads即可<br>The scraps file includes all the sequence not in the HQ (high quality) region, and the adapter sequences. The old bax format saved everything with indexes into the usable data. The subreads.bam contains all the usable data.</p>
<p>.subreads.bam A file contain usable reads without adapter;The Sequel System outputs one subreads.bam file per collection/cell, which contains unaligned base calls from high-quality regions.</p>
<p>.scraps.bam A file contain sequence data outside of the High Quality region, rejected subreads,excised adapter and possible barcode sequences, as well as spike-in control sequences. (The basecaller marks regions of single molecule sequence activity as high-quality.)</p>
<p>Ref:<br><a target="_blank" rel="noopener" href="http://www.zxzyl.com/archives/1044">http://www.zxzyl.com/archives/1044</a></p>
<p><a target="_blank" rel="noopener" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03751-8">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-03751-8</a></p>
<p><a target="_blank" rel="noopener" href="http://seqanswers.com/forums/showpost.php?p=202171&amp;postcount=9">http://seqanswers.com/forums/showpost.php?p=202171&amp;postcount=9</a></p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1209/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1209/" class="post-title-link" itemprop="url">谈一谈在变异解读过程中用到的几个不太熟悉的预测指标</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-01-20 10:25:23" itemprop="dateCreated datePublished" datetime="2021-01-20T10:25:23+08:00">2021-01-20</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>帅旸谈一谈在变异解读过程中用到的几个不太熟悉的预测指标：</p>
<h1 id="z-score"><a href="#z-score" class="headerlink" title="z score"></a>z score</h1><p>z score：这个指标指的是某个基因对missense的耐受程度，具体是指该基因所期望的missense数比上观察</p>
<p>到的missense数，如果z score&gt;3.09，则认为该基因对missense不耐受，根据公式我们可以看出如果比值越大，则基因对missense越不耐受。利用z score可以在我们使用ACMG指南PP2的时候使用。</p>
<h1 id="REVEL-score"><a href="#REVEL-score" class="headerlink" title="REVEL score"></a>REVEL score</h1><p>REVEL score：ClinGen SVI建议使用REVEL用来预测missense致病性。与其他常用missense致病性预测软件不同，REVEL整合了包括SIFT、PolyPhen、GERP++在内的13个软件的预测结果，对罕见变异的预测结果更加出色。当REVEL score&gt;0.75，&lt;0.15时分别使用ACMG指南PP3和BP4。</p>
<h1 id="GERP"><a href="#GERP" class="headerlink" title="GERP++"></a>GERP++</h1><p>GERP++ rejected substitutions” (RS) score：GERP++从基因进化速率角度预测位点保守性，具体是指该基因位点所期望的碱基替换次数减去观察到的碱基替换次数，可见分数越大，该位点保守性较强，当GERP++ RS score&gt;6.8时，认为该位点保守。当分析一个不影响剪切的同义突变时，如果RS score&lt;6.8，则可以使用ACMG指南BP7。</p>
<h1 id="dbscSNV-score"><a href="#dbscSNV-score" class="headerlink" title="dbscSNV score"></a>dbscSNV score</h1><p>dbscSNV score：dbscSNV含有两个不同的算法，用来预测变异是否影响截切，一个是基于adaptive boostin，一个是基于Random Forest。当两种算法得分均小于0.6时，则认为不影响剪切。</p>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p>ClinGen及Zhang, J., Yao, Y., He, H., &amp; Shen, J. (2020). Clinical Interpretation of Sequence Variants. Current Protocols in Human Genetics, 106(1), e98.</p>
<p>其他参数如pLI及 Haploinsufficiency score我们之前已经介绍过，请点击下文链接</p>
<p><a target="_blank" rel="noopener" href="http://www.zxzyl.com/archives/940">http://www.zxzyl.com/archives/940</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: 帅旸<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1199/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1199/" class="post-title-link" itemprop="url">推荐两个Rstudio Addins</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-01-09 16:48:23" itemprop="dateCreated datePublished" datetime="2021-01-09T16:48:23+08:00">2021-01-09</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <h1 id="datapasta"><a href="#datapasta" class="headerlink" title="datapasta"></a>datapasta</h1><p><a target="_blank" rel="noopener" href="https://github.com/MilesMcBain/datapasta/">https://github.com/MilesMcBain/datapasta/</a></p>
<p>还在手工的把excel的数据写成导到R里吗。不管横着还是竖着复制数据，datapasta可以自动、快速的把复制数据转成tibbles, data.frames, 或者 vectors格式。</p>
<p>更详细的参考<a target="_blank" rel="noopener" href="https://cran.r-project.org/web/packages/datapasta/vignettes/how-to-datapasta.html">https://cran.r-project.org/web/packages/datapasta/vignettes/how-to-datapasta.html</a></p>
<p><img src="https://raw.githubusercontent.com/lorenzwalthert/some_raw_data/master/styler_0.1.gif" alt="https://raw.githubusercontent.com/lorenzwalthert/some_raw_data/master/styler_0.1.gif"></p>
<h1 id="styler"><a href="#styler" class="headerlink" title="styler"></a>styler</h1><p><a target="_blank" rel="noopener" href="https://github.com/r-lib/styler">https://github.com/r-lib/styler</a></p>
<p>还嫌自己写出的代码不够美吗，styler 可以把代码格式化成tidyverse规则的风格。</p>
<p>The goal of styler is to provide non-invasive pretty-printing of R source code while adhering to the tidyverse formatting rules. styler can be customized to format code according to other style guides too.</p>
<p><img src="https://raw.githubusercontent.com/lorenzwalthert/some_raw_data/master/styler_0.1.gif" alt="https://raw.githubusercontent.com/lorenzwalthert/some_raw_data/master/styler_0.1.gif"></p>
<p>真香</p>
<h1 id="links"><a href="#links" class="headerlink" title="links"></a>links</h1><p><a target="_blank" rel="noopener" href="https://github.com/MilesMcBain/datapasta/">https://github.com/MilesMcBain/datapasta/</a></p>
<p><a target="_blank" rel="noopener" href="https://github.com/r-lib/styler">https://github.com/r-lib/styler</a></p>
<p><a target="_blank" rel="noopener" href="https://github.com/daattali/addinslist">https://github.com/daattali/addinslist</a></p>
<p><a target="_blank" rel="noopener" href="https://www.zhihu.com/question/398418315">https://www.zhihu.com/question/398418315</a></p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1195/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1195/" class="post-title-link" itemprop="url">Map NM ID to Gene Symbol</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-01-08 16:39:23" itemprop="dateCreated datePublished" datetime="2021-01-08T16:39:23+08:00">2021-01-08</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>新年快乐，21年的第一篇文章。</p>
<p>以前写过映射ENSEMBL ID 和 NCBI ID， <a target="_blank" rel="noopener" href="http://www.zxzyl.com/archives/736">http://www.zxzyl.com/archives/736</a>。</p>
<p>日常分析中，我们也会经常遇到其他的ID mapping的工作，这种工作不是基因ID转基因ID，而是转录本的ID转基因ID。</p>
<p>如果用的是refGene的注释，最简单了，直接用下面的命令即可</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">mysql --user=genome -N --host=genome-mysql.cse.ucsc.edu -A -D hg38 -e &quot;select name,name2 from refGene&quot;</span><br></pre></td></tr></table></figure>

<p>不过我也经常通过解析gtf文件获得，因为gtf有转本的ID，也有基因的symbol或者ID，只要有gtf文件就可以提取。本着不造轮子的精神，我利用的是现成的R包</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">library(plyranges)</span><br><span class="line">gr &lt;- read_gff(&quot;/path/to/gtf/or/gff&quot;) %&gt;% select(transcript_id, gene_id, gene_name)</span><br><span class="line">gr &lt;- unique(data.frame(gr))</span><br></pre></td></tr></table></figure>

<p>我也在自己的包里面写了一个函数得到ensembl，refseq，hgnc，gene symbol的对应关系，biomaRt比较慢，可以把结果保存成文件</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">devtools::install_github(<span class="string">&quot;ProfessionalFarmer/loonR&quot;</span>)</span><br><span class="line"><span class="comment"># 需要安装biomaRt和dplyr</span></span><br><span class="line">mapping.table &lt;- loonR::get_full_mapping_table()</span><br><span class="line"><span class="comment"># 保存mapping.table</span></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1186/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1186/" class="post-title-link" itemprop="url">Fixation index (FST)</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2020-11-02 21:03:23" itemprop="dateCreated datePublished" datetime="2020-11-02T21:03:23+08:00">2020-11-02</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <h1 id="Fixation-index-FST"><a href="#Fixation-index-FST" class="headerlink" title="Fixation index (FST)"></a>Fixation index (FST)</h1><p>整理来源 <a target="_blank" rel="noopener" href="http://www.uwyo.edu/dbmcd/popecol/maylects/fst.html%EF%BC%8C%E6%8A%8A%E8%BF%99%E4%B8%AA%E8%AF%BE%E7%A8%8B%E7%9A%84%E6%AD%A5%E9%AA%A4%E7%94%A8%E8%A1%A8%E6%A0%BC%E6%9D%A5%E8%A1%A8%E7%A4%BA">http://www.uwyo.edu/dbmcd/popecol/maylects/fst.html，把这个课程的步骤用表格来表示</a></p>
<table>
<thead>
<tr>
<th></th>
<th></th>
<th></th>
<th>Subpopulation 1</th>
<th>Subpopulation 2</th>
<th>Subpopulation 3</th>
<th>Total</th>
</tr>
</thead>
<tbody><tr>
<td></td>
<td>Genotype</td>
<td>AA</td>
<td>125</td>
<td>50</td>
<td>100</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>Aa</td>
<td>250</td>
<td>30</td>
<td>500</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>aa</td>
<td>125</td>
<td>20</td>
<td>400</td>
<td></td>
</tr>
<tr>
<td></td>
<td>Number of individual</td>
<td></td>
<td>500</td>
<td>100</td>
<td>1000</td>
<td>1600</td>
</tr>
<tr>
<td></td>
<td>Number of alleles</td>
<td></td>
<td>1000</td>
<td>200</td>
<td>2000</td>
<td>3200</td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 1.</td>
<td></td>
<td></td>
<td>Calculate the gene   (allele) frequencies</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>Observed allele frequency</td>
<td>A (p)</td>
<td>(125*2+250)/1000=0.5</td>
<td>(50*2+30)/200=0.65</td>
<td>(2*100+500)/2000=0.35</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>a (q)</td>
<td>0.5</td>
<td>0.35</td>
<td>0.65</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 2.</td>
<td></td>
<td></td>
<td>Calculate the expected   genotypic counts under Hardy-Weinberg Equilibrium, and then calculate the   excess or deficiency of homozygotes in each subpopulation.      Summary of homozygote deficiency or excess relative to HWE:          Pop. 1.  Observed = Expected: perfect fit          Pop. 2.  Excess of 15.5 homozygotes: some inbreeding          Pop. 3.  Deficiency of 45 homozygotes: outbred or   experiencing a Wahlund effect (isolate breaking).</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>Expected allele frequency</td>
<td>AA</td>
<td>500*0.5^2 = 125 (= observed)</td>
<td>100*0.65^2 = 42.25 (observed has excess of 7.75)</td>
<td>1,000*0.35^2 = 122.5   (observed has deficiency of 22.5)</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>Aa</td>
<td>500<em>2</em>0.5*0.5 = 250  (=   observed)</td>
<td>100<em>2</em>0.65*0.35 = 45.5 (observed has deficit of 15.5)</td>
<td>1,000<em>2</em>0.65*0.35 = 455   (observed has excess of 45)</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>aa</td>
<td>500*0.5^2 = 125 (= observed)</td>
<td>100*0.35^2 = 12.25 (observed has excess of 7.75)</td>
<td>1,000*0.35^2 = 422.5   (observed has deficiency of 22.5)</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 3.</td>
<td></td>
<td></td>
<td>Calculate the local   observed heterozygosities of each subpopulation (we will call them Hobs s,   where the s subscript refers to the sth of n populations – 3 in this   example).</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>Local observed   heterozygosities</td>
<td></td>
<td>250/500 = 0.5 (Hobs 1)</td>
<td>30/100 = 0.3 (Hobs 2)</td>
<td>500/1000 = 0.5(Hobs 3)</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 4.</td>
<td></td>
<td></td>
<td>Calculate the local   expected heterozygosity, or gene diversity, of each subpopulation       Hexp = 2pq</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>Local expected   heterozygosity</td>
<td></td>
<td>2<em>0.5</em>0.5=0.5 (Hexp 1)</td>
<td>2<em>0.65</em>3.5=0.455 (Hexp 2)</td>
<td>2<em>0.35</em>0.65=0.455 (Hexp 2)</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 5.</td>
<td></td>
<td></td>
<td>Calculate the local   inbreeding coefficient of each subpopulation      F = (Hexps -Hobs)/Hexp      [positive F means fewer heterozygotes than expected indicates   inbreeding]       [negative F means more heterozygotes   than expected means excess outbreeding]</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td>F1=(0.5—0.5)/0.5=0</td>
<td>F2=(0.455—0.3)/0.455=0.341</td>
<td>F3=(0.455—0.5)/0.455=-0.099</td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 6. and 7.</td>
<td></td>
<td></td>
<td>Calculate p-bar   (p-bar, the frequency of allele A) over the total population.      Calculate q-bar (q-bar, the frequency of allele a) over the total   population.      Check: p-bar + q-bar =  1.0</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td>the frequency of allele over the total population</td>
<td>p-bar</td>
<td>(0.5<em>1000+0.65</em>200+0.35*2000)/3200=0.4156</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>q-bar</td>
<td>(0.5<em>1000+0.35</em>200+0.65*2000)/3200=0.5844</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 8.</td>
<td></td>
<td></td>
<td>Calculate the global   heterozygosity indices (over Individuals, Subpopulations and Total   population)      HI based on observed heterozygosities in individuals in subpopulations         HS based on expected heterozygosities in subpopulations      HT based on expected heterozygosities for overall total population</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>HI (observed)</td>
<td>(0.5<em>500+0.3</em>100+0.5*1000)/1600=0.4875</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>HS (expected)</td>
<td>(0.5<em>500+0.455</em>100+0.455*1000)/1600=0.4691</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>HT (in overall total population)</td>
<td>2*p-bar *q-bar     = 2 * 0.4156 * 0.5844  = 0.4858</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 9.</td>
<td></td>
<td></td>
<td>Calculate the global   F-statistics      Compare and contrast the global FISbelow with the “local inbreeding   coefficient” Fs of Step 5.      Here we are using a weighted average of the individual heterozygosities   over all the subpopulations.      Both FIS and  Fs are, however, based   on the observed heterozygosities,              whereas FST and FIT are based   on expected heterozygosities.</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>FIS</td>
<td>(Hs-Hi)/Hs=(0.4691-0.4875)/0.4691=-0.0393</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>FST</td>
<td>(Ht-Hs)/Ht=(0.4858-0.4691)/0.4858=-0.0344</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td></td>
<td></td>
<td>FIT</td>
<td>(Ht-Hi)/Ht=(0.4858-0.4875)/0.4858=-0.0036</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td>Step 10</td>
<td></td>
<td>conclusions</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody></table>
<p>Finally, draw some   conclusions about the genetic structure of the population and its   subpopulations.      1)  One of the possible HWE   conclusions we could make:      Pop. 1 is consistent with HWE (results of Step </p>
<ol start="2">
<li>Two of the possible “local   inbreeding” conclusions we could make from Step 5:      </li>
</ol>
<p>Pop. 2 is inbred (results of Step 5), and    </p>
<p> Pop. 3 may have disassortative mating or be experiencing a Wahlund effect   (more heterozygotes than expected).      </p>
<ol start="3">
<li> Conclusion concerning overall   degree of genetic differentiation (FST)      </li>
</ol>
<p>Subdivision of populations, possibly due to genetic drift, accounts for approx. 3.4% of the total genetic variation (result of Eqn FST.8 FST   calculation in Step 9),      </p>
<ol start="4">
<li> No excess or deficiency of   heterozygotes over the total population (FIT    is nearly zero).</li>
</ol>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  


  
  <nav class="pagination">
    <a class="extend prev" rel="prev" href="/"><i class="fa fa-angle-left" aria-label="Previous page"></i></a><a class="page-number" href="/">1</a><span class="page-number current">2</span><a class="page-number" href="/page/3/">3</a><span class="space">&hellip;</span><a class="page-number" href="/page/17/">17</a><a class="extend next" rel="next" href="/page/3/"><i class="fa fa-angle-right" aria-label="Next page"></i></a>
  </nav>



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Jason</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">163</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">167</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>


<!--google_adsense-->

  <div class="links-of-blogroll-title">
    <i class="fa fa-google"></i> AdSense
    <ins class="adsbygoogle"
        style="display:block"
        data-ad-client="ca-pub-9370593480269435"
        data-ad-slot="8824957305"
        data-ad-format="auto"
        data-full-width-responsive="true"></ins>
    <script>
      (adsbygoogle = window.adsbygoogle || []).push({});
    </script>
  </div>

<!--/google_adsense-->



    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Jason</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

<script data-ad-client="ca-pub-9370593480269435" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<!-- PGAD -->
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-9370593480269435"
     data-ad-slot="8824957305"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
