<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"md.zxzyl.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>


  <meta name="description" content="不论是目标区域测序还是外显子组测序还是全基因组测序之后，我们会关注目标区域在特定深度下的覆盖度。比如20X的比例，50X的比例。如果只计算特定深度的覆盖度，我们了解不到其他深度下的覆盖度情况。如果每个都列出来，又不直观，这个时候用图片来表示就非常直观了。 获取统计数据我们需要知道在每个深度下碱基的比例，这个时候强大的bedtools就出场了。 bedtools coverage -sorted -">
<meta property="og:type" content="article">
<meta property="og:title" content="PLOT | 目标区域的测序覆盖度作图">
<meta property="og:url" content="https://md.zxzyl.com/archives/810/index.html">
<meta property="og:site_name" content="Zhongxu&#39;s blog">
<meta property="og:description" content="不论是目标区域测序还是外显子组测序还是全基因组测序之后，我们会关注目标区域在特定深度下的覆盖度。比如20X的比例，50X的比例。如果只计算特定深度的覆盖度，我们了解不到其他深度下的覆盖度情况。如果每个都列出来，又不直观，这个时候用图片来表示就非常直观了。 获取统计数据我们需要知道在每个深度下碱基的比例，这个时候强大的bedtools就出场了。 bedtools coverage -sorted -">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://md.zxzyl.com/wp/f4w/2020/2018-02-01-target-region-coverage.png">
<meta property="article:published_time" content="2018-02-01T07:56:04.000Z">
<meta property="article:modified_time" content="2018-02-01T07:56:04.000Z">
<meta property="article:author" content="Jason">
<meta property="article:tag" content="Coverage">
<meta property="article:tag" content="PLOT">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://md.zxzyl.com/wp/f4w/2020/2018-02-01-target-region-coverage.png">

<link rel="canonical" href="https://md.zxzyl.com/archives/810/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>PLOT | 目标区域的测序覆盖度作图 | Zhongxu's blog</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a3b5d3f466edb3e3f15e8238d8f04be8";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Zhongxu's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/810/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          PLOT | 目标区域的测序覆盖度作图
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2018-02-01 15:56:04" itemprop="dateCreated datePublished" datetime="2018-02-01T15:56:04+08:00">2018-02-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>不论是目标区域测序还是外显子组测序还是全基因组测序之后，我们会关注目标区域在特定深度下的覆盖度。比如20X的比例，50X的比例。如果只计算特定深度的覆盖度，我们了解不到其他深度下的覆盖度情况。如果每个都列出来，又不直观，这个时候用图片来表示就非常直观了。</p>
<h1 id="获取统计数据"><a href="#获取统计数据" class="headerlink" title="获取统计数据"></a>获取统计数据</h1><p>我们需要知道在每个深度下碱基的比例，这个时候强大的bedtools就出场了。</p>
<pre>bedtools coverage -sorted -hist -g genome.file  -b  samp.bam -a target_regions.bed ' grep ^all > samp.hist.all.txt</pre>

<p>-hist是为了获取目标区域的总结信息，以all开头，输出每个每个深度下碱基的比例，所以后续用grep ^all来过滤。</p>
<p>-sorted可以节省内存，但需要通过-g跟genome file来指定染色体的顺序。</p>
<p>genome file可以通过awk -v OFS=’\t’ {‘print $1,$2’} genome.fa.fai &gt; genome.file 得到。</p>
<p>假设我们得到了smp1.hist.all.txt、smp2.hist.all.txt、smp3.hist.all.txt、smp4.hist.all.txt、smp5.hist.all.txt、smp6.hist.all.txt、smp7.hist.all.txt、</p>
<h1 id="作图"><a href="#作图" class="headerlink" title="作图"></a>作图</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Get a list of the bedtools output files you&#x27;d like to read in</span></span><br><span class="line"></span><br><span class="line">print(labs &lt;- paste(<span class="string">&quot;&quot;</span>, gsub(<span class="string">&quot;.hist.all.txt&quot;</span>, <span class="string">&quot;&quot;</span>, files, perl=<span class="literal">TRUE</span>), sep=<span class="string">&quot;&quot;</span>))</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># Create lists to hold coverage and cumulative coverage for each alignment,</span></span><br><span class="line"><span class="comment"># and read the data into these lists.</span></span><br><span class="line">cov &lt;- <span class="built_in">list</span>()</span><br><span class="line">cov_cumul &lt;- <span class="built_in">list</span>()</span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:<span class="built_in">length</span>(files)) &#123;</span><br><span class="line">    cov[[i]] &lt;- read.table(files[i])</span><br><span class="line">    <span class="comment"># The value should be 1 at 0X.</span></span><br><span class="line">    cov_cumul[[i]] &lt;- <span class="number">1</span>-<span class="built_in">cumsum</span>(<span class="built_in">c</span>(<span class="number">0</span>,cov[[i]][,<span class="number">5</span>]))</span><br><span class="line">&#125;</span><br><span class="line"></span><br><span class="line">library(RColorBrewer)</span><br><span class="line">cols &lt;- brewer.pal(<span class="built_in">length</span>(cov), <span class="string">&quot;Dark2&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Save the graph to a file</span></span><br><span class="line">png(<span class="string">&quot;target-coverage-plots.png&quot;</span>, h=<span class="number">1000</span>, w=<span class="number">1000</span>, pointsize=<span class="number">20</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Create plot area, but do not plot anything. Add gridlines and axis labels.</span></span><br><span class="line">plot(cov[[<span class="number">1</span>]][<span class="number">2</span>:<span class="number">401</span>, <span class="number">2</span>], cov_cumul[[<span class="number">1</span>]][<span class="number">1</span>:<span class="number">400</span>], type=<span class="string">&#x27;n&#x27;</span>, xlab=<span class="string">&quot;Depth&quot;</span>, ylab=<span class="string">&quot;Fraction of capture target bases \u2265 depth&quot;</span>, ylim=<span class="built_in">c</span>(<span class="number">0</span>,<span class="number">1.0</span>), main=<span class="string">&quot;Target Region Coverage&quot;</span>)</span><br><span class="line">abline(v = <span class="number">20</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(v = <span class="number">50</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(v = <span class="number">80</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(v = <span class="number">100</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(h = <span class="number">0.50</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">abline(h = <span class="number">0.90</span>, col = <span class="string">&quot;gray60&quot;</span>)</span><br><span class="line">axis(<span class="number">1</span>, at=<span class="built_in">c</span>(<span class="number">20</span>,<span class="number">50</span>,<span class="number">80</span>), labels=<span class="built_in">c</span>(<span class="number">20</span>,<span class="number">50</span>,<span class="number">80</span>))</span><br><span class="line">axis(<span class="number">2</span>, at=<span class="built_in">c</span>(<span class="number">0.90</span>), labels=<span class="built_in">c</span>(<span class="number">0.90</span>))</span><br><span class="line">axis(<span class="number">2</span>, at=<span class="built_in">c</span>(<span class="number">0.50</span>), labels=<span class="built_in">c</span>(<span class="number">0.50</span>))</span><br><span class="line"></span><br><span class="line"><span class="comment"># Actually plot the data for each of the alignments (stored in the lists).</span></span><br><span class="line"><span class="keyword">for</span> (i <span class="keyword">in</span> <span class="number">1</span>:<span class="built_in">length</span>(cov)) points(cov[[i]][<span class="number">2</span>:<span class="number">401</span>, <span class="number">2</span>], cov_cumul[[i]][<span class="number">1</span>:<span class="number">400</span>], type=<span class="string">&#x27;l&#x27;</span>, lwd=<span class="number">3</span>, col=cols[i])</span><br><span class="line"></span><br><span class="line"><span class="comment"># Add a legend using the nice sample labeles rather than the full filenames.</span></span><br><span class="line">legend(<span class="string">&quot;topright&quot;</span>, legend=labs, col=cols, lty=<span class="number">1</span>, lwd=<span class="number">4</span>)</span><br><span class="line"></span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>

<p><img src="/wp/f4w/2020/2018-02-01-target-region-coverage.png"></p>
<p>图片也可以用ggplot2来做。横坐标是DP，纵坐标是大于等于&gt;=该DP的碱基比例。该图还有一个好处，是可以多个样本同时放在一张图上，便于进行样本间比较。</p>
<p>可以看出smp2的覆盖度最好，随着横左边的增加，下降最缓慢。smp1和smp3最不好，在20X以后，覆盖度显著下降。如果样本smp来自不同的测序实验，则smp2所用的方法是最有的。</p>
<p>最后还是那句话，图片表达的信息量多而且直观。让我们用图片说话。</p>
<p>主要CLONE参考：</p>
<p><a target="_blank" rel="noopener" href="http://www.gettinggeneticsdone.com/2014/03/visualize-coverage-exome-targeted-ngs-bedtools.html">http://www.gettinggeneticsdone.com/2014/03/visualize-coverage-exome-targeted-ngs-bedtools.html</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/Coverage/" rel="tag"># Coverage</a>
              <a href="/tags/PLOT/" rel="tag"># PLOT</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/archives/805/" rel="prev" title="Illumina下机FASTQ文件命名规则">
      <i class="fa fa-chevron-left"></i> Illumina下机FASTQ文件命名规则
    </a></div>
      <div class="post-nav-item">
    <a href="/archives/815/" rel="next" title="BCL文件与BCL2FAFSTQ程序简介">
      BCL文件与BCL2FAFSTQ程序简介 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#%E8%8E%B7%E5%8F%96%E7%BB%9F%E8%AE%A1%E6%95%B0%E6%8D%AE"><span class="nav-number">1.</span> <span class="nav-text">获取统计数据</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E4%BD%9C%E5%9B%BE"><span class="nav-number">2.</span> <span class="nav-text">作图</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Jason</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">163</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">167</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>


<!--google_adsense-->

  <div class="links-of-blogroll-title">
    <i class="fa fa-google"></i> AdSense
    <ins class="adsbygoogle"
        style="display:block"
        data-ad-client="ca-pub-9370593480269435"
        data-ad-slot="8824957305"
        data-ad-format="auto"
        data-full-width-responsive="true"></ins>
    <script>
      (adsbygoogle = window.adsbygoogle || []).push({});
    </script>
  </div>

<!--/google_adsense-->



    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Jason</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

<script data-ad-client="ca-pub-9370593480269435" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<!-- PGAD -->
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-9370593480269435"
     data-ad-slot="8824957305"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
