<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"md.zxzyl.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>


  <meta property="og:type" content="website">
<meta property="og:title" content="Zhongxu&#39;s blog">
<meta property="og:url" content="https://md.zxzyl.com/index.html">
<meta property="og:site_name" content="Zhongxu&#39;s blog">
<meta property="og:locale" content="en_US">
<meta property="article:author" content="Jason">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://md.zxzyl.com/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : true,
    isPost : false,
    lang   : 'en'
  };
</script>

  <title>Zhongxu's blog</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a3b5d3f466edb3e3f15e8238d8f04be8";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Zhongxu's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content index posts-expand">
            
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1450/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1450/" class="post-title-link" itemprop="url">临床模型的决策曲线分析解读</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-04-30 21:02:20" itemprop="dateCreated datePublished" datetime="2022-04-30T21:02:20+08:00">2022-04-30</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>朋友问如何解读决策曲线分析，正好我也顺便学习一下，看到这篇文章[A simple, step-by-step guide to interpreting decision curve analysis](A simple, step-by-step guide to interpreting decision curve analysis)，这篇文章引用也100多了。</p>
<p>[TOC]</p>
<p><img src="/wp/f4w/2022/2022-04-30-DCAInterpreting1.png"></p>
<h1 id="Abstract"><a href="#Abstract" class="headerlink" title="Abstract"></a>Abstract</h1><p>背景</p>
<p>决策曲线分析是一种评估预测模型和诊断测试的方法，该方法在 2006 年的出版物中引入。 现在文献中普遍报道了决策曲线，但对其含义的理解仍然存在广泛的误解和混淆。</p>
<p>评论总结</p>
<p>在本文中，我们一步一步的解读决策曲线分析，并回答了有关该方法的一些常见问题。 我们认为，解释决策曲线的许多困难可以通过将 y 轴重新标记为“收益”并将 x 轴重新标记为“偏好”来解决。 如果模型或测试在一系列临床合理偏好中具有最高水平的益处，则可以推荐用于临床使用。</p>
<p>结论</p>
<p>如果读者和作者遵循一些简单的指导方针，决策曲线很容易理解。</p>
<h1 id="引言"><a href="#引言" class="headerlink" title="引言"></a>引言</h1><p>决策曲线分析是一种评估预测模型和诊断测试的方法，由 Vickers 和 Elkin 在 2006 年的 Medical Decision Making刊物上提出。该方法试图克服传统统计指标的局限性，例如区分和校准，因为它们不能直接提供临床价值信息以及完整决策分析方法，这些方法不适用于常规生物统计学实践。</p>
<p>简而言之，与治疗所有患者或不治疗患者的默认策略相比，决策曲线分析计算一个或多个预测模型或诊断测试的临床“净收益（net benefit）”。净收益是基于一系列阈值概率（0-1）计算的。</p>
<p><strong>阈值概率（threshold probability）定义为需要进一步干预的疾病的最小概率</strong>，我们可以理解为风险概率。</p>
<p><strong>净收益（net benefit） = sensitivity × prevalence – (1 – specificity) × (1 – prevalence) × <em>w</em> （这个公式多了prevalence）</strong></p>
<p>对于给出疾病预测概率 p̂ 的预测模型，在特定的pt阈值下，w是比值比odds，等于p̂/(1-p̂)，阳性样本的 p̂ ≥ pt，进而可以计算灵敏度和特异性。净收益不同于判别和校准等准确度指标，因为它包含了基于模型或测试做出的决策的后果。有关决策曲线分析背景的更多信息，参阅 Vickers 等人。 </p>
<p>06年文章中简单的公式没有prevalence，如下</p>
<p><img src="/wp/f4w/2022/2022-04-30-DCAInterpreting5.png"></p>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/1450/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1447/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1447/" class="post-title-link" itemprop="url">蛋白延伸（extension）变异的hgvs命名</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-04-25 12:06:20" itemprop="dateCreated datePublished" datetime="2022-04-25T12:06:20+08:00">2022-04-25</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>翻译<a target="_blank" rel="noopener" href="https://varnomen.hgvs.org/recommendations/protein/variant/extension/">https://varnomen.hgvs.org/recommendations/protein/variant/extension/</a></p>
<p>这个突变类型不常见，hgvs的命名挺有意思的</p>
<h1 id="延伸突变定义"><a href="#延伸突变定义" class="headerlink" title="延伸突变定义"></a>延伸突变定义</h1><p>序列变化导致参考氨基酸序列在N或者C端多了一个或者多个氨基酸</p>
<h1 id="描述"><a href="#描述" class="headerlink" title="描述"></a>描述</h1><h2 id="p-Met1ext-5"><a href="#p-Met1ext-5" class="headerlink" title="p.Met1ext-5"></a>p.Met1ext-5</h2><p>N端格式: “prefix”“Met1”“ext”“position_new_initiation_site”, e.g. p.Met1ext-5</p>
<p>“prefix” = 前缀，用p.表示<br>“Met1” = 正常的翻译起始位点Met1<br>“ext” = 变化类型是延伸ext<br>“position_new_initiation_site” = 上游新的翻译起始位点-5</p>
<h2 id="e-g-p-Ter110GlnextTer17"><a href="#e-g-p-Ter110GlnextTer17" class="headerlink" title="e.g. p.Ter110GlnextTer17"></a>e.g. p.Ter110GlnextTer17</h2><p>C端格式：“prefix”“Ter_position”“new_amino_acid”“ext”“Ter”“position_new_termination_site”, e.g. p.Ter110GlnextTer17</p>
<p>“prefix” = 前缀，用p.表示<br>“Ter_position” = 正常的终止位点Ter110<br>“new_amino_acid” = 在终止位点编码的新的氨基酸Gln<br>“ext” = 变化类型是延伸ext<br>“Ter” = 终止位置Ter（或者用*表示）<br>“position_new_termination_site” = 下游新的终止位点17</p>
<h1 id="注意"><a href="#注意" class="headerlink" title="注意"></a>注意</h1>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/1447/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1373/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1373/" class="post-title-link" itemprop="url">Entrez Direct (EDirect)!!! Convert Biosample ID to Run ID</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-04-01 10:46:20" itemprop="dateCreated datePublished" datetime="2022-04-01T10:46:20+08:00">2022-04-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>解决：提取Biosample的信息，将Biosample Id转换为SRA Run ID。</p>
<p>我有一个NCBI的biosmaple ID， 比如SAMN02324197，我不知道Bio project的情况下，我想知道他的SRR Run ID。最简单的办法是直接在SRA中搜这个biosample的ID，网页中Project，Run的信息都有了。</p>
<p>不过我如果有上百个biosample，来自不同的project，总不能一个一个的查吧。我先是在biostar上的这个<a target="_blank" rel="noopener" href="https://www.biostars.org/p/97782/">https://www.biostars.org/p/97782/</a>看到了EDirect可以查GSE的样本，我就在想能不能查sra的。</p>
<p>我试了一下这个命令，能检索到</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">&gt; esearch -db sra -query &quot;SAMN02324197&quot;</span><br><span class="line"></span><br><span class="line">&lt;ENTREZ_DIRECT&gt;</span><br><span class="line">  &lt;Db&gt;sra&lt;/Db&gt;</span><br><span class="line">  &lt;WebEnv&gt;MCID_6243c437ec6f7a20fc0f452a&lt;/WebEnv&gt;</span><br><span class="line">  &lt;QueryKey&gt;1&lt;/QueryKey&gt;</span><br><span class="line">  &lt;Count&gt;1&lt;/Count&gt;</span><br><span class="line">  &lt;Step&gt;1&lt;/Step&gt;</span><br><span class="line">&lt;/ENTREZ_DIRECT&gt;</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>但是我想提取相关的信息呢，添加了efetch即可</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line">&gt; esearch -db sra -query &quot;SAMN02324197&quot; | \</span><br><span class="line">         efetch -format docsum</span><br><span class="line"></span><br><span class="line">&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot; ?&gt;</span><br><span class="line">&lt;!DOCTYPE DocumentSummarySet&gt;</span><br><span class="line">&lt;DocumentSummarySet status=&quot;OK&quot;&gt;</span><br><span class="line">  &lt;DbBuild&gt;Build220329-1838m.1&lt;/DbBuild&gt;</span><br><span class="line">  &lt;DocumentSummary&gt;</span><br><span class="line">    &lt;Id&gt;477293&lt;/Id&gt;</span><br><span class="line">    &lt;ExpXml&gt;</span><br><span class="line">      &lt;Summary&gt;</span><br><span class="line">        &lt;Title&gt;Genome sequence of Apis mellifera worker from the A lineage [Africa]-S300&lt;/Title&gt;</span><br><span class="line">        &lt;Platform instrument_model=&quot;Illumina HiSeq 2000&quot;&gt;ILLUMINA&lt;/Platform&gt;</span><br><span class="line">        &lt;Statistics total_runs=&quot;1&quot; total_spots=&quot;195798261&quot; total_bases=&quot;9789913050&quot; total_size=&quot;7003530482&quot; load_done=&quot;true&quot; cluster_name=&quot;public&quot;/&gt;</span><br><span class="line">      &lt;/Summary&gt;</span><br><span class="line">      &lt;Submitter acc=&quot;SRA096086&quot; center_name=&quot;York University&quot; contact_name=&quot;Amro Zayed&quot; lab_name=&quot;Zayed Lab&quot;/&gt;</span><br><span class="line">      &lt;Experiment acc=&quot;SRX339524&quot; ver=&quot;2&quot; status=&quot;public&quot; name=&quot;Genome sequence of Apis mellifera worker from the A lineage [Africa]-S300&quot;/&gt;</span><br><span class="line">      &lt;Study acc=&quot;SRP029219&quot; name=&quot;Apis mellifera population genomics project&quot;/&gt;</span><br><span class="line">      &lt;Organism taxid=&quot;7460&quot; ScientificName=&quot;Apis mellifera&quot;/&gt;</span><br><span class="line">      &lt;Sample acc=&quot;SRS473244&quot; name=&quot;&quot;/&gt;</span><br><span class="line">      &lt;Instrument ILLUMINA=&quot;Illumina HiSeq 2000&quot;/&gt;</span><br><span class="line">      &lt;Library_descriptor&gt;</span><br><span class="line">        &lt;LIBRARY_STRATEGY&gt;WGS&lt;/LIBRARY_STRATEGY&gt;</span><br><span class="line">        &lt;LIBRARY_SOURCE&gt;GENOMIC&lt;/LIBRARY_SOURCE&gt;</span><br><span class="line">        &lt;LIBRARY_SELECTION&gt;RANDOM&lt;/LIBRARY_SELECTION&gt;</span><br><span class="line">        &lt;LIBRARY_LAYOUT&gt;</span><br><span class="line">          &lt;SINGLE/&gt;</span><br><span class="line">        &lt;/LIBRARY_LAYOUT&gt;</span><br><span class="line">      &lt;/Library_descriptor&gt;</span><br><span class="line">      &lt;Bioproject&gt;PRJNA216922&lt;/Bioproject&gt;</span><br><span class="line">      &lt;Biosample&gt;SAMN02324197&lt;/Biosample&gt;</span><br><span class="line">    &lt;/ExpXml&gt;</span><br><span class="line">    &lt;Runs&gt;</span><br><span class="line">      &lt;Run acc=&quot;SRR957095&quot; total_spots=&quot;195798261&quot; total_bases=&quot;9789913050&quot; load_done=&quot;true&quot; is_public=&quot;true&quot; cluster_name=&quot;public&quot; static_data_available=&quot;true&quot;/&gt;</span><br><span class="line">    &lt;/Runs&gt;</span><br><span class="line">    &lt;ExtLinks/&gt;</span><br><span class="line">    &lt;CreateDate&gt;2014/02/05&lt;/CreateDate&gt;</span><br><span class="line">    &lt;UpdateDate&gt;2013/08/26&lt;/UpdateDate&gt;</span><br><span class="line">  &lt;/DocumentSummary&gt;</span><br><span class="line">&lt;/DocumentSummarySet&gt;</span><br></pre></td></tr></table></figure>

<p>这个时候能看到想要的信息都在这里面了，然后试着提取相关的信息</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">&gt; esearch -db sra -query &quot;SAMN02324197&quot; | \</span><br><span class="line">         efetch -format docsum | \</span><br><span class="line">         xtract -pattern DocumentSummary -element Bioproject,Biosample,Title</span><br><span class="line"></span><br><span class="line">PRJNA216922     SAMN02324197    Genome sequence of Apis mellifera worker from the A lineage [Africa]-S300</span><br></pre></td></tr></table></figure>

<p>但这个时候SRR的ID死活提取不来，翻了NCBI的手册看到可以通过block和@提取，这样相关的biosample对应的Run ID信息就有了。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">&gt; esearch -db sra -query &quot;SAMN02324197&quot; | \</span><br><span class="line">         efetch -format docsum | \</span><br><span class="line">         xtract -pattern DocumentSummary -element Bioproject,Biosample,Title -block Runs -element Run@acc</span><br><span class="line"></span><br><span class="line">PRJNA216922     SAMN02324197    Genome sequence of Apis mellifera worker from the A lineage [Africa]-S300       SRR957095</span><br></pre></td></tr></table></figure>

<p>批量查询就变得很简单了</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">&gt; cat biosample.ids.list| head -4 | xargs -L 1 -i sh -c &quot;esearch -db sra -query &#123;&#125; | efetch -format docsum | xtract -pattern DocumentSummary -element Bioproject,Biosample,Title -block Runs -element Run@acc&quot;</span><br><span class="line"></span><br><span class="line">PRJNA418874     SAMN08038389    Resequencing of Apis cerana: adult female body  SRR6301311</span><br><span class="line">PRJNA418874     SAMN08038319    Resequencing of Apis cerana: adult female body  SRR6301296</span><br><span class="line">PRJNA418874     SAMN08038413    Resequencing of Apis cerana: adult female body  SRR6301438</span><br><span class="line">PRJNA418874     SAMN08038427    Resequencing of Apis cerana: adult female body  SRR6301396</span><br></pre></td></tr></table></figure>

<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/books/NBK179288/">https://www.ncbi.nlm.nih.gov/books/NBK179288/</a></p>
<p><a target="_blank" rel="noopener" href="https://www.biostars.org/p/97782/">https://www.biostars.org/p/97782/</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1363/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1363/" class="post-title-link" itemprop="url">MobaXterm SSHTunnel/v2ray + Switch Omega + Proxifier</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-03-29 13:01:20" itemprop="dateCreated datePublished" datetime="2022-03-29T13:01:20+08:00">2022-03-29</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>这两年在大陆都是用学校的VPN访问外网，但最近VPN太烂太烂了，经常掉线。这怎么能忍，于是我想到了代理。花钱的事我又不干，我又有VPS，不如让VPS当作代理服务器。一共两种方式都可以实现：socks或v2ray。</p>
<h1 id="（1）MobaXterm"><a href="#（1）MobaXterm" class="headerlink" title="（1）MobaXterm"></a>（1）MobaXterm</h1><p>我一直用mobaXterm的Portabl版本（<a target="_blank" rel="noopener" href="https://mobaxterm.mobatek.net/%EF%BC%89%EF%BC%8C%E8%BF%99%E5%87%A0%E5%B9%B4%E8%B5%B0%E5%88%B0%E5%93%AA%E9%83%BD%E4%BC%9A%E6%8A%8AmobaXterm%E7%9A%84%E6%96%87%E4%BB%B6%E5%A4%B9%E5%90%8C%E6%AD%A5%E8%BF%87%E6%9D%A5%EF%BC%8C%E9%81%BF%E5%85%8D%E6%AF%8F%E6%AC%A1%E5%BB%BA%E7%AB%8B%E6%96%B0%E7%9A%84ssh%E3%80%82%E6%88%91%E4%B9%9F%E4%B8%80%E7%9B%B4%E7%9F%A5%E9%81%93MobaXterm%E6%9C%89%E4%B8%AATunneling%E7%9A%84%E6%A0%87%E7%AD%BE%EF%BC%8C%E4%BA%8E%E6%98%AF%E5%B0%B1%E7%A0%94%E7%A9%B6%E4%B8%8B%E4%BA%86MobaXterm%E8%87%AA%E5%B8%A6%E7%9A%84%E3%80%82%E8%BF%99%E4%B8%AA%E6%AF%94%E8%BE%83%E7%AE%80%E5%8D%95%EF%BC%8C%E6%88%91%E7%9B%B4%E6%8E%A5%E5%9C%A8mobaxterm%E4%B8%8A%E8%AE%BE%E7%BD%AE%E5%90%8E%EF%BC%8C%E5%B0%B1%E5%BB%BA%E7%AB%8B%E4%BA%86%E9%9A%A7%E9%81%93%E3%80%82">https://mobaxterm.mobatek.net/），这几年走到哪都会把mobaXterm的文件夹同步过来，避免每次建立新的ssh。我也一直知道MobaXterm有个Tunneling的标签，于是就研究下了MobaXterm自带的。这个比较简单，我直接在mobaxterm上设置后，就建立了隧道。</a></p>
<p>主要是建立SSH隧道，需要一个境外的ssh帐号。</p>
<p><img src="/wp/f4w/2022/2022-03-29-SocksProxy-1.png"></p>
<h1 id="（2）V2ray"><a href="#（2）V2ray" class="headerlink" title="（2）V2ray"></a>（2）V2ray</h1><p>V2ray虽然高级，但是麻烦啊，还要安装还是配置。我参考的是<a target="_blank" rel="noopener" href="https://www.v2fly.org/%EF%BC%8C%E6%9C%AC%E5%9C%B0%E7%94%A8%E7%9A%84%E6%98%AFv2rayN">https://www.v2fly.org/，本地用的是v2rayN</a> （<a target="_blank" rel="noopener" href="https://github.com/2dust/v2rayN%EF%BC%89%E3%80%82">https://github.com/2dust/v2rayN）。</a></p>
<p>无论是1还是2，都是建立和远程代理服务器的连接，然后设定本地localhost的一个特定端口，通过本地的这个端口访问。</p>
<h1 id="设置Switch-Omega"><a href="#设置Switch-Omega" class="headerlink" title="设置Switch Omega"></a>设置Switch Omega</h1><p>这个我从会翻墙的时候就在用，不多说，自己设置规则就行。如果你用了v2ray，还可以增加http代理的端口。</p>
<p><img src="/wp/f4w/2022/2022-03-29-SocksProxy-2.png"></p>
<h1 id="设置Proxifier"><a href="#设置Proxifier" class="headerlink" title="设置Proxifier"></a>设置Proxifier</h1><p>设置omega之后，网页可以访问了，但我还想让我电脑的特定软件也翻墙，于是我知道了proxifier，比如我专门让google drive和one drive走外网。本来是想利用windows自带的proxy来设置的，但那和proxifier更好用哈。</p>
<p><img src="/wp/f4w/2022/2022-03-29-SocksProxy-3.png"></p>
<p>PS：纪念下从12年就开始用的goagent/xxnet。</p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1358/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1358/" class="post-title-link" itemprop="url">False-positive genes with high frequency mutation</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-03-19 12:23:20" itemprop="dateCreated datePublished" datetime="2022-03-19T12:23:20+08:00">2022-03-19</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <h1 id="Mutational-heterogeneity-in-cancer-and-the-search-for-new-cancer-associated-genes"><a href="#Mutational-heterogeneity-in-cancer-and-the-search-for-new-cancer-associated-genes" class="headerlink" title="Mutational heterogeneity in cancer and the search for new cancer-associated genes"></a>Mutational heterogeneity in cancer and the search for new cancer-associated genes</h1><p><a target="_blank" rel="noopener" href="https://www.nature.com/articles/nature12213">https://www.nature.com/articles/nature12213</a></p>
<blockquote>
<p>After recognizing the problem of apparent false-positive findings, we reviewed the published literature and found that some of these potentially spurious genes have already been nominated as cancer-associated genes in recently published cancer genome studies: for example, LRP1B in glioblastoma and lung adenocarcinoma; CSMD3 in ovarian cancer; PCLO in DLBCL; MUC16 in lung squamous carcinoma11, breast cancer and DLBCL; MUC4 in melanoma; olfactory receptor OR2L13 in glioblastoma14; and TTN in breast cancer and other tumour types. We therefore set out to understand the source of the problem.</p>
</blockquote>
<h1 id="Mutational-heterogeneity-in-cancer-and-the-search-for-new-cancer-associated-genes-1"><a href="#Mutational-heterogeneity-in-cancer-and-the-search-for-new-cancer-associated-genes-1" class="headerlink" title="Mutational heterogeneity in cancer and the search for new cancer-associated genes"></a>Mutational heterogeneity in cancer and the search for new cancer-associated genes</h1><p><a target="_blank" rel="noopener" href="https://www.nature.com/articles/nature12213">https://www.nature.com/articles/nature12213</a></p>
<blockquote>
<p>As such, several large genes, including TTN, MUC19, MUC16, and KMT2D, had more false-positive results that could be screened by our study</p>
</blockquote>
<h1 id="Targeted-sequencing-reveals-the-somatic-mutation-landscape-in-a-Swedish-breast-cancer-cohort"><a href="#Targeted-sequencing-reveals-the-somatic-mutation-landscape-in-a-Swedish-breast-cancer-cohort" class="headerlink" title="Targeted sequencing reveals the somatic mutation landscape in a Swedish breast cancer cohort"></a>Targeted sequencing reveals the somatic mutation landscape in a Swedish breast cancer cohort</h1><p><a target="_blank" rel="noopener" href="https://www.nature.com/articles/s41598-020-74580-1">https://www.nature.com/articles/s41598-020-74580-1</a></p>
<blockquote>
<p>The high variance is due to two genes (TTN andMUC16) having extremely long lengths (35992 and 14508 aa respectively) compared to the rest of the protein coding genes. </p>
</blockquote>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1354/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1354/" class="post-title-link" itemprop="url">突变mutation signatures</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-03-16 12:35:20" itemprop="dateCreated datePublished" datetime="2022-03-16T12:35:20+08:00">2022-03-16</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>拿到突变的数据之后，一般会先看除了进行突变注释，看突变所在的基因和造成的氨基酸变化，还会看具体的碱基变化类型，共96种，以三个碱基为统计单位，看三联核苷酸中间的碱基变化类型的个数，以C变为A为例（等同负链G变为T），NCN&gt;NAN，N可以为任意碱基，所以有4*4=16种，所有情况为96种。<br><img src="/wp/f4w/2022/2022-03-16-MutationTypes_v3.jpg"></p>
<p>统计96种突变类型的具体数目，之后会进行突变的signature分析有三种：</p>
<ol>
<li>与已知的cosmic signature进行比较，看哪些signature比重高</li>
<li>利用非负矩阵分解NFM找novel的signature</li>
<li>分析1和2之后，看novel的signature和已知的signature的相似性</li>
</ol>
<p>突变的signature是内外部环境共同作用下造成的一些特定的变异特征，这些因素包括DNA修复或者复制缺陷，吸烟饮酒等等。</p>
<h1 id="找cosmic的signature"><a href="#找cosmic的signature" class="headerlink" title="找cosmic的signature"></a>找cosmic的signature</h1><p>利用R包deconstructSigs进行</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 假设你有一个数据框，包含了突变的信息，以包自带的sample.mut.ref为例</span></span><br><span class="line">head(sample.mut.ref)</span><br><span class="line"></span><br><span class="line"><span class="comment"># Convert to deconstructSigs input，得到96碱基突变数据框</span></span><br><span class="line"><span class="comment"># 指定染色体位置、突变碱基和样本名所在的列名</span></span><br><span class="line">sigs.input &lt;- mut.to.sigs.input(mut.ref = sample.mut.ref, </span><br><span class="line">                                sample.id = <span class="string">&quot;Sample&quot;</span>, </span><br><span class="line">                                chr = <span class="string">&quot;chr&quot;</span>, </span><br><span class="line">                                pos = <span class="string">&quot;pos&quot;</span>, </span><br><span class="line">                                ref = <span class="string">&quot;ref&quot;</span>, </span><br><span class="line">                                alt = <span class="string">&quot;alt&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 提取样本的mutation signature</span></span><br><span class="line"><span class="comment"># 这里的signature.nature2013或者signatures.cosmic就是参考的signature，如果你有自己的signature数据，比如SomaticSignatures包找到的，也可以在这个地方指定，这样就计算的是提供的singature</span></span><br><span class="line">sample_1 = whichSignatures(tumor.ref = sigs.input, </span><br><span class="line">                           signatures.ref = signatures.nature2013, <span class="comment"># 参考signature </span></span><br><span class="line">                           sample.id = <span class="number">1</span>,  <span class="comment"># 样本的barcode </span></span><br><span class="line">                           contexts.needed = <span class="literal">TRUE</span>,</span><br><span class="line">                           tri.counts.method = <span class="string">&#x27;default&#x27;</span> <span class="comment"># 是否进行normalzie</span></span><br><span class="line">                           )</span><br><span class="line">                           </span><br><span class="line">sample_1$weights[<span class="number">1</span>:<span class="number">3</span>]</span><br><span class="line"></span><br><span class="line"><span class="comment">#  Signature.1A Signature.1B Signature.2</span></span><br><span class="line"><span class="comment">#1            0    0.1564832           0</span></span><br><span class="line"> </span><br></pre></td></tr></table></figure>
<h1 id="找novel的signature"><a href="#找novel的signature" class="headerlink" title="找novel的signature"></a>找novel的signature</h1><p>利用SomaticSignatures包进行，这里找到的novel signature可以作为参考的signature用deconstructSigs，注意96突变类型数据框的名应一致。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br></pre></td><td class="code"><pre><span class="line">library(SomaticSignatures)</span><br><span class="line">library(SomaticCancerAlterations)</span><br><span class="line">library(BSgenome.Hsapiens.1000genomes.hs37d5)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">sca_metadata = scaMetadata() </span><br><span class="line"></span><br><span class="line"><span class="comment"># sca_data = unlist(scaLoadDatasets())  # 官网用的多种是癌症的数据</span></span><br><span class="line"><span class="comment"># 我们以TCGA的LUAD的全外数据为例</span></span><br><span class="line">sca_data = (scaLoadDatasets())$luad_tcga</span><br><span class="line"></span><br><span class="line">sca_data$study = factor(gsub(<span class="string">&quot;(.*)_(.*)&quot;</span>, <span class="string">&quot;\\1&quot;</span>, toupper(<span class="built_in">names</span>(sca_data))))</span><br><span class="line">sca_data = unname(subset(sca_data, Variant_Type %in% <span class="string">&quot;SNP&quot;</span>))</span><br><span class="line">sca_data = keepSeqlevels(sca_data, hsAutosomes(), pruning.mode = <span class="string">&quot;coarse&quot;</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># 构建对象，这里可以自己根据自己的数据构建VRanges对象</span></span><br><span class="line">sca_vr = VRanges(</span><br><span class="line">    seqnames = seqnames(sca_data),</span><br><span class="line">    ranges = ranges(sca_data),</span><br><span class="line">    ref = sca_data$Reference_Allele,</span><br><span class="line">    alt = sca_data$Tumor_Seq_Allele2,</span><br><span class="line">    sampleNames = sca_data$Patient_ID,</span><br><span class="line">    seqinfo = seqinfo(sca_data),</span><br><span class="line">    study = sca_data$study)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">sca_motifs = mutationContext(sca_vr, BSgenome.Hsapiens.1000genomes.hs37d5)</span><br><span class="line"><span class="comment"># 这里得到96碱基突变数据框</span></span><br><span class="line">sca_mm = motifMatrix(sca_motifs, normalize = <span class="literal">TRUE</span>)</span><br><span class="line"><span class="comment"># 可视化</span></span><br><span class="line">plotMutationSpectrum(sca_motifs)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 找最佳的signature数目，一种是NMF，一种是PCA方法</span></span><br><span class="line">n_sigs = <span class="number">2</span>:<span class="number">8</span></span><br><span class="line">gof_nmf = assessNumberSignatures(sca_mm, n_sigs, nReplicates = <span class="number">5</span>)</span><br><span class="line">gof_pca = assessNumberSignatures(sca_mm, n_sigs, pcaDecomposition)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 确定最佳的数目</span></span><br><span class="line">plotNumberSignatures(gof_nmf)</span><br><span class="line">plotNumberSignatures(gof_pca)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 找signature</span></span><br><span class="line">n_sigs = <span class="number">5</span> <span class="comment"># 假设最佳的是5</span></span><br><span class="line">sigs_nmf = identifySignatures(sca_mm, n_sigs, nmfDecomposition)</span><br><span class="line">sigs_pca = identifySignatures(sca_mm, n_sigs, pcaDecomposition)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">library(ggplot2)</span><br><span class="line"></span><br><span class="line"><span class="comment"># 看每个样本的signature的权重</span></span><br><span class="line">plotSignatureMap(sigs_nmf) + ggtitle(<span class="string">&quot;Somatic Signatures: NMF - Heatmap&quot;</span>)</span><br><span class="line"><span class="comment"># 看每个样本具体的突变类型统计</span></span><br><span class="line">plotSignatures(sigs_nmf) + ggtitle(<span class="string">&quot;Somatic Signatures: NMF - Barchart&quot;</span>)</span><br><span class="line"><span class="comment"># 没放图，具体的图可以参考 https://bioconductor.org/packages/release/bioc/vignettes/SomaticSignatures/inst/doc/SomaticSignatures-vignette.html</span></span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>另外还可以通过cosine similarity 比较两个signature之间或者样本之间的相似性。</p>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p><a target="_blank" rel="noopener" href="https://github.com/raerose01/deconstructSigs">https://github.com/raerose01/deconstructSigs</a></p>
<p><a target="_blank" rel="noopener" href="https://bioconductor.org/packages/release/bioc/vignettes/SomaticSignatures/inst/doc/SomaticSignatures-vignette.html">https://bioconductor.org/packages/release/bioc/vignettes/SomaticSignatures/inst/doc/SomaticSignatures-vignette.html
</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1348/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1348/" class="post-title-link" itemprop="url">Prognostic versus predictive value of biomarkers</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2022-02-15 12:23:20" itemprop="dateCreated datePublished" datetime="2022-02-15T12:23:20+08:00">2022-02-15</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <h1 id="1"><a href="#1" class="headerlink" title="1"></a>1</h1><p>A prognostic biomarker provides information about the patients overall cancer outcome, regardless of therapy, whilst a predictive biomarker gives information about the effect of a therapeutic intervention. A predictive biomarker can be a target for therapy. </p>
<p>REF: PMID 18396036, “Prognostic versus predictive value of biomarkers in oncology”</p>
<h1 id="2"><a href="#2" class="headerlink" title="2"></a>2</h1><p>The term prognostic value refers to a genetic factor’s ability to project the natural history of disease in relation to another factor (such as treatment or environmental exposure or another genetic factor; henceforth referred to as treatment) by discriminating between good versus bad prognosis, thereby providing insights into whom to treat with novel modalities.</p>
<p>The term predictive value refers to a genetic factor’s ability to project the benefit of treatment under a suitable definition of the term benefit [Sawyers 2008; Italiano 2011]. When the magnitude of treatment benefit changes with the level of the genetic factor, a gene-treatment interaction can be included in a relevant statistical model to represent this change. </p>
<p>REF: PMID 26349638, “Prognostic and predictive values and statistical interactions in the era of targeted treatment”</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1309/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1309/" class="post-title-link" itemprop="url">群体遗传学中基于Fst&Pi的选择消除分析</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-11-01 12:02:20" itemprop="dateCreated datePublished" datetime="2021-11-01T12:02:20+08:00">2021-11-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <h1 id="Fst衡量群体分化程度"><a href="#Fst衡量群体分化程度" class="headerlink" title="Fst衡量群体分化程度"></a>Fst衡量群体分化程度</h1><p>1说明两个population是完全独立的。0说明两个population之间自由interbreeding。Fst值越大，说明genetic distance越远。值越低，说明大多数的genetic variation是发生在同一个population的。</p>
<p>Wright建议，实际研究中，F ST为0～0.05:群体间遗传分化很小，可以不考虑；<br>F ST为0.05～0.15，群体间存在中等程度的遗传分化；<br>F ST为0.15~0.25，群体间遗传分化较大；<br>F ST为0.25以上，群体间有很大的遗传分化。</p>
<p>其中 代表 Weir &amp; Cockerham 的 Fst。F 统计量反映了群体结构的变化，它受不同因素的影响，比如突变，遗传漂变，近亲交配，选择作用或 Wahlund 效应（指一个种群中由于亚种群的结构导致的异质性的下降）。在中性进化条件下，F 统计量的大小主要决定于遗传漂变和迁移等因素的影响，如果种群中一个等位基因因为对于特定生境的适合度较高而经历适应性选择，那么其频率的升高会增大种群分化水平，反映在 F 统计量上就是有较高的 Fst 值</p>
<p><img src="/wp/f4w/2021/2021-11-01-FST.jpg"></p>
<p>vcftools就可以做</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line"># 如果是vcf.gz请用--gzvcf</span><br><span class="line">vcftools --vcf pop.vcf --weir-fst-pop Pop1.txt --weir-fst-pop Pop2.txt --out P1vsP2.Fst </span><br><span class="line"></span><br><span class="line">$ head -5 P1vsP2.Fst.weir.fst</span><br><span class="line">CHROM   POS     WEIR_AND_COCKERHAM_FST</span><br><span class="line">chr1     1216    -0.155606</span><br><span class="line">chr1     1226    -0.0884707</span><br><span class="line">chr1     1480    0.0448135</span><br><span class="line">chr1     1481    0.0275202</span><br><span class="line"></span><br><span class="line"># 默认基于位点进行统计，如果需要滑窗统计，可以添加--fst-window-size和--fst-window-step选项</span><br><span class="line"></span><br><span class="line">$ head -5 P1vsp2.Fst.windowed.weir.fst</span><br><span class="line">CHROM   BIN_START       BIN_END N_VARIANTS      WEIGHTED_FST    MEAN_FST</span><br><span class="line">chr1     1       40000   79      0.178739        0.0986989</span><br><span class="line">chr1     20001   60000   84      0.21267 0.143746</span><br><span class="line">chr1     40001   80000   118     0.315279        0.19392</span><br><span class="line">chr1     60001   100000  147     0.361274        0.215205</span><br></pre></td></tr></table></figure>

<h1 id="Pi衡量遗传多样性"><a href="#Pi衡量遗传多样性" class="headerlink" title="Pi衡量遗传多样性"></a>Pi衡量遗传多样性</h1><p>Pi主要用来衡量nucleotide divergency</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br></pre></td><td class="code"><pre><span class="line"># vcftools同样可以计算pi值，如果是vcf.gz请用--gzvcf</span><br><span class="line">vcftools --vcf pop1.vcf --site-pi --out P1.Pi</span><br><span class="line">vcftools --vcf pop2.vcf --site-pi --out P2.Pi</span><br><span class="line"></span><br><span class="line">$ head -5 P1.Pi.sites.pi</span><br><span class="line">CHROM   POS     PI</span><br><span class="line">chr1     1216    0.5</span><br><span class="line">chr1     1226    0.6</span><br><span class="line">chr1     1480    0</span><br><span class="line">chr1     1481    0</span><br><span class="line"></span><br><span class="line"># 如果需要滑窗处理可以添加--window-pi和--window-pi-step选项</span><br><span class="line"></span><br><span class="line">$ head -5 P1.Pi.windowed.pi</span><br><span class="line">CHROM   BIN_START       BIN_END N_VARIANTS      PI</span><br><span class="line">chr1     1       40000   69      0.000217064</span><br><span class="line">chr1     20001   60000   74      0.000241674</span><br><span class="line">chr1     40001   80000   91      0.000350297</span><br><span class="line">chr1     60001   100000  111     0.000413938</span><br></pre></td></tr></table></figure>

<h1 id="联合分析"><a href="#联合分析" class="headerlink" title="联合分析"></a>联合分析</h1><p>Fst &amp; Pi 已被证实是一种很有效力地检测选择消除区域的方法，特别是在挖掘与生存环境密切相关的功能区时，往往可以得到较强的选择信号，二者共同筛选较强的选择信号（比如Fst和Pi都处于Top 5%），便于目标基因的筛选。文章中常画的图是一个二维的散点图，如果要研究群体1和群体2，横坐标是log2(Pi Ratio)=log2(P1.Pi/P2.Pi)，纵坐标为 P1 vs P2的Fst值，并且把对应的频率分布图画在右方和上方，右上方的点是受选择的区域。</p>
<p>有了区域之后，可以进一步做富集分析或者验证候选基因。</p>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p><a target="_blank" rel="noopener" href="https://vcftools.github.io/man_latest.html">https://vcftools.github.io/man_latest.html</a><br><a target="_blank" rel="noopener" href="https://blog.csdn.net/q623928815/article/details/78627610">https://blog.csdn.net/q623928815/article/details/78627610</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1298/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1298/" class="post-title-link" itemprop="url">肿瘤免疫表型-cold (excluded, desert) and hot</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-10-11 05:09:20" itemprop="dateCreated datePublished" datetime="2021-10-11T05:09:20+08:00">2021-10-11</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>根据免疫状态，肿瘤可以分为hot，肿瘤内外富集免疫浸润淋巴细胞，有明显的免疫浸润性；excluded，免疫细胞没有浸润到肿瘤内部而富集在肿瘤外部，desert，肿瘤不怎么有免疫浸润细胞。</p>
<p>理论上讲，hot tumor更能从免疫治疗获得收益。</p>
<h3 id="Tumor-immune-phenotypes"><a href="#Tumor-immune-phenotypes" class="headerlink" title="Tumor immune phenotypes"></a>Tumor immune phenotypes</h3><p><img src="/wp/f4w/2021/2021-10-10TumorImmunePhenotypes1.jpg"></p>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/1298/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1283/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/1283/" class="post-title-link" itemprop="url">C++14 standard requested but CXX14 is not defined</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2021-09-09 08:04:20" itemprop="dateCreated datePublished" datetime="2021-09-09T08:04:20+08:00">2021-09-09</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>在安装R包的时候遇到报错，C++14 standard requested but CXX14 is not defined</p>
<p>查了很多办法，刚开始是根据<a target="_blank" rel="noopener" href="https://github.com/stan-dev/rstan/issues/892%E4%BF%AE%E6%94%B9.R%E4%B8%8B%E9%9D%A2%E7%9A%84Makevars%EF%BC%8C">https://github.com/stan-dev/rstan/issues/892修改.R下面的Makevars，</a></p>
<p>但是包另外一个错g++: error: unrecognized command line option ‘-std=c++14’</p>
<p>于是继续查到c++1y这个问题，但依然没有解决问题。</p>
<p>复盘了一下，感觉是gcc的问题，所以升级了最新的gcc</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"># 系统是CentOS</span><br><span class="line">sudo yum install centos-release-scl</span><br><span class="line">sudo yum install devtoolset-10</span><br><span class="line">scl enable devtoolset-10 bash</span><br></pre></td></tr></table></figure>

<p>但是装包的时候新版的gcc依然不能别识别，所以修改Makevars，最终用了如下的配置,重点是指定了新版的g++和c++的路径，这样问题就解决了</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">MAKEFLAGS = -j18</span><br><span class="line"></span><br><span class="line">## C++ flags</span><br><span class="line">CXX=g++</span><br><span class="line">CXX11=g++</span><br><span class="line">CXX14=/opt/rh/devtoolset-9/root/usr/bin/g++</span><br><span class="line">CXX17=g++</span><br><span class="line"></span><br><span class="line">CXXFLAGS=-O3 -march=native -Wno-ignored-attributes</span><br><span class="line">CXX11FLAGS=-O3 -march=native -Wno-ignored-attributes</span><br><span class="line">CXX14FLAGS=-O3 -march=native -Wno-ignored-attributes</span><br><span class="line">CXX17FLAGS=-O3 -march=native -Wno-ignored-attributes</span><br><span class="line"></span><br><span class="line">CXXPICFLAGS=-fPIC</span><br><span class="line">CXX11PICFLAGS=-fPIC</span><br><span class="line">CXX14PICFLAGS=-fPIC</span><br><span class="line">CXX17PICFLAGS=-fPIC</span><br><span class="line"></span><br><span class="line">CXX11STD=-std=c++11</span><br><span class="line">CXX14STD=-std=c++14</span><br><span class="line">CXX17STD=-std=c++17</span><br><span class="line"></span><br><span class="line">## C flags</span><br><span class="line">CC=/opt/rh/devtoolset-10/root/usr/bin/gcc</span><br><span class="line">FLAGS=-O3 -march=native</span><br><span class="line"></span><br><span class="line">## Fortran flags</span><br><span class="line">FC=gfortran</span><br><span class="line">F77=gfortran</span><br><span class="line">FFLAGS=-O3 -march=native</span><br><span class="line">FCFLAGS=-O3 -march=native</span><br></pre></td></tr></table></figure>

<p>总结：<br>1，upgrade gcc</p>
<p>2, specify the absolute gcc and g++ path </p>
<p>如果可以的话，建议把整个系统的gcc都替换成新版的</p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  


  
  <nav class="pagination">
    <span class="page-number current">1</span><a class="page-number" href="/page/2/">2</a><span class="space">&hellip;</span><a class="page-number" href="/page/17/">17</a><a class="extend next" rel="next" href="/page/2/"><i class="fa fa-angle-right" aria-label="Next page"></i></a>
  </nav>



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Jason</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">163</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">167</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>


<!--google_adsense-->

  <div class="links-of-blogroll-title">
    <i class="fa fa-google"></i> AdSense
    <ins class="adsbygoogle"
        style="display:block"
        data-ad-client="ca-pub-9370593480269435"
        data-ad-slot="8824957305"
        data-ad-format="auto"
        data-full-width-responsive="true"></ins>
    <script>
      (adsbygoogle = window.adsbygoogle || []).push({});
    </script>
  </div>

<!--/google_adsense-->



    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Jason</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

<script data-ad-client="ca-pub-9370593480269435" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<!-- PGAD -->
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-9370593480269435"
     data-ad-slot="8824957305"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
