<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"md.zxzyl.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>


  <meta property="og:type" content="website">
<meta property="og:title" content="Zhongxu&#39;s blog">
<meta property="og:url" content="https://md.zxzyl.com/page/9/index.html">
<meta property="og:site_name" content="Zhongxu&#39;s blog">
<meta property="og:locale" content="en_US">
<meta property="article:author" content="Jason">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="https://md.zxzyl.com/page/9/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : true,
    isPost : false,
    lang   : 'en'
  };
</script>

  <title>Zhongxu's blog</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a3b5d3f466edb3e3f15e8238d8f04be8";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Zhongxu's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content index posts-expand">
            
      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/789/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/789/" class="post-title-link" itemprop="url">测序中加入Phix的作用</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2017-12-01 23:40:54" itemprop="dateCreated datePublished" datetime="2017-12-01T23:40:54+08:00">2017-12-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>测序建库的时候，会加入一定比例的Phix，那么Phix文库有什么作用呢，我转了两篇文章，方便大家理解。Phix文库最主要的目的1）是调节碱基平衡，改善测序仪的空间校正，便于后期提高base calling的准确性，2）由于Phix序列已知基因组较小，在测序的过程中Illumina的测序仪就开始将测的read与phix基因组进行比较，预估测序指标。我也遇到过，Illumina工程师在维护测序仪时，用Phix文库测试。转载内容详见下文</p>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/789/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/788/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/788/" class="post-title-link" itemprop="url">Calculate phastCon Score for a gene ---- 计算基因的phastCon平均分，判断基因保守型</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-11-04 21:45:25" itemprop="dateCreated datePublished" datetime="2016-11-04T21:45:25+08:00">2016-11-04</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>Calculate phastCon Score for a gene —- 计算基因的phastCon平均分，判断基因保守型</p>
<p>PhastCon socre is the score from 0 to 1 to show the conservation level.</p>
<p>A score showing the posterior probability that phastCons’s phylogenetic hidden Markov model (HMM) is in its most conserved state at that base position.</p>
<p>The phastCons scores represent probabilities of negative selection and range between 0 and 1. </p>
<p>Short highly-conserved regions and long moderately conserved regions can both obtain high scores. </p>
<p>也就是说如果某个位点或者一段序列的phastCon分值高的话，表示保守型较高。</p>
<h1 id="1，下载phast-score的wigFix格式文件（以19号染色体为测试）"><a href="#1，下载phast-score的wigFix格式文件（以19号染色体为测试）" class="headerlink" title="1，下载phast score的wigFix格式文件（以19号染色体为测试）"></a>1，下载phast score的wigFix格式文件（以19号染色体为测试）</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">wget -c http://hgdownload.cse.ucsc.edu/goldenPath/hg19/phastCons46way/placentalMammals/chr19.phastCons46way.placental.wigFix.gz</span><br><span class="line">gunzip -k chr19.phastCons46way.placental.wigFix.gz</span><br><span class="line"></span><br><span class="line">head chr19.phastCons46way.placental.wigFix</span><br><span class="line"></span><br><span class="line">fixedStep chrom=chr19 start=60001 step=1</span><br><span class="line">0.412</span><br><span class="line">0.407</span><br><span class="line">0.394</span><br><span class="line">0.393</span><br><span class="line">0.381</span><br><span class="line">0.381</span><br><span class="line">0.369</span><br><span class="line">0.345</span><br><span class="line">0.291</span><br></pre></td></tr></table></figure>

<p>wigFix格式是 fixed 的 Wiggle 格式，根据上面的前几行，应该能够看出该文件表示19号染色体，从60001位置开始，每一步step（base）的phastCon分值。</p>
<h1 id="利用bedops，将wig转成bed文件"><a href="#利用bedops，将wig转成bed文件" class="headerlink" title="利用bedops，将wig转成bed文件"></a>利用bedops，将wig转成bed文件</h1><p>bedops的官网 <a target="_blank" rel="noopener" href="https://bedops.readthedocs.io/en/latest/">https://bedops.readthedocs.io/en/latest/</a> ， 转成bed文件便于利用bed工具进行处理。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">wig2bed chr19.phastCons46way.placental.wigFix &gt; chr19.phastCons46way.placental.bed</span><br></pre></td></tr></table></figure>

<p>如果需要合并的话，运行下面的命令即可，这里以chr19染色体为例，就不合并了。<br>bedops –everything chr*.bed &gt; vertebrate.phastCons46.bed</p>
<h1 id="将gtf文件分割成已基因为单位的gtf文件"><a href="#将gtf文件分割成已基因为单位的gtf文件" class="headerlink" title="将gtf文件分割成已基因为单位的gtf文件"></a>将gtf文件分割成已基因为单位的gtf文件</h1><p>gtf文件中只有chr19染色体的基因注释信息。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">mkdir test</span><br><span class="line">awk -F &#x27;t&#x27; &#x27;&#123;split($9,a,&quot;;&quot;); split(a[1],a,&quot; &quot;); gsub(&quot;&quot;&quot;, &quot;&quot;, a[2]); x=&quot;test/&quot;a[2]&quot;.gtf&quot;; print &gt;&gt; (&quot;test/&quot;a[2]&quot;.gtf&quot;); close(x);&#125;&#x27; Homo_sapiens.GRCh38.85.gtf &amp;&amp; rm test/.gtf</span><br></pre></td></tr></table></figure>




<h1 id="将gft文件转成bed文件"><a href="#将gft文件转成bed文件" class="headerlink" title="将gft文件转成bed文件"></a>将gft文件转成bed文件</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">rm test/*bed</span><br><span class="line">for fn in `ls test/*gtf`; do</span><br><span class="line">    gene=`echo $fn &#x27; cut -d . -f 1`</span><br><span class="line">    gtf2bed $fn &#x27; sort -k1,1 -k2,2n &gt;&gt; $gene.bed</span><br><span class="line">    rm $fn</span><br><span class="line">done</span><br></pre></td></tr></table></figure>


<p>gtf2bed是<a target="_blank" rel="noopener" href="http://wp.zxzyl.com/?p=270">前面文章</a>中提到的，只要能将gtf转成bed就可。</p>
<h1 id="利用bedtools工具计算每个序列的平均phastCon-score"><a href="#利用bedtools工具计算每个序列的平均phastCon-score" class="headerlink" title="利用bedtools工具计算每个序列的平均phastCon score"></a>利用bedtools工具计算每个序列的平均phastCon score</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"># 最新版的bedtools有split选项</span><br><span class="line">for fn in `ls test/*.bed`;do</span><br><span class="line">bedtools map -split -sorted -a $fn -b chr19.phastCons46way.placental.bed -o mean &#x27; awk -F &#x27;t&#x27; &#x27;&#123;printf (&quot;%st%4fn&quot;, $1,$NF)&#125;&#x27;</span><br><span class="line">done</span><br></pre></td></tr></table></figure>


<p>bedtools官网 <a target="_blank" rel="noopener" href="http://bedtools.readthedocs.io/">bedtools.readthedocs.io</a> ，mean表示计算平均分。结果如下，平均分越高越保守。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">chr19   531711  542092  ENST00000215574 0.388707</span><br><span class="line">chr19   531768  536715  ENST00000586788 0.317417</span><br><span class="line">chr19   532031  542092  ENST00000586283 0.431241</span><br><span class="line">chr19   532099  541585  ENST00000607527 0.643244</span><br><span class="line">chr19   535836  542087  ENST00000606065 0.404337</span><br><span class="line">chr19   536952  542042  ENST00000606400 0.288264</span><br><span class="line">chr19   541216  541661  ENST00000593036 0.434159</span><br><span class="line">chr19   489175  505342  ENST00000587541 0.032945</span><br><span class="line">chr19   496453  505207  ENST00000613880 0.186624</span><br><span class="line">chr19   496453  505207  ENST00000617201 0.194969&lt;</span><br></pre></td></tr></table></figure>




<h1 id="参考："><a href="#参考：" class="headerlink" title="参考："></a>参考：</h1><p><strong><a target="_blank" rel="noopener" href="https://www.biostars.org/p/150152/">https://www.biostars.org/p/150152/</a></strong></p>
<p><strong><a target="_blank" rel="noopener" href="http://ccg.vital-it.ch/mga/mm9/phastcons/phastcons.html">http://ccg.vital-it.ch/mga/mm9/phastcons/phastcons.html</a></strong></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>########################################################################################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/787/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/787/" class="post-title-link" itemprop="url">思考---基因组中那些“看起来”很严重的突变往往不致病</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-10-19 22:19:28" itemprop="dateCreated datePublished" datetime="2016-10-19T22:19:28+08:00">2016-10-19</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>在全基因组数据或者外显子数据得到的突变通过注释之后，每个人或多多少都含有一定数目的”看起来”很严重的突变，比如剪切位点上突变，终止密码子突变，移码突变等等（正如《ExAC-外显子组聚合数据库：揭露那些”披着狼皮的羊”基因》标题中的描述———披着狼皮的羊）。</p>
<p>但真实情况是，人并没有疾病表型，或者疾病只与其中的个别突变有关系，其他的突变都没有影响。这是我一直思考的问题，偶然机会，我向业内做遗传咨询的一个博士（感谢）提出这个问题，讨论之后认为主要有以下几点原因：</p>
<p><strong>一，</strong>很多基因不是必须基因，即使全部删除也不影响机体的正常功能。</p>
<p><strong>二，</strong>隐性基因即使有一份被破坏，不影响另外一份行使功能。</p>
<p><strong>三，</strong>如果参考的是文献中的致病突变，有很多是假的。也就说参考文献有可能存在假的结果。</p>
<p><strong>四，</strong>机体有很强的代偿能力，即使真的是有缺陷的基因，机体也有可能代偿其缺陷而不影响正常机体功能，比如有一些个体携带确定致病的基因突变，表型仍正常。</p>
<p>欢迎您留言补充。</p>
<p><strong>PS：</strong>虽然在以往的生物学研究中，我们发现了很多（经典案例的）携带有某个突变会导致疾病，但这种突变往往是罕见病的致病突变。也不绝对，可能有些人确实携带我们认为的致病突变，但由于机体的其他通路代偿功能，补偿了该突变造成的损害。因为机体是一个复杂的过程，复杂到现在我们都不清楚对生命的过程有多少了解。过往的经典研究案例越来越少，从复杂的整体层面来研究或许能够突破现在的瓶颈。</p>
<p>DNA双螺旋的发现是由于 物理和化学中X光射线衍射的介入。细胞微观结构的发现是由于 物理和电子中的电子显微镜的介入。每一次生物学的飞跃都与其他学科的介入，带来新的研究方式有关。</p>
<p>现在，正是这个时代，高通量测序技术和计算机技术的飞速发展，生物领域或许正在下一次飞跃的黎明。</p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/786/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/786/" class="post-title-link" itemprop="url">[转载]蛋白质组学及蛋白质鉴定常用的网站和数据库</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-10-18 23:29:11" itemprop="dateCreated datePublished" datetime="2016-10-18T23:29:11+08:00">2016-10-18</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>PS：我不经常用蛋白质数据库，不过看到这个总结的挺详细的，就转过来了。</p>
<h1 id="一、蛋白质数据库"><a href="#一、蛋白质数据库" class="headerlink" title="一、蛋白质数据库"></a>一、蛋白质数据库</h1><hr>
<h2 id="1-UniProt-The-Universal-Protein-Resource"><a href="#1-UniProt-The-Universal-Protein-Resource" class="headerlink" title="1.UniProt (The Universal Protein Resource)"></a>1.UniProt (The Universal Protein Resource)</h2><p>  网址：<a target="_blank" rel="noopener" href="http://www.uniprot.org/">http://www.uniprot.org/</a><br>        <a target="_blank" rel="noopener" href="http://www.ebi.ac.uk/uniprot/">http://www.ebi.ac.uk/uniprot/</a><br>  简介：由EBI(欧洲生物信息研究所)、PIR(蛋白信息资源)和SIB(瑞士生物信息研究所)合作建立而成，提供详细的蛋白质序列、功能信息，如蛋白质功能描述、结构域结构、转录后修饰、修饰位点、变异度、二级结构、三级结构等，同时提供其他数据库，包括序列数据库、三维结构数据库、2-D凝聚电泳数据库、蛋白质家族数据库的相应链接。</p>
<h2 id="2-PIR-Protein-Information-Resource"><a href="#2-PIR-Protein-Information-Resource" class="headerlink" title="2.PIR(Protein Information Resource)"></a>2.PIR(Protein Information Resource)</h2><p>  网址：<a target="_blank" rel="noopener" href="http://pir.georgetown.edu/">http://pir.georgetown.edu/</a><br>  简介：致力于提供及时的、高质量、最广泛的注释，其下的数据库有iProClass、PIRSF、PIR-PSD、PIR-NREF、UniPort，与90多个生物数据库(蛋白家族、蛋白质功能、蛋白质网络、蛋白质互作、基因组等数据库)存在着交叉应用。</p>
<h2 id="3-BRENDA-enzyme-database"><a href="#3-BRENDA-enzyme-database" class="headerlink" title="3.BRENDA(enzyme database)"></a>3.BRENDA(enzyme database)</h2><p>  网址：<a target="_blank" rel="noopener" href="http://www.brenda-enzymes.org/">http://www.brenda-enzymes.org</a><br>  简介：酶数据库，提供酶的分类、命名法、生化反应、专一性、结构、细胞定位、提取方法、文献、应用与改造及相关疾病的数据。</p>
<h2 id="4-CORUM-collection-of-experimentally-verified-mammalian-protein-complexes"><a href="#4-CORUM-collection-of-experimentally-verified-mammalian-protein-complexes" class="headerlink" title="4.CORUM(collection of experimentally verified mammalian protein complexes)"></a>4.CORUM(collection of experimentally verified mammalian protein complexes)</h2><p>  网址：<a target="_blank" rel="noopener" href="http://mips.gsf.de/genre/proj/corum/index.html">http://mips.gsf.de/genre/proj/corum/index.html</a><br>  简介：哺乳动物蛋白复合物数据库，提供的数据包括蛋白复合物名称、亚基、功能、相关文献等</p>
<h2 id="5-CyBase-cyclic-protein-database"><a href="#5-CyBase-cyclic-protein-database" class="headerlink" title="5.CyBase(cyclic protein database)"></a>5.CyBase(cyclic protein database)</h2><p>  网址：<a target="_blank" rel="noopener" href="http://research1t.imb.uq.edu.au/cybase">http://research1t.imb.uq.edu.au/cybase</a><br>  简介：环状蛋白数据库，提供环状蛋白的序列、结构等数据，提供环化蛋白预测服务。</p>
<h2 id="6-DB-PABP"><a href="#6-DB-PABP" class="headerlink" title="6.DB-PABP"></a>6.DB-PABP</h2><p>  网址：<a target="_blank" rel="noopener" href="http://pabp.bcf.ku.edu/DB_PABP/">http://pabp.bcf.ku.edu/DB_PABP/</a><br>  简介：聚阴离子结合蛋白数据库。聚阴离子结合蛋白与聚阴离子的互作在胞内定位、运输、蛋白质折叠等生命过程中起重要作用，此外许多与神经衰退疾病相关的蛋白质均为聚阴离子结合蛋白。该数据库提供已被鉴定的聚阴离子结合蛋白的数据，与NCBI蛋白数据库存在交叉应用。</p>
<h2 id="7-IUPHAR-DB"><a href="#7-IUPHAR-DB" class="headerlink" title="7.IUPHAR-DB"></a>7.IUPHAR-DB</h2><p>  网址：<a target="_blank" rel="noopener" href="http://www.iuphar-db.org/">http://www.iuphar-db.org</a><br>  简介：G蛋白偶联受体、离子通道数据库。提供这些蛋白的基因、功能、结构、配体、表达图谱、信号转导机制、多样性等数据。</p>
<h2 id="8-GLIDA"><a href="#8-GLIDA" class="headerlink" title="8.GLIDA"></a>8.GLIDA</h2><p>  网址：<a target="_blank" rel="noopener" href="http://pharminfo.pharm.kyoto-u.ac.jp/services/glida/">http://pharminfo.pharm.kyoto-u.ac.jp/services/glida/</a><br>　简介：G蛋白偶联受体-配体数据库，提供G蛋白偶联受体-配体互作数据、配体数据、G蛋白偶联受体数据、同源受体关系网、保守识别区，为新药发现提供了支持。</p>
<h2 id="9-LOCATE"><a href="#9-LOCATE" class="headerlink" title="9.LOCATE"></a>9.LOCATE</h2><p>  网址：<a target="_blank" rel="noopener" href="http://locate.imb.uq.edu.au/">http://locate.imb.uq.edu.au/</a><br>  简介：哺乳动物蛋白质亚细胞定位数据库</p>
<h2 id="10-InterPro"><a href="#10-InterPro" class="headerlink" title="10.InterPro"></a>10.InterPro</h2><p>  网址：<a target="_blank" rel="noopener" href="http://www.ebi.ac.uk/interpro/">http://www.ebi.ac.uk/interpro/</a><br>  简介：蛋白质综合数据库，从大量的数据库中整合而成的包括蛋白质结构域、蛋白质家族、功能位点等信息的数据库。</p>
<h2 id="11-OKCAM"><a href="#11-OKCAM" class="headerlink" title="11.OKCAM"></a>11.OKCAM</h2><p>  网址：<a target="_blank" rel="noopener" href="http://okcam.cbi.pku.edu.cn/">http://okcam.cbi.pku.edu.cn</a><br>  简介：人体细胞粘附分子数据库。</p>
<h1 id="二、蛋白质组数据库"><a href="#二、蛋白质组数据库" class="headerlink" title="二、蛋白质组数据库"></a>二、蛋白质组数据库</h1><hr>
<h2 id="1-GELBANK"><a href="#1-GELBANK" class="headerlink" title="1.GELBANK"></a>1.GELBANK</h2>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/786/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/785/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/785/" class="post-title-link" itemprop="url">Axel - Linux下多线程下载工具</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-09-29 12:02:13" itemprop="dateCreated datePublished" datetime="2016-09-29T12:02:13+08:00">2016-09-29</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>在linux环境下，用wget下载大文件，实在是件痛苦的事情，下载速度慢。这非常的不科学，于是找到了axel这个工具，可以实现在linux下多线程下载。并且可以实现断点续传。 Axel项目网站 <a target="_blank" rel="noopener" href="https://wilmer.gaa.st/main.php/axel.html">https://wilmer.gaa.st/main.php/axel.html</a></p>
<h1 id="安装"><a href="#安装" class="headerlink" title="安装"></a>安装</h1><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">wget -c https://wilmer.gaa.st/downloads/axel-1.0b.tar.gz</span><br><span class="line">tar zxvf axel-1.0b.tar.gz</span><br><span class="line"><span class="built_in">cd</span> axel-1.0b/</span><br><span class="line">./configure</span><br><span class="line">make</span><br><span class="line">make instal</span><br></pre></td></tr></table></figure>

<h1 id="或者"><a href="#或者" class="headerlink" title="或者"></a>或者</h1><p>apt-get install axel 参数 -n 指定线程数 -o 指定另存为目录 -s 指定每秒的最大比特数 -q 静默模式</p>
<h1 id="测试"><a href="#测试" class="headerlink" title="测试"></a>测试</h1><p>比如从UCSC上下载 938M 的参考基因组序列gz格式文件</p>
<p>axel -n 8 <a target="_blank" rel="noopener" href="http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz">http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz</a></p>
<p>Initializing download: <a target="_blank" rel="noopener" href="http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz">http://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz</a><br>File size: 983659424 bytes<br>Opening output file hg38.fa.gz<br>Starting download</p>
<p>[  0%]  ………. ………. ……….  [  41.3KB/s]<br>[  0%]  ………. ………. ……….  [ 179.5KB/s]<br>[  0%]  ………. ………. ……….  [ 423.4KB/s]<br>[  0%]  ………. ………. ……….  [ 583.9KB/s]<br>[  0%]  ………. ………. ……….  [ 772.0KB/s]<br>[  0%]  ………. ………. ……….  [ 801.8KB/s]<br>[  0%]  ………. ………. ……….  [ 802.1KB/s]<br>[  0%]  ………. ………. ……….  [ 813.1KB/s]<br>[  1%]  ………. ………. ……….  [ 843.5KB/s]<br>[ 99%]  ………. ………. ……….  [ 666.7KB/s]<br>[ 99%]  ………. ……<br>Connection 7 finished<br>        ,,,,,,,,,, ,,,,,,,,,, ,,,,,,….  [ 666.6KB/s]<br>[ 99%]  ………. ………. ……….  [ 666.7KB/s]<br>[ 99%]  ………. …….<br>Connection 2 finished<br>        ,,,,,,,,,, ,,,,,,,,,, ,,,,,,,…  [ 666.5KB/s]<br>[ 99%]  ………. ………. ……….  [ 663.1KB/s]<br>[ 99%]  ………. ………. ……….  [ 662.8KB/s]<br>[100%]  ….<br>Connection 0 finished</p>
<p>Downloaded 938.1 megabytes in 24:09 seconds. (662.81 KB/s)</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/770/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/770/" class="post-title-link" itemprop="url">利用genome music分析癌症样本中显著突变的基因和相关通路</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-09-28 20:41:56" itemprop="dateCreated datePublished" datetime="2016-09-28T20:41:56+08:00">2016-09-28</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>MuSiC   the Mutational Significance In Cancer (MuSiC) suite of tools<br>官网地址 <a target="_blank" rel="noopener" href="http://gmt.genome.wustl.edu/packages/genome-music/index.html">http://gmt.genome.wustl.edu/packages/genome-music/index.html</a></p>
<h1 id="功能其主要功能"><a href="#功能其主要功能" class="headerlink" title="功能其主要功能"></a>功能其主要功能</h1><ol>
<li>Apply statistical methods to identify significantly mutated genes</li>
<li>Highlight significantly altered pathways</li>
<li>Investigate the proximity of amino acid mutations in the same gene</li>
<li>Search for gene-based or site-based correlations to mutations and relationships between mutations themselves</li>
<li>Correlate mutations to clinical features, using typical correlation measures, and generalized linear models</li>
<li>Cross-reference findings with relevant databases such as Pfam, COSMIC, and OMIM</li>
</ol>
<h1 id="软件使用的分析思路"><a href="#软件使用的分析思路" class="headerlink" title="软件使用的分析思路"></a>软件使用的分析思路</h1><p>1，生成VCF文件<br>2，将VCF文件转换成maf（Mutation Annotation Format）文件<br>3，合并maf文件为一个文件<br>4，MuSiC bmr calc-covg 生成 <strong>covg</strong> 文件<br>5，MuSiC bmr calc-bmr 生成 <strong>gene_mrs</strong> 文件<br>6，MuSiC smg 生成 <strong>significantly mutated genes</strong><br>7，Music path-scan 生成 <strong>sm_pathways</strong></p>
<h1 id="安装"><a href="#安装" class="headerlink" title="安装"></a>安装</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">sudo apt-add-repository &quot;deb http://apt.genome.wustl.edu/ubuntu lucid-genome main&quot;</span><br><span class="line">wget https://apt.genome.wustl.edu/ubuntu/files/genome-institute.asc</span><br><span class="line">sudo apt-key add genome-institute.asc</span><br><span class="line">sudo apt-get update</span><br><span class="line"># fix dependency error Bit::Vector</span><br><span class="line">sudo cpan install Bit::Vector  </span><br><span class="line">sudo apt-get install genome-music</span><br></pre></td></tr></table></figure>




<h1 id="1，生成VCF文件"><a href="#1，生成VCF文件" class="headerlink" title="1，生成VCF文件"></a>1，生成VCF文件</h1><p>可以用varscan2，GATK muTect2等</p>
<h1 id="2，将VCF文件转换成maf"><a href="#2，将VCF文件转换成maf" class="headerlink" title="2，将VCF文件转换成maf"></a>2，将VCF文件转换成maf</h1><p>参见<a target="_blank" rel="noopener" href="http://wp.zxzyl.com/?p=273">前期文章提到的oncotator</a>，要注意通过设置 –annotate-manual=’Tumor_Sample_Barcode:Tumor_sample_name’ 设置癌症的样本名（对应第16列）</p>
<h1 id="3，合并maf文件"><a href="#3，合并maf文件" class="headerlink" title="3，合并maf文件"></a>3，合并maf文件</h1><p># 人懒，这个文件中间有maf的标题栏，没有去掉 ( ° △ °’’’)︴<br><code>cat *.maf &#39; grep -v &quot;^#&quot; &gt; myMAF.tsv</code></p>
<h1 id="4，MuSiC-bmr-calc-covg-生成-covg-文件"><a href="#4，MuSiC-bmr-calc-covg-生成-covg-文件" class="headerlink" title="4，MuSiC bmr calc-covg 生成 covg 文件"></a>4，MuSiC bmr calc-covg 生成 covg 文件</h1><p><strong>count covered bases per-gene for each given tumor-normal pair of BAMs.</strong>  </p>
<p>–bam-list        一个专门记录样本的文件，需要配对的bam(需要先index)，共三列，sample_name（样本名） normal_bam（正常组织bam文件） tumor_bam（肿瘤组织bam文件），每个配对样本的sample_name名称要和第二部设置的名称一致。<br>格式如下</p>
<p>aaa    /path/to/bam/aaa_normal.bam        /path/to/bam/aaa_tumor.bam<br>bbb    /path/to/bam/bbb_normal.bam        /path/to/bam/bbb_tumor.bam</p>
<p>–roi-file        The regions of interest (ROIs) of each gene are typically regions targeted for sequencing or are merged exon loci (from multiple transcripts) of genes with 2-bp flanks (splice junctions).<br>roi（基于1 base），可以从  <a target="_blank" rel="noopener" href="https://github.com/ding-lab/calc-roi-covg/blob/master/data/ensembl_67_cds_ncrna_and_splice_sites_hg19">https://github.com/ding-lab/calc-roi-covg/blob/master/data/ensembl_67_cds_ncrna_and_splice_sites_hg19</a> 下载到，格式如下，要注意染色体要与maf对应，比如都有chr三个字符。如果maf中的染色体带chr，可以awk ‘{print “chr”$0}’ old.roi &gt; new.chr.roi 。roi文件中的基因名要和maf中对应。</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">chr1\t11867\t12229\tDDX11L1</span><br><span class="line">chr1\t12611\t12723\tDDX11L1</span><br><span class="line">chr1\t29552\t30041\tMIR1302-11</span><br></pre></td></tr></table></figure>




<p>如何生成自己需要的roi文件，过几天我可能会讲。</p>
<p>运行命令</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">genome music bmr calc-covg </span><br><span class="line">    --bam-list bam_list  </span><br><span class="line">    --output-dir output_dir/   </span><br><span class="line">    --reference-sequence ref.fa  </span><br><span class="line">    --roi-file new.chr.roi </span><br></pre></td></tr></table></figure>




<p>文件夹结构</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">.</span><br><span class="line">├── gene_covgs</span><br><span class="line">│   ├── aaa.covg</span><br><span class="line">│   └── bbb.covg</span><br><span class="line">├── roi_covgs</span><br><span class="line">│   ├── aaa.covg</span><br><span class="line">│   └── bbb.covg</span><br><span class="line">└── total_covgs</span><br><span class="line">2 directories, 5 files</span><br></pre></td></tr></table></figure>



<h1 id="5，MuSiC-bmr-calc-bmr-生成-gene-mrs-文件"><a href="#5，MuSiC-bmr-calc-bmr-生成-gene-mrs-文件" class="headerlink" title="5，MuSiC bmr calc-bmr 生成 gene_mrs 文件"></a>5，MuSiC bmr calc-bmr 生成 gene_mrs 文件</h1>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/770/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/769/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/769/" class="post-title-link" itemprop="url">oncotator对VCF进行注释，并生成MAF格式文件</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-09-06 16:00:01" itemprop="dateCreated datePublished" datetime="2016-09-06T16:00:01+08:00">2016-09-06</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>MAF格式Mutation Annotation Format (MAF) ，是TCGA组织对突变进行注释的格式。一些和癌症分析相关的软件，经常要求MAF格式文件作为输入。而现在经过GATK或samtools检测出突变的格式一般为VCF格式，的注释软件，即使经过SNPEff和annovar注释（当然还有VEP），结果依然为VCF格式或者tab分割的文件等。</p>
<p>MAF中每一列是一种注释信息，由于包含的注释信息太多（<a target="_blank" rel="noopener" href="https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+%28MAF%29+Specification;WIKISESSIONID=52395E56D6223DF1AB5840164882D34A">详见格式</a>），单纯的通过写脚本转换SNPEff或者annovar的注释文件，会变得非常麻烦而且考虑的问题可能不完全（有人实现过，通过Ensembl的VEP对VCF注释，然后转换，可以在github上搜索到）。</p>
<p>这里介绍注释软件oncotator，可以注释VCF文件，并直接生成MAF格式，相当于将VCF格式转换成MAF格式。Broad institute开发的，用起来放心哈。</p>
<h2 id="首先安装python-virtualenv"><a href="#首先安装python-virtualenv" class="headerlink" title="首先安装python virtualenv"></a>首先安装python virtualenv</h2><p>virtualenv可以创建一个虚拟的python环境，在虚拟环境中安装的包，不影响机器已经安装的python包。</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">curl -O https://pypi.python.org/packages/<span class="built_in">source</span>/v/virtualenv/virtualenv-14.0.6.tar.gz</span><br><span class="line">tar -xvzf virtualenv-14.0.6.tar.gz </span><br><span class="line"><span class="built_in">cd</span> virtualenv-14.0.6/</span><br><span class="line">python setup.py install --prefix=~/pypackages/</span><br><span class="line"><span class="built_in">cd</span> ..</span><br><span class="line">rm -rf virtualenv-14.0.6*</span><br></pre></td></tr></table></figure>




<h2 id="安装oncotator依赖"><a href="#安装oncotator依赖" class="headerlink" title="安装oncotator依赖"></a>安装oncotator依赖</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">git clone  https://github.com/broadinstitute/oncotator.git</span><br><span class="line">cd oncotator</span><br><span class="line">mkdir env</span><br><span class="line">bash scripts/create_oncotator_venv.sh -e ./env</span><br></pre></td></tr></table></figure>


<h2 id="安装oncotator"><a href="#安装oncotator" class="headerlink" title="安装oncotator"></a>安装oncotator</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">source ./env/bin/activate</span><br><span class="line">python setup.py install</span><br><span class="line">deactivate</span><br></pre></td></tr></table></figure>




<h2 id="下载注释数据文件"><a href="#下载注释数据文件" class="headerlink" title="下载注释数据文件"></a>下载注释数据文件</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">wget -c http://www.broadinstitute.org/~lichtens/oncobeta/oncotator_v1_ds_Jan262015.tar.gz</span><br><span class="line">tar -xvzf  oncotator_v1_ds_Jan262015.tar.gz</span><br></pre></td></tr></table></figure>

<p>注释内容如下，包括基因组注释，蛋白注释，癌症突变注释等，用的基因组是GENCODE的hg19版本。<br><strong>Genomic Annotations</strong></p>
<p>Gene, transcript, and functional consequence annotations using GENCODE for hg19.<br>Reference sequence around a variant.<br>GC content around a variant.<br>Human DNA Repair Gene annotations from Wood et al.</p>
<p><strong>Protein Annotations</strong></p>
<p>Site-specific protein annotations from UniProt.<br>Functional impact predictions from dbNSFP.</p>
<p><strong>Cancer Variant Annotations</strong></p>
<p>Observed cancer mutation frequency annotations from COSMIC.<br>Cancer gene and mutation annotations from the Cancer GenCensus.<br>Overlapping mutations from the Cancer Cell Line Encyclopedia.<br>Cancer gene annotations from the Familial Cancer Database.<br>Cancer variant annotations from ClinVar.</p>
<p><strong>Non-Cancer Variant Annotations</strong></p>
<p>Common SNP annotations from dbSNP.<br>Variant annotations from 1000 Genomes.<br>Variant annotations from NHLBI GO Exome Sequencing Project (ESP).</p>
<h1 id="运行oncotator"><a href="#运行oncotator" class="headerlink" title="运行oncotator"></a>运行oncotator</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">#进入虚拟环境env</span><br><span class="line">source ./env/bin/activate</span><br><span class="line"># WGS = whole genome sequencing, WXS = whole exome sequencing</span><br><span class="line"># 通过annotate-manual设置MAF注释，比如样本barcode，样本来源等，至于氨基酸突变等，会自动进行注释</span><br><span class="line">oncotator --input_format=VCF --db-dir=./oncotator_v1_ds_Jan262014 --output_format=TCGAMAF --tx-mode=EFFECT --collapse-number-annotations --annotate-default=&#x27;Sequencer:Illumina HiSeq&#x27; --annotate-manual=&#x27;Sequence_Source:WGS&#x27; --annotate-manual=&#x27;Tumor_Sample_Barcode:Tumor&#x27;  --annotate-manual=&#x27;Matched_Norm_Sample_Barcode:normal&#x27;  input.vcf output.maf hg19</span><br><span class="line">deactivate</span><br></pre></td></tr></table></figure>




<h1 id="结果"><a href="#结果" class="headerlink" title="结果"></a>结果</h1><p><img src="/wp/f4w/2020/2016-09-06-oncotator-annotation-maf.png"></p>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/769/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/768/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/768/" class="post-title-link" itemprop="url">Convert gtf to bed12 format --- gtf2bed</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-08-30 21:29:32" itemprop="dateCreated datePublished" datetime="2016-08-30T21:29:32+08:00">2016-08-30</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>Some software requires bed12 format, not gtf/gff. So a convertion work should be done.</p>
<h1 id="bedops-gtf2bed"><a href="#bedops-gtf2bed" class="headerlink" title="bedops gtf2bed"></a>bedops gtf2bed</h1><p>Easy way to get your result —- <a target="_blank" rel="noopener" href="http://bedops.readthedocs.io/">bedops</a></p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">gtf2bed &lt; Homo_sapiens.GRCh38.85.gtf &#x27; head</span><br><span class="line">1       11868   14409   ENST00000456328 0       +       11868   14409   0       3       359,109,1189,   0,744,1352,</span><br><span class="line">1       12009   13670   ENST00000450305 0       +       12009   13670   0       6       48,49,85,78,154,218,    0,169,603,965,1211,1443,</span><br><span class="line">1       17368   17436   ENST00000619216 0       -       17368   17436   0       1       68,     0,</span><br><span class="line">1       14403   29570   ENST00000488147 0       -       14403   29570   0       11      98,34,152,159,198,136,137,147,99,154,37,        0,601,1392,2203,2454,2829,3202,3511,3864,10334,15130,</span><br><span class="line">1       29553   31097   ENST00000473358 0       +       29553   31097   0       3       486,104,122,    0,1010,1422,</span><br><span class="line">1       30266   31109   ENST00000469289 0       +       30266   31109   0       2       401,134,        0,709,</span><br><span class="line">1       30365   30503   ENST00000607096 0       +       30365   30503   0       1       138,    0,</span><br><span class="line">1       34553   36081   ENST00000417324 0       -       34553   36081   0       3       621,205,361,    0,723,1167,</span><br><span class="line">1       35244   36073   ENST00000461467 0       -       35244   36073   0       2       237,353,        0,476,</span><br><span class="line">1       52472   53312   ENST00000606857 0       +       52472   53312   0       1       840,    0,</span><br></pre></td></tr></table></figure>


<p>Here we use UCSC’s package to convert gtf format file to bed12 format file.</p>
<h1 id="Download-UCSC-package"><a href="#Download-UCSC-package" class="headerlink" title="Download UCSC package"></a>Download UCSC package</h1><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">wget -c http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/gtfToGenePred</span><br><span class="line">wget -c http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/genePredToBed</span><br><span class="line">chmod 755 gtfToGenePred genePredToBed</span><br></pre></td></tr></table></figure>




<h2 id="Convert-gtf-to-bed12"><a href="#Convert-gtf-to-bed12" class="headerlink" title="Convert gtf to bed12"></a>Convert gtf to bed12</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">time gtfToGenePred Homo_sapiens.GRCh38.85.gtf test.genePhred &amp;&amp; genePredToBed test.genePhred result.bed &amp;&amp; rm test.genePhred </span><br><span class="line"></span><br><span class="line">real    0m14.959s</span><br><span class="line">user    0m14.387s</span><br><span class="line">sys     0m0.567s</span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">head reult.bed </span><br><span class="line">1       11868   14409   ENST00000456328 0       +       14409   14409   0       3       359,109,1189,   0,744,1352,</span><br><span class="line">1       12009   13670   ENST00000450305 0       +       13670   13670   0       6       48,49,85,78,154,218,    0,169,603,965,1211,1443,</span><br><span class="line">1       14403   29570   ENST00000488147 0       -       29570   29570   0       11      98,34,152,159,198,136,137,147,99,154,37,       0,601,1392,2203,2454,2829,3202,3511,3864,10334,15130,</span><br><span class="line">1       17368   17436   ENST00000619216 0       -       17436   17436   0       1       68,     0,</span><br><span class="line">1       29553   31097   ENST00000473358 0       +       31097   31097   0       3       486,104,122,    0,1010,1422,</span><br><span class="line">1       30266   31109   ENST00000469289 0       +       31109   31109   0       2       401,134,        0,709,</span><br><span class="line">1       30365   30503   ENST00000607096 0       +       30503   30503   0       1       138,    0,</span><br><span class="line">1       34553   36081   ENST00000417324 0       -       36081   36081   0       3       621,205,361,    0,723,1167,</span><br><span class="line">1       35244   36073   ENST00000461467 0       -       36073   36073   0       2       237,353,        0,476,</span><br><span class="line">1       52472   53312   ENST00000606857 0       +       53312   53312   0       1       840,    0,</span><br></pre></td></tr></table></figure>


<p>Cheers. </p>
<h1 id="Another-tool"><a href="#Another-tool" class="headerlink" title="Another tool"></a>Another tool</h1><p>I also recommend a perl script – <a target="_blank" rel="noopener" href="https://github.com/ExpressionAnalysis/ea-utils/blob/master/clipper/gtf2bed">gtf2bed</a>. This script works fine to convert gtf to bed format. Thanks for ea-utils project.</p>
<h1 id="Ref"><a href="#Ref" class="headerlink" title="Ref"></a>Ref</h1><p><a target="_blank" rel="noopener" href="https://groups.google.com/a/soe.ucsc.edu/forum/#!topic/genome/fTNuZOS-CYA">https://groups.google.com/a/soe.ucsc.edu/forum/#!topic/genome/fTNuZOS-CYA</a></p>
<p><a target="_blank" rel="noopener" href="https://expressionanalysis.github.io/ea-utils/">https://expressionanalysis.github.io/ea-utils/</a></p>
<p><a target="_blank" rel="noopener" href="http://bedops.readthedocs.io/">bedops.readthedocs.io</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/767/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/767/" class="post-title-link" itemprop="url">topGO包进行-GO富集分析-做有向无环图</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-08-16 16:23:15" itemprop="dateCreated datePublished" datetime="2016-08-16T16:23:15+08:00">2016-08-16</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>GOdata = new(“topGOdata”, ontology = “MF”, allGenes = geneList,annot = annFUN.gene2GO, gene2GO = geneID2GO)</p>
<p>利用topGO进行分析，最重要的是构建topGO对象，构建topGO需要两个参数：</p>
<p>1，topGO需要基因和GO号的对应关系</p>
<p>2，基因列表，用来标记背景基因（所有基因）及差异基因</p>
<h1 id="一，获取ensembl和GO号的对应关系-geneID2GO"><a href="#一，获取ensembl和GO号的对应关系-geneID2GO" class="headerlink" title="一，获取ensembl和GO号的对应关系: geneID2GO"></a>一，获取ensembl和GO号的对应关系: geneID2GO</h1><p>如果你有现成的gene id和go id的对应关系，文件格式为</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gene_ID制表符GO_ID1, GO_ID2, GO_ID3, ...</span><br></pre></td></tr></table></figure>

<p>每行，则可以利用readMappings读取该文件（topGO包里面的函数）。</p>
<p>而我有一个cuffdiff的结果文件gene_exp.diff，用的是ensembl的注释GFF文件。首先需要获得ensembl和GO的对应关系，这里利用biomaRt包（前面的<a target="_blank" rel="noopener" href="http://wp.zxzyl.com/?p=182">文章</a>中有讲，这里不详细介绍）。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">library(biomaRt)</span><br><span class="line">genes = useEnsembl(biomart=<span class="string">&quot;ensembl&quot;</span>,dataset=<span class="string">&quot;hsapiens_gene_ensembl&quot;</span>)</span><br><span class="line"><span class="comment"># 得到go信息和gene</span></span><br><span class="line">gene2goInfo &lt;- getBM(<span class="built_in">attributes</span>=<span class="built_in">c</span>(<span class="string">&#x27;ensembl_gene_id&#x27;</span>,<span class="string">&#x27;go_id&#x27;</span>,<span class="string">&#x27;entrezgene&#x27;</span>,<span class="string">&#x27;name_1006&#x27;</span>,<span class="string">&#x27;go_linkage_type&#x27;</span>,<span class="string">&#x27;namespace_1003&#x27;</span>), mart = genes)</span><br><span class="line"><span class="comment"># 过滤</span></span><br><span class="line">gene2goInfo=gene2goInfo[gene2goInfo$go_id != <span class="string">&quot;&quot;</span>, ]</span><br><span class="line"><span class="comment"># 上述是一个gene对应一个go id，需要合并为一个gene对应多个go id，利用by函数（神器）</span></span><br><span class="line">geneID2GO = by(gene2goInfo$go_id, gene2goInfo$ensembl_gene_id, <span class="keyword">function</span>(x) <span class="built_in">as.character</span>(x))</span><br></pre></td></tr></table></figure>



<p>geneID2GO是符合topGO要求的，gene2goInfo和geneID2GO的格式如下</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br></pre></td><td class="code"><pre><span class="line">&gt; head(gene2goInfo)</span><br><span class="line">  ensembl_gene_id      go_id entrezgene                         name_1006</span><br><span class="line"><span class="number">3</span> ENSG00000281614 GO:<span class="number">0005886</span>       <span class="number">3635</span>                   plasma membrane</span><br><span class="line"><span class="number">4</span> ENSG00000281614 GO:<span class="number">0005829</span>       <span class="number">3635</span>                           cytosol</span><br><span class="line"><span class="number">5</span> ENSG00000281614 GO:<span class="number">0005515</span>       <span class="number">3635</span>                   protein binding</span><br><span class="line"><span class="number">6</span> ENSG00000281614 GO:<span class="number">0007165</span>       <span class="number">3635</span>               signal transduction</span><br><span class="line"><span class="number">7</span> ENSG00000281614 GO:<span class="number">0005856</span>       <span class="number">3635</span>                      cytoskeleton</span><br><span class="line"><span class="number">8</span> ENSG00000281614 GO:<span class="number">0050852</span>       <span class="number">3635</span> <span class="built_in">T</span> cell receptor signaling pathway</span><br><span class="line">  go_linkage_type     namespace_1003</span><br><span class="line"><span class="number">3</span>             IEA cellular_component</span><br><span class="line"><span class="number">4</span>             TAS cellular_component</span><br><span class="line"><span class="number">5</span>             IPI molecular_function</span><br><span class="line"><span class="number">6</span>             TAS biological_process</span><br><span class="line"><span class="number">7</span>             IEA cellular_component</span><br><span class="line"><span class="number">8</span>             TAS biological_process</span><br><span class="line"></span><br><span class="line">&gt; head(geneID2GO)</span><br><span class="line">$ENSG00000000003</span><br><span class="line">[<span class="number">1</span>] <span class="string">&quot;GO:0004871&quot;</span> <span class="string">&quot;GO:0005515&quot;</span> <span class="string">&quot;GO:0005887&quot;</span> <span class="string">&quot;GO:0070062&quot;</span> <span class="string">&quot;GO:0007166&quot;</span></span><br><span class="line">[<span class="number">6</span>] <span class="string">&quot;GO:0043123&quot;</span> <span class="string">&quot;GO:0039532&quot;</span> <span class="string">&quot;GO:1901223&quot;</span></span><br><span class="line"></span><br><span class="line">$ENSG00000000005</span><br><span class="line">[<span class="number">1</span>] <span class="string">&quot;GO:0016021&quot;</span> <span class="string">&quot;GO:0005737&quot;</span> <span class="string">&quot;GO:0005515&quot;</span> <span class="string">&quot;GO:0005635&quot;</span> <span class="string">&quot;GO:0016525&quot;</span></span><br><span class="line">[<span class="number">6</span>] <span class="string">&quot;GO:0001886&quot;</span> <span class="string">&quot;GO:0001937&quot;</span> <span class="string">&quot;GO:0071773&quot;</span> <span class="string">&quot;GO:0035990&quot;</span></span><br></pre></td></tr></table></figure>

<h1 id="二，构建基因列表：geneList"><a href="#二，构建基因列表：geneList" class="headerlink" title="二，构建基因列表：geneList"></a>二，构建基因列表：geneList</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">diff=read.table(&quot;./gene_exp.diff&quot;,sep=&quot;t&quot;,header=TRUE)</span><br><span class="line"># 差异表达基因</span><br><span class="line">interesting_genes=factor(diff[diff$significant==&quot;yes&quot;,]$gene_id)</span><br><span class="line"># 所有基因</span><br><span class="line">all_genes &lt;- diff$gene_id</span><br><span class="line"># 构建基因列表</span><br><span class="line">geneList &lt;- factor(as.integer (all_genes %in% interesting_genes))</span><br><span class="line">names(geneList)=all_genes</span><br></pre></td></tr></table></figure>


<h1 id="三，构建topGO对象"><a href="#三，构建topGO对象" class="headerlink" title="三，构建topGO对象"></a>三，构建topGO对象</h1><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">source(&quot;https://bioconductor.org/biocLite.R&quot;)</span><br><span class="line">chooseBioCmirror()</span><br><span class="line">biocLite(&quot;topGO&quot;)</span><br><span class="line">biocLite(&quot;org.Hs.eg.db&quot;)</span><br><span class="line">biocLite(&quot;GO.db&quot;)</span><br><span class="line">biocLite(&quot;Rgraphviz&quot;)</span><br><span class="line">library(topGO)</span><br><span class="line"></span><br><span class="line">GOdata &lt;- new(&quot;topGOdata&quot;, ontology = &quot;MF&quot;, allGenes = geneList,annot = annFUN.gene2GO, gene2GO = geneID2GO)</span><br><span class="line"></span><br><span class="line">Building most specific GOs .....</span><br><span class="line">( 3893 GO terms found. )</span><br><span class="line"></span><br><span class="line">Build GO DAG topology ..........</span><br><span class="line">( 4356 GO terms and 5490 relations. )</span><br><span class="line"></span><br><span class="line">Annotating nodes ...............</span><br><span class="line">( 16722 genes annotated to the GO terms. )</span><br></pre></td></tr></table></figure>

<h1 id="四，基因富集分析"><a href="#四，基因富集分析" class="headerlink" title="四，基因富集分析"></a>四，基因富集分析</h1><p>Fisher检验，通过2*2的列联表的进行计算。当然还有其他检验，比如KS，KS.elim。</p>
<p>—————-非差异表达基因—差异表达基因<br>注释到A通路——– 20 ———– 50<br>没有注释到A通路—- 1870 ——— 80</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">resultFisher &lt;- runTest(GOdata, algorithm = <span class="string">&quot;classic&quot;</span>, statistic = <span class="string">&quot;fisher&quot;</span>)</span><br><span class="line">allRes &lt;- GenTable(GOdata,classicFisher = resultFisher, orderBy = <span class="string">&quot;classicFisher&quot;</span>, topNodes = <span class="number">10</span>)</span><br></pre></td></tr></table></figure>


<p>根据orderBy的参数对象进行排序，得到结果如下</p>
<p>GO.ID                                        Term Annotated Significant<br>1 GO:0032794           GTPase activating protein binding        14           2<br>2 GO:0015278            calcium-release channel activity        16           2<br>3 GO:0099604       ligand-gated calcium channel activity        16           2<br>4 GO:0005527                           macrolide binding        19           2<br>5 GO:0005528                               FK506 binding        19           2<br>6 GO:0001225 RNA polymerase II transcription coactiva…         1           1</p>
<p>Expected classicFisher<br>1     0.05        0.0012<br>2     0.06        0.0016<br>3     0.06        0.0016<br>4     0.07        0.0023<br>5     0.07        0.0023<br>6     0.00        0.0038</p>
<h1 id="五，生成有向无环图-directed-acycline-praph，DAG）"><a href="#五，生成有向无环图-directed-acycline-praph，DAG）" class="headerlink" title="五，生成有向无环图(directed acycline praph，DAG）"></a>五，生成有向无环图(directed acycline praph，DAG）</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">svg(<span class="string">&quot;go.dag.svg&quot;</span>)</span><br><span class="line">showSigOfNodes(GOdata, score(resultFisher), firstSigNodes = <span class="number">5</span>, useInfo = <span class="string">&#x27;all&#x27;</span>)</span><br><span class="line">dev.off()</span><br></pre></td></tr></table></figure>




<p><img src="/wp/f4w/2020/2016-08-16-go.dag.png">  </p>
          <!--noindex-->
            <div class="post-button">
              <a class="btn" href="/archives/767/#more" rel="contents">
                Read more &raquo;
              </a>
            </div>
          <!--/noindex-->
        
      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  

      
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/766/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h2 class="post-title" itemprop="name headline">
          
            <a href="/archives/766/" class="post-title-link" itemprop="url">Ubuntu下安装java和多版本java共存控制</a>
        </h2>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2016-08-06 12:22:27" itemprop="dateCreated datePublished" datetime="2016-08-06T12:22:27+08:00">2016-08-06</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
          <p>教你通过命令jdk7，jdk8就可以优雅的切换java版本</p>
<p>我在机器上已有java7（java1.7），目前而言java7应该是用的最广泛的，被java8取代还需要一段时间。不过我遇到最新版的picard要求java8版本，才遇到了安装新版java的问题，并且我不想删掉老版本java，我希望很方便的调用。</p>
<p>我查找的方法介绍，大部分都是通过update-alternatives –config java来选择，个人不喜欢这种方法。下面介绍一种比较优雅的方法，通过一个命令就能切换java版本。</p>
<h1 id="下载jdk"><a href="#下载jdk" class="headerlink" title="下载jdk"></a>下载jdk</h1><p>下载页面  <a target="_blank" rel="noopener" href="http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html">http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html</a></p>
<p>选在对应的版本，如果版本不对应，在运行java时会报 <strong>no such file or directory</strong>。</p>
<p>一般放在  /usr/jvm/jdk   /opt/Java/jdk/   /usr/lib/jvm _等目录下，当然也可以放在其他地方。</p>
<p><strong>解压缩下载后的文件</strong></p>
<h1 id="优雅的控制java版本"><a href="#优雅的控制java版本" class="headerlink" title="优雅的控制java版本"></a>优雅的控制java版本</h1><p><strong>编辑<del>/.profile或者</del>/.bashrc</strong></p>
<h2 id="设置两个版本的路径"><a href="#设置两个版本的路径" class="headerlink" title="设置两个版本的路径"></a>设置两个版本的路径</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">export JAVA7_HOME=&#x27;/path/to/jdk1.7.**&#x27;</span><br><span class="line">export JAVA8_HOME=&#x27;/path/to/jdk1.8.***&#x27;</span><br></pre></td></tr></table></figure>


<h2 id="设置默认java版本"><a href="#设置默认java版本" class="headerlink" title="设置默认java版本"></a>设置默认java版本</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">export JAVA_HOME=$JAVA7_HOME</span><br><span class="line">export JRE_HOME=$JAVA_HOME/jre</span><br><span class="line">export CLASSPATH=.:$JAVA_HOME/lib:$JRE_HOME/lib:$CLASSPATH</span><br><span class="line">export PATH=$JAVA_HOME/bin:$JRE_HOME/bin:$PATH</span><br></pre></td></tr></table></figure>


<h2 id="转换版本命令"><a href="#转换版本命令" class="headerlink" title="转换版本命令"></a>转换版本命令</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">alias jdk8=&#x27;export JAVA_HOME=$JAVA8_HOME &amp;&amp; export JRE_HOME=$JAVA_HOME/jre &amp;&amp; export CLASSPATH=.:$JAVA_HOME/lib:$JRE_HOME/lib:$CLASSPATH &amp;&amp; export PATH=$JAVA_HOME/bin:$JRE_HOME/bin:$PATH&#x27;</span><br><span class="line">alias jdk7=&#x27;export JAVA_HOME=$JAVA7_HOME &amp;&amp; export JRE_HOME=$JAVA_HOME/jre &amp;&amp; export CLASSPATH=.:$JAVA_HOME/lib:$JRE_HOME/lib:$CLASSPATH &amp;&amp; export PATH=$JAVA_HOME/bin:$JRE_HOME/bin:$PATH&#x27;</span><br></pre></td></tr></table></figure>


<p>还有一步 source ~/.profile或者source ~/.bashrc</p>
<p>这样就可以通过jdk7命令切换到java7版本，通过jdk8命令切换到java8版本。默认是java7版本。这样可以避免通过 update-java-alternatives 命令来配置和控制java版本了。</p>
<h1 id="参考"><a href="#参考" class="headerlink" title="参考"></a>参考</h1><p><a target="_blank" rel="noopener" href="http://stackoverflow.com/questions/26252591/mac-os-x-and-multiple-java-versions">http://stackoverflow.com/questions/26252591/mac-os-x-and-multiple-java-versions</a></p>
<p><a target="_blank" rel="noopener" href="http://jingyan.baidu.com/article/647f0115bb26817f2048a871.html">http://jingyan.baidu.com/article/647f0115bb26817f2048a871.html</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>####################################################################</p>

      
    </div>

    
    
    
      <footer class="post-footer">
        <div class="post-eof"></div>
      </footer>
  </article>
  
  
  


  
  <nav class="pagination">
    <a class="extend prev" rel="prev" href="/page/8/"><i class="fa fa-angle-left" aria-label="Previous page"></i></a><a class="page-number" href="/">1</a><span class="space">&hellip;</span><a class="page-number" href="/page/8/">8</a><span class="page-number current">9</span><a class="page-number" href="/page/10/">10</a><span class="space">&hellip;</span><a class="page-number" href="/page/17/">17</a><a class="extend next" rel="next" href="/page/10/"><i class="fa fa-angle-right" aria-label="Next page"></i></a>
  </nav>



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Jason</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">163</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">167</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>


<!--google_adsense-->

  <div class="links-of-blogroll-title">
    <i class="fa fa-google"></i> AdSense
    <ins class="adsbygoogle"
        style="display:block"
        data-ad-client="ca-pub-9370593480269435"
        data-ad-slot="8824957305"
        data-ad-format="auto"
        data-full-width-responsive="true"></ins>
    <script>
      (adsbygoogle = window.adsbygoogle || []).push({});
    </script>
  </div>

<!--/google_adsense-->



    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Jason</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

<script data-ad-client="ca-pub-9370593480269435" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<!-- PGAD -->
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-9370593480269435"
     data-ad-slot="8824957305"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
