<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"md.zxzyl.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>


  <meta name="description" content="I always use gtf file and retrieve gene information. There isn’t a highly flexible tool to solve my demand. I modified the code from “https:&#x2F;&#x2F;github.com&#x2F;Jverma&#x2F;GFF-Parser”, thanks Jverma. This tool wi">
<meta property="og:type" content="article">
<meta property="og:title" content="Parse gtf">
<meta property="og:url" content="https://md.zxzyl.com/archives/1171/index.html">
<meta property="og:site_name" content="Zhongxu&#39;s blog">
<meta property="og:description" content="I always use gtf file and retrieve gene information. There isn’t a highly flexible tool to solve my demand. I modified the code from “https:&#x2F;&#x2F;github.com&#x2F;Jverma&#x2F;GFF-Parser”, thanks Jverma. This tool wi">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="https://md.zxzyl.com/wp/f4w/2020/2020-10-02-gtf-parser.gif">
<meta property="article:published_time" content="2020-09-30T17:03:23.000Z">
<meta property="article:modified_time" content="2020-09-30T17:03:23.000Z">
<meta property="article:author" content="Jason">
<meta property="article:tag" content="Python">
<meta property="article:tag" content="gtf">
<meta property="article:tag" content="code">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://md.zxzyl.com/wp/f4w/2020/2020-10-02-gtf-parser.gif">

<link rel="canonical" href="https://md.zxzyl.com/archives/1171/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>Parse gtf | Zhongxu's blog</title>
  


  <script>
    var _hmt = _hmt || [];
    (function() {
      var hm = document.createElement("script");
      hm.src = "https://hm.baidu.com/hm.js?a3b5d3f466edb3e3f15e8238d8f04be8";
      var s = document.getElementsByTagName("script")[0];
      s.parentNode.insertBefore(hm, s);
    })();
  </script>




  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">Zhongxu's blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="https://md.zxzyl.com/archives/1171/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Jason">
      <meta itemprop="description" content="">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="Zhongxu's blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          Parse gtf
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2020-10-01 01:03:23" itemprop="dateCreated datePublished" datetime="2020-10-01T01:03:23+08:00">2020-10-01</time>
            </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/Default-Category/" itemprop="url" rel="index"><span itemprop="name">Default Category</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>I always use gtf file and retrieve gene information. There isn’t a highly flexible tool to solve my demand. I modified the code from “<a target="_blank" rel="noopener" href="https://github.com/Jverma/GFF-Parser%E2%80%9D">https://github.com/Jverma/GFF-Parser”</a>, thanks Jverma. This tool will be easier to use.</p>
<p><img src="/wp/f4w/2020/2020-10-02-gtf-parser.gif"></p>
<h1 id="Usage"><a href="#Usage" class="headerlink" title="Usage"></a>Usage</h1><p>Basically, there are three parameters.</p>
<p>id: either transcript id or gene id.</p>
<p>attType: attribute defined in gtf file. E.g. feature (column 3), gene_name, exon_number, transcript_id in column 9</p>
<p>attValue: the attribute value you want to search for.</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">&gt;&gt;&gt; </span><span class="keyword">import</span> sys</span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span><span class="keyword">from</span> gtfParser <span class="keyword">import</span> gtfParser</span><br><span class="line"></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span>gtf = gtfParser(<span class="string">&quot;example.gtf&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span><span class="comment"># Get all exons in CDK7</span></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span>gtf.getRecordsByID(<span class="string">&quot;CDK7&quot;</span>, <span class="string">&quot;feature&quot;</span>, <span class="string">&quot;exon&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span><span class="comment"># Get all features of transcript_id defined as &quot;NM_001324069&quot; in gene &quot;CDK7&quot;</span></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span>gtf.getRecordsByID(<span class="string">&quot;CDK7&quot;</span>, <span class="string">&quot;transcript_id&quot;</span>, <span class="string">&quot;NM_001324069&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span><span class="comment"># Get start codon where feature was defined as &quot;start_codon&quot; in transcript &quot;NM_001324069&quot;</span></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span>gtf.getRecordsByID(<span class="string">&quot;NM_001324069&quot;</span>, <span class="string">&quot;feature&quot;</span>, <span class="string">&quot;start_codon&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span><span class="comment"># Get a exon where its id is &quot;NM_001324078.1&quot; in &quot;NM_001324078&quot; transcript</span></span><br><span class="line"><span class="meta">&gt;&gt;&gt; </span>gtf.getRecordsByID(<span class="string">&quot;NM_001324078&quot;</span>, <span class="string">&quot;exon_id&quot;</span>, <span class="string">&quot;NM_001324078.1&quot;</span>)</span><br></pre></td></tr></table></figure>



<span id="more"></span>

<h1 id="Example-gtf"><a href="#Example-gtf" class="headerlink" title="Example gtf"></a>Example gtf</h1><p>Here is an simple example of gtf file. You can use to test. A subset from refSeq.hg38.gtf.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br></pre></td><td class="code"><pre><span class="line">chr5	hg38.refGene.ucsc	transcript	69234886	69277430	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;;  gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69234886	69235041	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;1&quot;; exon_id &quot;NM_001324069.1&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69235394	69235453	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;2&quot;; exon_id &quot;NM_001324069.2&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69250745	69250831	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;3&quot;; exon_id &quot;NM_001324069.3&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69250829	69250831	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;3&quot;; exon_id &quot;NM_001324069.3&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69252418	69252451	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;4&quot;; exon_id &quot;NM_001324069.4&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69252418	69252451	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;4&quot;; exon_id &quot;NM_001324069.4&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69254602	69254669	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;5&quot;; exon_id &quot;NM_001324069.5&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69254602	69254669	.	+	2	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;5&quot;; exon_id &quot;NM_001324069.5&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69255460	69255528	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;6&quot;; exon_id &quot;NM_001324069.6&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69255460	69255528	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;6&quot;; exon_id &quot;NM_001324069.6&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69258043	69258153	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;7&quot;; exon_id &quot;NM_001324069.7&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69258043	69258153	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;7&quot;; exon_id &quot;NM_001324069.7&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69259818	69259936	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;8&quot;; exon_id &quot;NM_001324069.8&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69259818	69259936	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;8&quot;; exon_id &quot;NM_001324069.8&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69262205	69262304	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;9&quot;; exon_id &quot;NM_001324069.9&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69262205	69262304	.	+	1	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;9&quot;; exon_id &quot;NM_001324069.9&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69269207	69269293	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;10&quot;; exon_id &quot;NM_001324069.10&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69269207	69269293	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;10&quot;; exon_id &quot;NM_001324069.10&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69272892	69273041	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;11&quot;; exon_id &quot;NM_001324069.11&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69272892	69273041	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;11&quot;; exon_id &quot;NM_001324069.11&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69276543	69276690	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;12&quot;; exon_id &quot;NM_001324069.12&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69276543	69276690	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;12&quot;; exon_id &quot;NM_001324069.12&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69277107	69277430	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;13&quot;; exon_id &quot;NM_001324069.13&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69277107	69277132	.	+	2	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;13&quot;; exon_id &quot;NM_001324069.13&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	start_codon	69250829	69250831	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;3&quot;; exon_id &quot;NM_001324069.3&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	stop_codon	69277133	69277135	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324069&quot;; exon_number &quot;13&quot;; exon_id &quot;NM_001324069.13&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	transcript	69234886	69277430	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;;  gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69234886	69235041	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;1&quot;; exon_id &quot;NM_001324078.1&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69235394	69235453	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;2&quot;; exon_id &quot;NM_001324078.2&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69254602	69254667	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;3&quot;; exon_id &quot;NM_001324078.3&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69255460	69255528	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;4&quot;; exon_id &quot;NM_001324078.4&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69255511	69255528	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;4&quot;; exon_id &quot;NM_001324078.4&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69258043	69258153	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;5&quot;; exon_id &quot;NM_001324078.5&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69258043	69258153	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;5&quot;; exon_id &quot;NM_001324078.5&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69259818	69259936	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;6&quot;; exon_id &quot;NM_001324078.6&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69259818	69259936	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;6&quot;; exon_id &quot;NM_001324078.6&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69262205	69262304	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;7&quot;; exon_id &quot;NM_001324078.7&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69262205	69262304	.	+	1	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;7&quot;; exon_id &quot;NM_001324078.7&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69269207	69269293	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;8&quot;; exon_id &quot;NM_001324078.8&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69269207	69269293	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;8&quot;; exon_id &quot;NM_001324078.8&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69272892	69273041	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;9&quot;; exon_id &quot;NM_001324078.9&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69272892	69273041	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;9&quot;; exon_id &quot;NM_001324078.9&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69276543	69276690	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;10&quot;; exon_id &quot;NM_001324078.10&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69276543	69276690	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;10&quot;; exon_id &quot;NM_001324078.10&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	exon	69277107	69277430	.	+	.	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;11&quot;; exon_id &quot;NM_001324078.11&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	CDS	69277107	69277132	.	+	2	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;11&quot;; exon_id &quot;NM_001324078.11&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	start_codon	69255511	69255513	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;4&quot;; exon_id &quot;NM_001324078.4&quot;; gene_name &quot;CDK7&quot;;</span><br><span class="line">chr5	hg38.refGene.ucsc	stop_codon	69277133	69277135	.	+	0	gene_id &quot;CDK7&quot;; transcript_id &quot;NM_001324078&quot;; exon_number &quot;11&quot;; exon_id &quot;NM_001324078.11&quot;; gene_name &quot;CDK7&quot;;</span><br></pre></td></tr></table></figure>

<h1 id="My-code"><a href="#My-code" class="headerlink" title="My code"></a>My code</h1><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!/usr/bin/env python</span></span><br><span class="line"><span class="comment"># -*- coding: utf-8 -*-</span></span><br><span class="line"><span class="comment">#############################################################</span></span><br><span class="line"><span class="comment">###     Inspired by https://github.com/Jverma/GFF-Parser  ###</span></span><br><span class="line"><span class="comment">#############################################################</span></span><br><span class="line"></span><br><span class="line"><span class="comment">####################################################################</span></span><br><span class="line"><span class="comment"># &gt;&gt;&gt; from gtfParser import gtfParser</span></span><br><span class="line"><span class="comment"># &gt;&gt;&gt; gtf = gtfParser(input_file)</span></span><br><span class="line"><span class="comment">#</span></span><br><span class="line"><span class="comment"># &gt;&gt;&gt; gtf.getRecordsByID(&quot;TTN&quot;, &quot;feature&quot;, &quot;exon&quot;)</span></span><br><span class="line"><span class="comment"># &gt;&gt;&gt; gtf.getRecordsByID(&quot;TTN&quot;, &quot;transcript_id&quot;, &quot;NM_001367552&quot;)</span></span><br><span class="line"><span class="comment"># &gt;&gt;&gt; gtf.getRecordsByID(&quot;NM_013416&quot;, &quot;feature&quot;, &quot;start_codon&quot;)</span></span><br><span class="line"><span class="comment"># &gt;&gt;&gt; gtf.getRecordsByID(&quot;NR_024540&quot;, &quot;exon_id&quot;, &quot;NR_024540.5&quot;)</span></span><br><span class="line"><span class="comment">####################################################################</span></span><br><span class="line"></span><br><span class="line"><span class="class"><span class="keyword">class</span> <span class="title">gtfParser</span>:</span></span><br><span class="line">    <span class="function"><span class="keyword">def</span> <span class="title">__init__</span>(<span class="params">self, input_file</span>):</span></span><br><span class="line">        <span class="keyword">import</span> sys</span><br><span class="line">        self.data = &#123;&#125;</span><br><span class="line">        self.<span class="built_in">dict</span> = &#123;&#125;</span><br><span class="line">        self.gene_attributes_dict = &#123;&#125; <span class="comment"># ZZX</span></span><br><span class="line">        self.transcriptID_geneID_dict = &#123;&#125; <span class="comment"># ZZX</span></span><br><span class="line"></span><br><span class="line">        sys.stderr.write(<span class="string">&quot;#####################\nParsing reference gtf file: &quot;</span> +</span><br><span class="line">                         input_file +</span><br><span class="line">                         <span class="string">&#x27;\n#####################\n&#x27;</span>)</span><br><span class="line"></span><br><span class="line">        <span class="keyword">for</span> line <span class="keyword">in</span> <span class="built_in">open</span>(input_file):</span><br><span class="line">            <span class="keyword">if</span> line.startswith(<span class="string">&quot;#&quot;</span>): <span class="keyword">continue</span></span><br><span class="line">            record = line.strip().split(<span class="string">&quot;\t&quot;</span>)</span><br><span class="line">            sequence_name = record[<span class="number">0</span>]</span><br><span class="line">            source = record[<span class="number">1</span>]</span><br><span class="line">            feature = record[<span class="number">2</span>]</span><br><span class="line">            start = <span class="built_in">int</span>(record[<span class="number">3</span>])</span><br><span class="line">            end = <span class="built_in">int</span>(record[<span class="number">4</span>])</span><br><span class="line">            <span class="keyword">if</span> (record[<span class="number">5</span>] != <span class="string">&#x27;.&#x27;</span>):</span><br><span class="line">                score = record[<span class="number">5</span>]</span><br><span class="line">            <span class="keyword">else</span>:</span><br><span class="line">                score = <span class="literal">None</span></span><br><span class="line">            strand = record[<span class="number">6</span>]</span><br><span class="line">            <span class="keyword">if</span> (record[<span class="number">7</span>] != <span class="string">&#x27;.&#x27;</span>):</span><br><span class="line">                frame = record[<span class="number">7</span>]</span><br><span class="line">            <span class="keyword">else</span>:</span><br><span class="line">                frame = <span class="literal">None</span></span><br><span class="line">            attributes = record[<span class="number">8</span>].split(<span class="string">&#x27;;&#x27;</span>) <span class="comment"># please note the separator between each attribute</span></span><br><span class="line">            attributes = [x.strip() <span class="keyword">for</span> x <span class="keyword">in</span> attributes[<span class="number">0</span>:-<span class="number">1</span>]] <span class="comment"># ZZX</span></span><br><span class="line">            attributes = &#123;x.split(<span class="string">&#x27; &#x27;</span>)[<span class="number">0</span>]: x.split(<span class="string">&#x27; &#x27;</span>)[<span class="number">1</span>].strip(<span class="string">&quot;\&quot;&quot;</span>) <span class="keyword">for</span> x <span class="keyword">in</span> attributes <span class="keyword">if</span> <span class="string">&quot; &quot;</span> <span class="keyword">in</span> x&#125; <span class="comment"># ZZX</span></span><br><span class="line"></span><br><span class="line">            <span class="keyword">if</span> <span class="keyword">not</span> (sequence_name <span class="keyword">in</span> self.data): self.data[sequence_name] = []</span><br><span class="line">            alpha = &#123;<span class="string">&#x27;source&#x27;</span>: source, <span class="string">&#x27;feature&#x27;</span>: feature, <span class="string">&#x27;start&#x27;</span>: start, <span class="string">&#x27;end&#x27;</span>: end, <span class="string">&#x27;score&#x27;</span>: score, <span class="string">&#x27;strand&#x27;</span>: strand,</span><br><span class="line">                     <span class="string">&#x27;frame&#x27;</span>: frame&#125;</span><br><span class="line">            <span class="comment"># python 3 .items(), python 2 .iteritems() ZZX</span></span><br><span class="line">            <span class="keyword">for</span> k, v <span class="keyword">in</span> attributes.items(): alpha[k] = v</span><br><span class="line">            self.data[sequence_name].append(alpha)</span><br><span class="line"></span><br><span class="line">        <span class="comment"># ZZX</span></span><br><span class="line">        <span class="keyword">for</span> k, v <span class="keyword">in</span> self.data.items():</span><br><span class="line">            <span class="keyword">for</span> alpha <span class="keyword">in</span> v:</span><br><span class="line">                gene_id = alpha[<span class="string">&quot;gene_id&quot;</span>]</span><br><span class="line">                transcript_id = alpha[<span class="string">&quot;transcript_id&quot;</span>]</span><br><span class="line"></span><br><span class="line">                self.transcriptID_geneID_dict[transcript_id] = gene_id</span><br><span class="line">                <span class="keyword">if</span> gene_id <span class="keyword">in</span> self.gene_attributes_dict.keys():</span><br><span class="line">                    self.gene_attributes_dict[gene_id].append(alpha)</span><br><span class="line">                <span class="keyword">else</span>:</span><br><span class="line">                    self.gene_attributes_dict[gene_id] = <span class="built_in">list</span>()</span><br><span class="line">                    self.gene_attributes_dict[gene_id].append(alpha)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">    <span class="function"><span class="keyword">def</span> <span class="title">getRecordsByID</span>(<span class="params">self, <span class="built_in">id</span>, attType, attValue</span>):</span></span><br><span class="line">        <span class="string">&quot;&quot;&quot;</span></span><br><span class="line"><span class="string">        Gets all the features for a given gene</span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string"></span></span><br><span class="line"><span class="string">        :param id: Identifier of the gene (specified by gene_id) or mRNA (specified by transcript_id)</span></span><br><span class="line"><span class="string">        :param attType: Any attribute list in gtf file.</span></span><br><span class="line"><span class="string">        :param attValue:</span></span><br><span class="line"><span class="string">        :return: A list of features related to ID where restricted by attType and attValue.</span></span><br><span class="line"><span class="string">        &quot;&quot;&quot;</span></span><br><span class="line"></span><br><span class="line">        att_list = []</span><br><span class="line">        <span class="keyword">if</span> <span class="built_in">id</span> <span class="keyword">in</span> self.gene_attributes_dict.keys():</span><br><span class="line">            <span class="keyword">for</span> x <span class="keyword">in</span> self.gene_attributes_dict[<span class="built_in">id</span>]:</span><br><span class="line">                <span class="keyword">if</span> ( attType <span class="keyword">in</span> x.keys() <span class="keyword">and</span>  x[attType] == attValue ):</span><br><span class="line">                    att_info = x</span><br><span class="line">                    att_list.append(att_info)</span><br><span class="line">        <span class="keyword">elif</span> <span class="built_in">id</span> <span class="keyword">in</span> self.transcriptID_geneID_dict.keys():</span><br><span class="line">            <span class="keyword">for</span> x <span class="keyword">in</span> self.gene_attributes_dict[self.transcriptID_geneID_dict[<span class="built_in">id</span>]]:</span><br><span class="line">                <span class="keyword">if</span> ( attType <span class="keyword">in</span> x.keys() <span class="keyword">and</span>  x[attType] == attValue <span class="keyword">and</span> x[<span class="string">&quot;transcript_id&quot;</span>] == <span class="built_in">id</span>):</span><br><span class="line">                    att_info = x</span><br><span class="line">                    att_list.append(att_info)</span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            sys.stderr.write(<span class="string">&quot;Could not find ID &quot;</span>+<span class="built_in">id</span>+<span class="string">&#x27;\n&#x27;</span>)</span><br><span class="line">            sys.exit(<span class="number">1</span>)</span><br><span class="line">        <span class="keyword">return</span> att_list</span><br></pre></td></tr></table></figure>



<p>Code available in GitHub: </p>
<p><a target="_blank" rel="noopener" href="https://github.com/ProfessionalFarmer/lake/blob/master/gtfParser.py">https://github.com/ProfessionalFarmer/lake/blob/master/gtfParser.py</a></p>
<p>#####################################################################<br>#版权所有 转载请告知 版权归作者所有 如有侵权 一经发现 必将追究其法律责任<br>#Author: Jason<br>#####################################################################</p>

    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/Python/" rel="tag"># Python</a>
              <a href="/tags/gtf/" rel="tag"># gtf</a>
              <a href="/tags/code/" rel="tag"># code</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/archives/1136/" rel="prev" title="整合多组学数据的通路富集分析-ActivePathways">
      <i class="fa fa-chevron-left"></i> 整合多组学数据的通路富集分析-ActivePathways
    </a></div>
      <div class="post-nav-item">
    <a href="/archives/1186/" rel="next" title="Fixation index (FST)">
      Fixation index (FST) <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#Usage"><span class="nav-number">1.</span> <span class="nav-text">Usage</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#Example-gtf"><span class="nav-number">2.</span> <span class="nav-text">Example gtf</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#My-code"><span class="nav-number">3.</span> <span class="nav-text">My code</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Jason</p>
  <div class="site-description" itemprop="description"></div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">163</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
        <span class="site-state-item-count">1</span>
        <span class="site-state-item-name">categories</span>
      </div>
      <div class="site-state-item site-state-tags">
        <span class="site-state-item-count">167</span>
        <span class="site-state-item-name">tags</span>
      </div>
  </nav>
</div>



      </div>


<!--google_adsense-->

  <div class="links-of-blogroll-title">
    <i class="fa fa-google"></i> AdSense
    <ins class="adsbygoogle"
        style="display:block"
        data-ad-client="ca-pub-9370593480269435"
        data-ad-slot="8824957305"
        data-ad-format="auto"
        data-full-width-responsive="true"></ins>
    <script>
      (adsbygoogle = window.adsbygoogle || []).push({});
    </script>
  </div>

<!--/google_adsense-->



    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2022</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Jason</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

<script data-ad-client="ca-pub-9370593480269435" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<!-- PGAD -->
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-9370593480269435"
     data-ad-slot="8824957305"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
